<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Fabio Bedin | MS-Unit" />


<title>Integration old analysis T1 vs TME1</title>

<script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.18/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">OC_Patients_comparisons</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/FabioBedin/OC_Nezi">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Integration old analysis T1 vs TME1</h1>
<h4 class="author">Fabio Bedin | MS-Unit</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-06-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>OC_Nezi/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210520code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210520)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210520code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210520)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomFabioBedinOCNezitree5c1eab18835ca28e03437baebdee0b3d37b6a39ctargetblank5c1eab1a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/FabioBedin/OC_Nezi/tree/5c1eab18835ca28e03437baebdee0b3d37b6a39c" target="_blank">5c1eab1</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomFabioBedinOCNezitree5c1eab18835ca28e03437baebdee0b3d37b6a39ctargetblank5c1eab1a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/FabioBedin/OC_Nezi/tree/5c1eab18835ca28e03437baebdee0b3d37b6a39c" target="_blank">5c1eab1</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.RData

Untracked files:
    Untracked:  T1_vs_T2.tex
    Untracked:  analysis/T2_R_vs_T2_S_gene_name.txt
    Untracked:  analysis/test.cys
    Untracked:  code/custom_functions.R
    Untracked:  data/anova_table_t1_t2.txt
    Untracked:  data/anova_table_t1_t2_filt_1500.txt
    Untracked:  data/anova_table_t1_tme1.txt
    Untracked:  data/anova_table_t1_tme1_filt_1500.txt
    Untracked:  data/anova_table_t2_tme2.txt
    Untracked:  data/anova_table_t2_tme2_filt_1500.txt
    Untracked:  data/expdesign_SI_S_R.tsv
    Untracked:  data/expdesign_old_exp.tsv
    Untracked:  data/proteinGroups.txt
    Untracked:  gene_name_BP_anova.txt
    Untracked:  site_libs/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Integration_old_analysis_T1_TME1.Rmd</code>) and HTML (<code>docs/Integration_old_analysis_T1_TME1.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/FabioBedin/OC_Nezi/blob/5c1eab18835ca28e03437baebdee0b3d37b6a39c/analysis/Integration_old_analysis_T1_TME1.Rmd" target="_blank">5c1eab1</a>
</td>
<td>
FabioBedin
</td>
<td>
2021-06-28
</td>
<td>
New analysis with filter at 1500 protein and first test for external imanges
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/FabioBedin/OC_Nezi/5c1eab18835ca28e03437baebdee0b3d37b6a39c/docs/Integration_old_analysis_T1_TME1.html" target="_blank">5c1eab1</a>
</td>
<td>
FabioBedin
</td>
<td>
2021-06-28
</td>
<td>
New analysis with filter at 1500 protein and first test for external imanges
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/FabioBedin/OC_Nezi/533e873d5d0be4b56e44fbc7b0711122fba68f11/docs/Integration_old_analysis_T1_TME1.html" target="_blank">533e873</a>
</td>
<td>
FabioBedin
</td>
<td>
2021-06-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/FabioBedin/OC_Nezi/blob/e35e7be6280a04ef0627bb6011c8b2980e46cb51/analysis/Integration_old_analysis_T1_TME1.Rmd" target="_blank">e35e7be</a>
</td>
<td>
FabioBedin
</td>
<td>
2021-06-17
</td>
<td>
wflow_publish(here::here(c(“analysis/index.Rmd”, “analysis/Integration_old_analysis.Rmd”,
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="integration-old-analysis" class="section level1">
<h1>Integration old analysis</h1>
<hr />
<pre class="r"><code>knitr::opts_chunk$set(fig.align = &quot;center&quot;)
library(&quot;MBQN&quot;)
library(&quot;dplyr&quot;)
library(&quot;tidyr&quot;)
library(&quot;DEP&quot;)
library(&quot;SummarizedExperiment&quot;)
library(&quot;preprocessCore&quot;)
library(&quot;tibble&quot;)
library(&quot;ggplot2&quot;)
library(&quot;enrichR&quot;)
library(&quot;DT&quot;)
library(&quot;stringr&quot;)
library(&quot;patchwork&quot;)
library(&quot;here&quot;)
library(&quot;datapasta&quot;)
library(&quot;visdat&quot;)
library(&quot;naniar&quot;)</code></pre>
<pre class="r"><code>source(here::here(&quot;code/custom_functions.R&quot;))</code></pre>
<p>Summary of previeus results of <strong>unique</strong> proteins:</p>
<pre class="r"><code>knitr::include_graphics(&quot;assets/Nezi_unique_t1_Vs_tme1.png&quot;, error = F)</code></pre>
<p><img src="assets/Nezi_unique_t1_Vs_tme1.png" width="150%" style="display: block; margin: auto;" /></p>
<hr />
<div id="dep-analysis-comparisons" class="section level2 tabset tabset-fade tabset-pills">
<h2>DEP analysis comparisons</h2>
<p>First, I analyzed the data by imputing the missing values. This gave me the possibility to get significant proteins even with more stringent filters: <strong>FDR = 0.05</strong> and <strong>FC = 1</strong>. New results from single patient comparisons are then shown, as well as a direct visualization of how unique proteins in the first analysis behave in this second.</p>
<pre class="r"><code>data &lt;- read.csv(here::here(&quot;data/proteinGroups.txt&quot;), header = TRUE,stringsAsFactors = FALSE, sep = &quot;\t&quot;)

data &lt;- data[data$Reverse != &quot;+&quot; &amp; data$Potential.contaminant != &quot;+&quot; &amp; data$Only.identified.by.site != &quot;+&quot;,]

data_unique &lt;- make_unique(data, &quot;Gene.names&quot;, &quot;Protein.IDs&quot;, delim = &quot;;&quot;)</code></pre>
<pre class="r"><code>expdesign &lt;- read.table(here::here(&quot;data/expdesign_old_exp.tsv&quot;), header = T, stringsAsFactors = F)

cond_1&lt;-&quot;T1_P1&quot;
cond_2&lt;-&quot;TME1_P1&quot;

conditions&lt;-c(cond_1,cond_2)

expdesign &lt;- subset(expdesign, condition %in% conditions)

test&lt;- paste(cond_1,cond_2,sep=&quot;_vs_&quot;)

columns&lt;-match(paste(&quot;LFQ.intensity.&quot;,expdesign$label,sep=&quot;&quot;),colnames(data_unique))

data_se &lt;- make_se(data_unique, columns, expdesign)</code></pre>
<div id="t1_p1_vs_tme1_p1" class="section level3">
<h3>T1_P1_vs_TME1_P1</h3>
<pre class="r"><code>data_filt &lt;- filter_missval(data_se, thr = 1)

set.seed(20210520)
data_imp_man &lt;- impute(data_filt, fun = &quot;man&quot;, shift = 1.8, scale = 0.3)

data_diff &lt;- test_diff_BH(data_imp_man, type = &quot;all&quot;)

dep &lt;- add_rejections(data_diff, alpha = 0.05, lfc = 1)

results&lt;-get_results(dep)</code></pre>
<p>There are <strong>27</strong> significant proteins in <strong>T1_P1_vs_TME1_P1</strong> comparison.</p>
<div id="visualization-of-the-results" class="section level4 tabset">
<h4><strong>Visualization of the results:</strong></h4>
<hr />
<div id="volcano-plot" class="section level5">
<h5>Volcano Plot</h5>
<pre class="r"><code>plot_volcano(dep, contrast=test, add_names=T,label_size=4,adjusted = F)</code></pre>
<p><img src="figure/Integration_old_analysis_T1_TME1.Rmd/V3-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="results-table" class="section level5">
<h5>Results Table</h5>
<pre class="r"><code>results %&gt;% 
  filter(significant) %&gt;%
  select(-significant, -ends_with(&quot;centered&quot;)) %&gt;% 
  pivot_longer(
   cols = ends_with(c(&quot;p.val&quot;, &quot;p.adj&quot;, &quot;significant&quot;, &quot;ratio&quot;)),
   names_to = c(&quot;comparison&quot;, &quot;.value&quot;),
   names_pattern = &quot;(.*_.*_.*_.*_.*)_(.*)&quot;) %&gt;% 
  arrange(name, p.adj) %&gt;% 
  relocate(comparison) %&gt;% 
  select(-comparison) %&gt;% 
  mutate(across(ends_with(c(&quot;p.val&quot;, &quot;p.adj&quot;)), format, scientific = T, digits = 2)) %&gt;% 
  mutate(significant = str_to_title(significant)) %&gt;% 
  DT::datatable(options = list(
  columnDefs = list(list(className = &#39;dt-center&#39;, targets = 1:6)),
  pageLength = 10))</code></pre>
<div id="htmlwidget-351ff91fcdcd073b4a62" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-351ff91fcdcd073b4a62">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27"],["ATAD3B","CD59","CD81","COX4I1","CPNE1","EBNA1BP2","GPX8","HNRNPAB","IGHV3-15","LAMB3","MAL2","NIT2","NPM1","PABPN1","PDIA3","PTGES3","PTGS1","RAB5A","RBM3","RPL18A","S100A1","SEC61B","SLC25A6","SNCG","SNRPD3","SNRPG","SRP9"],["Q5T9A4","E9PNW4","E9PRJ8","P13073","Q99829","H7C2Q8","Q8TED1","D6R9P3","A0A0B4J1V0","Q13751","Q969L2","Q9NQR4","P06748","Q86U42","H7BZJ3","Q15185","A0A087X296","P20339","P98179","M0R117","P23297","P60468","P12236","O76070","P62318","Q49AN9","P49458"],["2.2e-06","2.9e-04","3.0e-04","2.1e-04","4.7e-04","5.8e-04","5.3e-04","1.4e-04","1.5e-04","5.4e-04","6.0e-05","9.4e-05","2.7e-04","3.5e-04","6.8e-05","5.5e-05","4.6e-04","9.8e-05","4.3e-04","5.7e-05","2.8e-04","7.4e-05","2.0e-04","3.1e-04","1.6e-04","6.2e-05","4.8e-04"],["4.9e-03","3.6e-02","3.6e-02","3.3e-02","4.4e-02","4.8e-02","4.6e-02","2.9e-02","2.9e-02","4.6e-02","2.3e-02","2.4e-02","3.6e-02","3.9e-02","2.3e-02","2.3e-02","4.4e-02","2.4e-02","4.4e-02","2.3e-02","3.6e-02","2.3e-02","3.3e-02","3.6e-02","2.9e-02","2.3e-02","4.4e-02"],["True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True"],[9.8,4.6,4.22,4.3,4.11,3.74,-3.99,4.61,-5.09,4.41,4.75,7.87,5.32,4.69,4.88,5.11,4.81,4.84,4.19,5.57,5.9,5.31,5.64,4.53,5.78,6.95,4.26]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>name<\/th>\n      <th>ID<\/th>\n      <th>p.val<\/th>\n      <th>p.adj<\/th>\n      <th>significant<\/th>\n      <th>ratio<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-center","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"pageLength":10,"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="old-plot-unique" class="section level5">
<h5>Old plot + Unique</h5>
<pre class="r"><code>unique_vec &lt;- c(&quot;AARS2&quot;, &quot;ABCA8&quot;, &quot;BAZ1B&quot;, &quot;CCDC58&quot;, &quot;COMT&quot;, &quot;FAF1&quot;, &quot;FUT8&quot;, &quot;GATAD2B&quot;, &quot;INTS5&quot;, &quot;MBLAC2&quot;, &quot;MCCC1&quot;, &quot;NDUFA6&quot;, &quot;NDUFV2&quot;, &quot;NELFB&quot;, &quot;PABPN1&quot;, &quot;PLIN1&quot;, &quot;PLSCR1&quot;, &quot;POLR2E&quot;, &quot;PPIF&quot;, &quot;PTGES2&quot;, &quot;PTGES3&quot;, &quot;PTGS1&quot;, &quot;QTRT1&quot;, &quot;RPL18&quot;, &quot;RPL37A&quot;, &quot;SMNDC1&quot;, &quot;SNRPD3&quot;, &quot;TIMM10&quot;, &quot;TPD52&quot;, &quot;XRCC1&quot;, &quot;CAV1&quot;, &quot;EFEMP1&quot;, &quot;PON1&quot;, &quot;PRR12&quot;, &quot;SNX7&quot;, &quot;STX18&quot;, &quot;ACAD8&quot;, &quot;AGK&quot;, &quot;BRD4&quot;, &quot;CACNA2D1&quot;, &quot;CAPS&quot;, &quot;CCDC58&quot;, &quot;CIRH1A&quot;, &quot;IRGQ&quot;, &quot;LIPE&quot;, &quot;METTL14&quot;, &quot;MFF&quot;, &quot;MRPS24&quot;, &quot;NOL6&quot;, &quot;PPIF&quot;, &quot;PRKCD&quot;, &quot;PRKCI&quot;, &quot;PTGES3&quot;, &quot;RAB18&quot;, &quot;RPL37A&quot;, &quot;SNCG&quot;, &quot;TBC1D15&quot;, &quot;THY1&quot;, &quot;TKTL1&quot;, &quot;TM9SF3&quot;, &quot;UQCC2&quot;, &quot;ZNF185&quot;)

plot_volcano_3(dep, contrast=test, add_names=T,label_size=4,adjusted = F)</code></pre>
<p><img src="figure/Integration_old_analysis_T1_TME1.Rmd/V2-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="unique-table" class="section level5">
<h5>Unique Table</h5>
<pre class="r"><code>results %&gt;%  select(starts_with(&quot;name&quot;) | starts_with(test)) %&gt;% filter(name %in% unique_vec) %&gt;% arrange(across(ends_with(&quot;p.adj&quot;))) %&gt;% mutate(across(2:3, format, scientific = T, digits = 2)) %&gt;% dplyr::rename_with(~ tolower(gsub(paste0(test,&quot;_&quot;), &quot;&quot;, .x)), .cols = starts_with(test)) %&gt;% DT::datatable()</code></pre>
<div id="htmlwidget-182a2e5cf3ae80855303" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-182a2e5cf3ae80855303">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50"],["PTGES3","SNRPD3","SNCG","PABPN1","PTGS1","AARS2","MRPS24","SNX7","MBLAC2","PRKCD","NDUFA6","BRD4","ZNF185","BAZ1B","MFF","MCCC1","AGK","PON1","PPIF","TKTL1","PTGES2","RPL37A","IRGQ","CACNA2D1","FAF1","CIRH1A","INTS5","CCDC58","PLIN1","QTRT1","ACAD8","PRR12","LIPE","PRKCI","TM9SF3","UQCC2","CAV1","RPL18","NELFB","RAB18","TPD52","THY1","NDUFV2","METTL14","NOL6","EFEMP1","PLSCR1","XRCC1","CAPS","COMT"],["5.5e-05","1.6e-04","3.1e-04","3.5e-04","4.6e-04","8.1e-04","8.9e-04","1.0e-03","1.4e-03","1.6e-03","3.3e-03","4.4e-03","7.3e-03","8.5e-03","1.2e-02","1.2e-02","1.4e-02","1.4e-02","1.5e-02","1.5e-02","2.0e-02","2.7e-02","2.7e-02","3.0e-02","3.1e-02","3.6e-02","5.7e-02","5.8e-02","6.7e-02","7.5e-02","7.6e-02","8.2e-02","8.4e-02","8.4e-02","9.4e-02","9.5e-02","1.1e-01","1.1e-01","1.3e-01","1.3e-01","1.4e-01","1.4e-01","2.0e-01","2.1e-01","2.9e-01","3.0e-01","3.0e-01","3.0e-01","5.5e-01","8.3e-01"],["2.3e-02","2.9e-02","3.6e-02","3.9e-02","4.4e-02","5.4e-02","5.4e-02","5.6e-02","5.8e-02","5.9e-02","7.4e-02","8.3e-02","1.0e-01","1.1e-01","1.3e-01","1.3e-01","1.3e-01","1.3e-01","1.4e-01","1.4e-01","1.4e-01","1.6e-01","1.6e-01","1.7e-01","1.7e-01","1.9e-01","2.3e-01","2.4e-01","2.5e-01","2.6e-01","2.7e-01","2.7e-01","2.7e-01","2.7e-01","2.9e-01","2.9e-01","3.1e-01","3.2e-01","3.3e-01","3.3e-01","3.4e-01","3.4e-01","4.1e-01","4.2e-01","5.0e-01","5.1e-01","5.2e-01","5.2e-01","7.2e-01","9.1e-01"],[true,true,true,true,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],[5.11,5.78,4.53,4.69,4.81,4.44,3.7,-3.49,3.8,3.85,3.08,5.22,3.05,3.48,3.62,3.2,3.69,-2.71,2.88,4.51,2.59,2.92,2.24,3,3.08,2.12,2.36,2.05,2.01,2.46,1.66,-1.61,2.18,1.73,1.79,1.87,2.11,1.67,1.47,2.3,1.71,1.51,1.37,2.03,1.06,-1.49,1.36,0.966,0.65,0.241]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>name<\/th>\n      <th>p.val<\/th>\n      <th>p.adj<\/th>\n      <th>significant<\/th>\n      <th>ratio<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":5},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>expdesign &lt;- read.table(here::here(&quot;data/expdesign_old_exp.tsv&quot;), header = T, stringsAsFactors = F)

cond_1&lt;-&quot;T1_P2&quot;
cond_2&lt;-&quot;TME1_P2&quot;

conditions&lt;-c(cond_1,cond_2)

expdesign &lt;- subset(expdesign, condition %in% conditions)

test&lt;- paste(cond_1,cond_2,sep=&quot;_vs_&quot;)

columns&lt;-match(paste(&quot;LFQ.intensity.&quot;,expdesign$label,sep=&quot;&quot;),colnames(data_unique))

data_se &lt;- make_se(data_unique, columns, expdesign)</code></pre>
</div>
</div>
</div>
<div id="t1_p2_vs_tme1_p2" class="section level3">
<h3>T1_P2_vs_TME1_P2</h3>
<pre class="r"><code>data_filt &lt;- filter_missval(data_se, thr = 1)

set.seed(20210520)
data_imp_man &lt;- impute(data_filt, fun = &quot;man&quot;, shift = 1.8, scale = 0.3)

data_diff &lt;- test_diff_BH(data_imp_man, type = &quot;all&quot;)

dep &lt;- add_rejections(data_diff, alpha = 0.05, lfc = 1)

results&lt;-get_results(dep)</code></pre>
<p>There are <strong>162</strong> significant proteins in <strong>T1_P2_vs_TME1_P2</strong> comparison.</p>
<div id="visualization-of-the-results-1" class="section level4 tabset">
<h4><strong>Visualization of the results:</strong></h4>
<hr />
<div id="volcano-plot-1" class="section level5">
<h5>Volcano Plot</h5>
<pre class="r"><code>plot_volcano(dep, contrast=test, add_names=T,label_size=4,adjusted = F)</code></pre>
<p><img src="figure/Integration_old_analysis_T1_TME1.Rmd/V31-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="results-table-1" class="section level5">
<h5>Results Table</h5>
<pre class="r"><code>results %&gt;% 
  filter(significant) %&gt;%
  select(-significant, -ends_with(&quot;centered&quot;)) %&gt;% 
  pivot_longer(
   cols = ends_with(c(&quot;p.val&quot;, &quot;p.adj&quot;, &quot;significant&quot;, &quot;ratio&quot;)),
   names_to = c(&quot;comparison&quot;, &quot;.value&quot;),
   names_pattern = &quot;(.*_.*_.*_.*_.*)_(.*)&quot;) %&gt;% 
  arrange(name, p.adj) %&gt;% 
  relocate(comparison) %&gt;% 
  select(-comparison) %&gt;% 
  mutate(across(ends_with(c(&quot;p.val&quot;, &quot;p.adj&quot;)), format, scientific = T, digits = 2)) %&gt;% 
  mutate(significant = str_to_title(significant)) %&gt;% 
  DT::datatable(options = list(
  columnDefs = list(list(className = &#39;dt-center&#39;, targets = 1:6)),
  pageLength = 10))</code></pre>
<div id="htmlwidget-bf470b789b28d3d7f57f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-bf470b789b28d3d7f57f">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162"],["A0A2R8YHD2","AARS2","ABCA8","ACAD8","ACTC1","AGK","AHDC1","ANP32B","ANP32E","APOH","ASRGL1","BASP1","BGN.1","BTF3L4","C8orf82","CAB39L","CACNA2D1","CAPS","CCDC58","CDC42","CEP170","CFB","CFH","CILP","CIRH1A","CLDN3","COL12A1","COL14A1","COL1A1","COL1A2","COL3A1","COL6A1","COL6A2","COL6A3","COL6A3.1","COPG2","CPNE1","CPSF7","DNAJC5","DNPH1","DRAP1","ECHDC3","EEFSEC","EIF3I","EPPK1","FAM136A","FAM83H","FBLN1","FDX1","FN1","GALNT5","GATC","GC","H2AFY","HNRNPD.1","HNRNPDL.1","HNRNPK.1","HSPE1","IDH1","IGHV1OR15-1","IGHV3-15","IGKC","IGKV3OR2-268","IMMT","ITIH1","JCHAIN","KHDRBS1","KIAA2013","LGALS1","LLGL2","LMCD1","LMOD1","LRRFIP2","LSM3","LUM","MAGEA4","MAPK1","MAPK9","MBLAC2","METTL14","MRI1","MRPL41","MRPL49","MVD","MXRA7","MYL9","MYO18A","NCF2","NDUFA6","NDUFAF5","NELFB","NFU1","NIPSNAP3A","NONO","NPTN","NUBP1","NUP188","P0DP04","PABPN1","PARD3B","PARP10","PDHX","PDS5A","PGAM5","PLRG1","PLS1","PLSCR1","POLDIP3","PON1","POSTN.1","PPA1","PPIF","PPIH","PPP1CA","PRKCD","PTGES3","PYCARD","RAB18","RAB43","RABL6","RALBP1","RANBP1","RBM27","RMND1","S100A1","SERPINA3","SERPINA7","SHMT2","SIGLEC16","SLC25A10","SLC2A1","SLC9A3R2","SMNDC1","SNCG","SNRPD3","SNRPG","SNTB1","SNX7","SORBS1","SPINT2","STMN1","STX18","TBCD","TGFBI","TKTL1","TM9SF2","TM9SF3","TP53RK","TPM1","TPM1.1","TPM2","TRAPPC12","UBE2C","UQCC2","VCPIP1","VDAC3","VPS13C","WISP2","XPO4","YIF1B","ZC3H18","ZNF185"],["A0A2R8YHD2","Q5JTZ9","A0A0A0MSU4","Q9UKU7","P68032","E9PC15","Q5TGY3","Q92688","Q9BTT0","P02749","Q7L266","P80723","P21810","Q96K17","H0YF29","Q9H9S4","P54289","K7ES72","C9JQ41","P60953","H0Y2V6","B4E1Z4","P08603","O75339","H3BSH7","O15551","D6RGG3","Q05707","P02452","A0A087WTA8","P02461","A0A087X0S5","P12110","E7ENL6","P12111","Q9UBF2","Q99829","F5H669","Q9H3Z4","H0Y8X4","E9PNC7","Q96DC8","P57772","Q13347","A0A075B730","Q96C01","Q6ZRV2","B1AHL2","P10109","P02751","Q7Z7M9","H0YIV9","D6RF35","O75367","Q14103","A0A087WUK2","Q5T6W2","P61604","O75874","A0A075B7D0","A0A0B4J1V0","P01834","P01624","C9J406","P19827","D6RHJ6","Q07666","Q8IYS2","P09382","J3QRV5","Q9NZU5","P29536","Q9Y608","P62310","P51884","P43358","P28482","J3KNK1","Q68D91","Q9HCE5","Q9BV20","Q8IXM3","Q13405","P53602","P84157","P24844","A0A0D9SFK2","P19878","R4GN43","B3KR61","A0A5H1ZRP4","Q9UMS0","Q9UFN0","Q15233","Q9Y639","P53384","Q5SRE5","P0DP04","Q86U42","A0A087WYL6","E9PK67","O00330","Q29RF7","Q96HS1","O43660","Q14651","C9J7K9","F6VRR5","P27169","Q15063","Q15181","P30405","C9JQD4","P62136","Q05655","Q15185","Q9ULZ3","Q9NP72","Q86YS6","Q3YEC7","Q15311","P43487","Q9P2N5","A0A2R8Y4P5","P23297","P01011","P05543","P34897","A6NMB1","Q9UBX3","P11166","Q15599","O75940","O76070","P62318","Q49AN9","Q13884","Q9UNH6","Q9BX66","K7EM91","P16949","Q9P2W9","J3KR97","Q15582","P51854","Q99805","Q9HD45","Q96S44","A0A0S2Z4G6","H7BYY1","Q5TCU3","Q8WVT3","O00762","Q9BRT2","Q96JH7","Q9Y277","Q709C8","O76076","F2Z2X4","Q5BJH7","E7ERS3","O15231"],["1.7e-03","3.9e-05","5.5e-05","2.7e-03","1.2e-03","1.3e-04","1.1e-04","1.5e-03","7.1e-04","2.6e-03","7.2e-04","2.1e-03","2.0e-04","1.5e-03","2.6e-03","1.5e-03","1.5e-04","2.1e-03","2.7e-05","6.6e-04","6.1e-07","3.1e-03","1.7e-03","1.1e-03","1.7e-05","2.1e-03","7.3e-04","9.4e-04","7.4e-05","2.8e-04","6.5e-04","2.0e-03","1.5e-03","1.6e-03","3.2e-04","5.2e-04","1.6e-03","2.3e-03","2.5e-03","2.8e-04","1.1e-03","4.1e-05","1.2e-04","2.9e-03","2.8e-03","4.5e-04","9.2e-04","1.4e-04","7.6e-04","2.8e-04","5.5e-07","9.1e-04","9.8e-04","1.6e-03","9.3e-04","1.4e-04","1.9e-03","1.2e-03","1.6e-03","1.6e-04","2.6e-05","2.4e-03","8.0e-04","9.9e-04","1.0e-03","8.6e-04","1.4e-03","5.3e-04","2.6e-03","4.1e-04","2.3e-03","2.6e-03","2.7e-03","6.8e-04","1.9e-05","7.9e-04","1.6e-03","9.9e-05","7.6e-05","1.0e-03","4.9e-04","2.0e-03","2.3e-03","1.7e-04","1.8e-03","1.4e-05","3.8e-04","1.5e-03","2.8e-03","1.9e-04","2.2e-03","6.6e-04","3.8e-05","2.9e-03","2.2e-03","2.1e-03","2.5e-03","1.1e-04","5.6e-05","5.9e-04","3.0e-03","1.2e-03","2.3e-03","2.1e-03","5.3e-04","1.2e-03","2.9e-04","1.9e-03","5.3e-04","1.7e-03","2.0e-03","2.4e-05","1.8e-03","5.8e-04","1.1e-03","4.5e-05","2.1e-03","1.2e-04","1.9e-08","4.5e-04","1.1e-03","2.0e-05","2.5e-03","2.4e-04","3.9e-04","2.0e-03","2.1e-03","1.7e-03","1.9e-03","2.1e-03","9.4e-04","1.6e-04","2.3e-03","2.2e-05","2.8e-07","4.3e-04","3.0e-03","1.1e-04","7.1e-06","1.5e-03","5.4e-04","8.1e-05","7.0e-04","1.9e-03","1.9e-03","2.8e-03","1.4e-03","1.8e-03","5.1e-06","4.1e-04","2.2e-03","4.9e-05","7.8e-04","6.2e-04","6.7e-04","2.9e-03","2.9e-03","1.3e-03","2.2e-04","1.1e-03","1.9e-03","3.3e-04"],["4.0e-02","6.3e-03","7.0e-03","4.7e-02","3.4e-02","1.1e-02","1.0e-02","3.8e-02","2.6e-02","4.6e-02","2.6e-02","4.2e-02","1.3e-02","3.8e-02","4.5e-02","3.8e-02","1.2e-02","4.2e-02","5.1e-03","2.6e-02","4.0e-04","4.9e-02","3.9e-02","3.3e-02","5.1e-03","4.2e-02","2.6e-02","3.0e-02","8.6e-03","1.7e-02","2.6e-02","4.1e-02","3.8e-02","3.9e-02","1.8e-02","2.4e-02","3.9e-02","4.3e-02","4.5e-02","1.7e-02","3.2e-02","6.3e-03","1.0e-02","4.8e-02","4.8e-02","2.2e-02","3.0e-02","1.1e-02","2.7e-02","1.7e-02","4.0e-04","3.0e-02","3.1e-02","3.9e-02","3.0e-02","1.1e-02","4.1e-02","3.4e-02","3.9e-02","1.2e-02","5.1e-03","4.3e-02","2.8e-02","3.1e-02","3.1e-02","2.9e-02","3.7e-02","2.4e-02","4.5e-02","2.1e-02","4.3e-02","4.5e-02","4.7e-02","2.6e-02","5.1e-03","2.8e-02","3.9e-02","1.0e-02","8.6e-03","3.1e-02","2.3e-02","4.2e-02","4.3e-02","1.2e-02","4.1e-02","5.1e-03","2.0e-02","3.8e-02","4.8e-02","1.3e-02","4.2e-02","2.6e-02","6.3e-03","4.8e-02","4.2e-02","4.2e-02","4.4e-02","1.0e-02","7.0e-03","2.5e-02","4.9e-02","3.4e-02","4.3e-02","4.2e-02","2.4e-02","3.4e-02","1.7e-02","4.1e-02","2.4e-02","3.9e-02","4.1e-02","5.1e-03","4.1e-02","2.5e-02","3.2e-02","6.5e-03","4.2e-02","1.0e-02","4.9e-05","2.2e-02","3.2e-02","5.1e-03","4.5e-02","1.5e-02","2.1e-02","4.1e-02","4.2e-02","3.9e-02","4.1e-02","4.2e-02","3.0e-02","1.2e-02","4.3e-02","5.1e-03","3.6e-04","2.1e-02","4.9e-02","1.0e-02","3.1e-03","3.8e-02","2.4e-02","8.8e-03","2.6e-02","4.1e-02","4.1e-02","4.8e-02","3.8e-02","4.1e-02","2.7e-03","2.1e-02","4.2e-02","6.8e-03","2.8e-02","2.6e-02","2.6e-02","4.8e-02","4.8e-02","3.4e-02","1.4e-02","3.3e-02","4.1e-02","1.8e-02"],["True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True"],[-3.22,6.5,4.6,3.37,-4.81,4.12,-4.43,3.66,4.31,-3.43,4.24,-3.42,-6.33,4.9,3.88,4.23,3.91,3,4.94,-3.99,7.28,-4.11,-4.01,-3.11,5.06,-3.93,-7.14,-5.48,-6.27,-6.72,-5.61,-4.22,-4.68,-4.9,-6.69,3.54,3.75,3.8,-3.07,4.32,3.31,5.77,4.3,-4.48,3.58,3.94,-3.23,-5.23,4.11,-6.24,-8.44,4.33,-4.93,4.86,4.56,5.14,4.64,3.68,3.83,-5.01,-5.14,-3.9,-3.56,3.31,-4.62,-3.73,5.15,3.77,-3.37,-4.41,-4.14,-3.2,-4.45,-3.55,-7.34,4.87,4.1,-4.06,4.58,3.76,4.07,3.02,3.24,3.85,5.07,-5.32,-4.31,5.68,3.32,-3.79,3.73,3.28,5.38,4.05,3.3,3.51,5.05,-4.2,5.45,-3.94,-3.46,3.67,7.26,3.62,3.93,3.9,4.34,3.59,-3.69,-3.9,3.79,5.29,4.24,4.06,3.8,5.35,3.97,4.96,-10.9,3.53,-3.81,5.17,3.38,11.1,5.58,-3.71,-3.01,3.51,-5.84,3.09,-3.58,3.98,2.78,5.22,7.55,5.59,-3.99,-4.03,-5.95,3.16,3.99,-4.96,7.69,-5.32,6.54,3.84,3.79,3.4,-7.48,-5.78,-4.27,5.17,-3.2,3.37,3.98,5.04,3.21,3.92,-4.03,6.49,7.86,3.75]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>name<\/th>\n      <th>ID<\/th>\n      <th>p.val<\/th>\n      <th>p.adj<\/th>\n      <th>significant<\/th>\n      <th>ratio<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-center","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"pageLength":10,"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="old-plot-unique-1" class="section level5">
<h5>Old plot + Unique</h5>
<pre class="r"><code>unique_vec &lt;- c(&quot;AARS2&quot;, &quot;ABCA8&quot;, &quot;BAZ1B&quot;, &quot;CCDC58&quot;, &quot;COMT&quot;, &quot;FAF1&quot;, &quot;FUT8&quot;, &quot;GATAD2B&quot;, &quot;INTS5&quot;, &quot;MBLAC2&quot;, &quot;MCCC1&quot;, &quot;NDUFA6&quot;, &quot;NDUFV2&quot;, &quot;NELFB&quot;, &quot;PABPN1&quot;, &quot;PLIN1&quot;, &quot;PLSCR1&quot;, &quot;POLR2E&quot;, &quot;PPIF&quot;, &quot;PTGES2&quot;, &quot;PTGES3&quot;, &quot;PTGS1&quot;, &quot;QTRT1&quot;, &quot;RPL18&quot;, &quot;RPL37A&quot;, &quot;SMNDC1&quot;, &quot;SNRPD3&quot;, &quot;TIMM10&quot;, &quot;TPD52&quot;, &quot;XRCC1&quot;, &quot;CAV1&quot;, &quot;EFEMP1&quot;, &quot;PON1&quot;, &quot;PRR12&quot;, &quot;SNX7&quot;, &quot;STX18&quot;, &quot;IGHV3-15&quot;)

plot_volcano_3(dep, contrast=test, add_names=T,label_size=4,adjusted = F)</code></pre>
<p><img src="figure/Integration_old_analysis_T1_TME1.Rmd/V4-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="unique-table-1" class="section level5">
<h5>Unique Table</h5>
<pre class="r"><code>results %&gt;%  select(starts_with(&quot;name&quot;) | starts_with(test)) %&gt;% filter(name %in% unique_vec) %&gt;% arrange(across(ends_with(&quot;p.adj&quot;))) %&gt;% mutate(across(2:3, format, scientific = T, digits = 2)) %&gt;% dplyr::rename_with(~ tolower(gsub(paste0(test,&quot;_&quot;), &quot;&quot;, .x)), .cols = starts_with(test)) %&gt;% DT::datatable()</code></pre>
<div id="htmlwidget-df6d39fe7eb8d564bdfa" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-df6d39fe7eb8d564bdfa">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34"],["SNRPD3","CCDC58","IGHV3-15","PPIF","AARS2","PTGES3","ABCA8","PABPN1","MBLAC2","STX18","SNX7","PLSCR1","PON1","NELFB","SMNDC1","NDUFA6","MCCC1","EFEMP1","TPD52","PLIN1","RPL37A","PTGS1","GATAD2B","TIMM10","PRR12","FAF1","CAV1","PTGES2","QTRT1","BAZ1B","COMT","POLR2E","RPL18","NDUFV2"],["2.8e-07","2.7e-05","2.6e-05","2.4e-05","3.9e-05","4.5e-05","5.5e-05","5.6e-05","7.6e-05","8.1e-05","1.1e-04","2.9e-04","5.3e-04","2.2e-03","2.3e-03","2.8e-03","6.6e-03","6.6e-03","7.0e-03","7.4e-03","7.4e-03","1.1e-02","1.3e-02","1.6e-02","2.7e-02","4.5e-02","5.0e-02","1.2e-01","1.3e-01","1.5e-01","3.5e-01","4.2e-01","4.3e-01","7.4e-01"],["3.6e-04","5.1e-03","5.1e-03","5.1e-03","6.3e-03","6.5e-03","7.0e-03","7.0e-03","8.6e-03","8.8e-03","1.0e-02","1.7e-02","2.4e-02","4.2e-02","4.3e-02","4.8e-02","6.8e-02","6.9e-02","7.1e-02","7.2e-02","7.2e-02","9.3e-02","9.8e-02","1.1e-01","1.4e-01","1.8e-01","1.9e-01","3.2e-01","3.3e-01","3.6e-01","5.9e-01","6.4e-01","6.5e-01","8.5e-01"],[true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],[7.55,4.94,-5.14,5.29,6.5,5.35,4.6,5.45,4.58,-4.96,-4.03,4.34,-3.69,3.73,2.78,3.32,3.29,-3.66,2.6,3,3.33,3.54,2.63,2.93,-3.16,2.76,-1.75,1.65,2.21,1.86,1.16,0.785,0.917,-0.285]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>name<\/th>\n      <th>p.val<\/th>\n      <th>p.adj<\/th>\n      <th>significant<\/th>\n      <th>ratio<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":5},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>expdesign &lt;- read.table(here::here(&quot;data/expdesign_old_exp.tsv&quot;), header = T, stringsAsFactors = F)

cond_1&lt;-&quot;T1_P3&quot;
cond_2&lt;-&quot;TME1_P3&quot;

conditions&lt;-c(cond_1,cond_2)

expdesign &lt;- subset(expdesign, condition %in% conditions)

test&lt;- paste(cond_1,cond_2,sep=&quot;_vs_&quot;)

columns&lt;-match(paste(&quot;LFQ.intensity.&quot;,expdesign$label,sep=&quot;&quot;),colnames(data_unique))

data_se &lt;- make_se(data_unique, columns, expdesign)</code></pre>
</div>
</div>
</div>
<div id="t1_p3_vs_tme1_p3" class="section level3">
<h3>T1_P3_vs_TME1_P3</h3>
<pre class="r"><code>data_filt &lt;- filter_missval(data_se, thr = 1)

set.seed(20210520)
data_imp_man &lt;- impute(data_filt, fun = &quot;man&quot;, shift = 1.8, scale = 0.3)

data_diff &lt;- test_diff_BH(data_imp_man, type = &quot;all&quot;)

dep &lt;- add_rejections(data_diff, alpha = 0.05, lfc = 1)

results&lt;-get_results(dep)</code></pre>
<p>There are <strong>199</strong> significant proteins in <strong>T1_P3_vs_TME1_P3</strong> comparison.</p>
<div id="visualization-of-the-results-2" class="section level4 tabset">
<h4><strong>Visualization of the results:</strong></h4>
<hr />
<div id="volcano-plot-2" class="section level5">
<h5>Volcano Plot</h5>
<pre class="r"><code>plot_volcano(dep, contrast=test, add_names=T,label_size=4,adjusted = F)</code></pre>
<p><img src="figure/Integration_old_analysis_T1_TME1.Rmd/V32-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="results-table-2" class="section level5">
<h5>Results Table</h5>
<pre class="r"><code>results %&gt;% 
  filter(significant) %&gt;%
  select(-significant, -ends_with(&quot;centered&quot;)) %&gt;% 
  pivot_longer(
   cols = ends_with(c(&quot;p.val&quot;, &quot;p.adj&quot;, &quot;significant&quot;, &quot;ratio&quot;)),
   names_to = c(&quot;comparison&quot;, &quot;.value&quot;),
   names_pattern = &quot;(.*_.*_.*_.*_.*)_(.*)&quot;) %&gt;% 
  arrange(name, p.adj) %&gt;% 
  relocate(comparison) %&gt;% 
  select(-comparison) %&gt;% 
  mutate(across(ends_with(c(&quot;p.val&quot;, &quot;p.adj&quot;)), format, scientific = T, digits = 2)) %&gt;% 
  mutate(significant = str_to_title(significant)) %&gt;% 
  DT::datatable(options = list(
  columnDefs = list(list(className = &#39;dt-center&#39;, targets = 1:6)),
  pageLength = 10))</code></pre>
<div id="htmlwidget-a6309664db19a50210e2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a6309664db19a50210e2">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199"],["A2M","AARS","ABHD10","APOA2","ARF6","ARHGDIA","ARL6IP5","ARMC10","ASMTL","ASNS","BANF1","BGN.1","C20orf27","C3.1","C4B","C9","CA1","CA2","CCDC58","CCDC80","CD47","CDH11","CDH2","CDV3","CEP170","CFB","CFH","CHCHD6","CKAP4","CKMT1A","CLASP1","COL12A1","COL18A1","COL1A1","COL1A2","COL3A1","COL4A2","COL6A1","COL6A3","COL6A3.1","COX6C","CPSF6","CRIP1","CRIP2","CROCC","CRYAB","CS","CST1","CTHRC1","CTSC","CUL1","CYCS","DCTN6","DDX23","DDX58","DERL1","DES","DNPEP","ECHDC3","EIF3D","ELAVL1","EMILIN1","EPB41","EPS8L2","ETFB","FAF1","FBLIM1","FBLN1","FGA","FGB","FGG","FKBP8","FN1","FNTA","FUT8","GBAS","GLG1","GM2A","GOLGA4","GPX7","H1FX","H2AFY2","HBB","HBD","HBG1","hCG_2014768","HIST2H2AC","HK2","HMGB1","HMGB2","HMGB3","HMGCL","HNRNPA2B1","HNRNPAB","HNRNPD.1","HNRNPDL.1","HNRNPH1","HNRNPK.1","HPCAL1","HPX","HSPA14","HSPE1","IGHA1","IGHG1","IGHG3","IGHV3-15","IGHV3-21","IGKC","IGKV2D-28","IGKV3OR2-268","ITIH1","ITIH2","KATNAL2","KCTD12","KDELC2","KLKB1","KRT18","LAMB1","LRRFIP2","LUM","LXN","LZIC","MAOA","MARCKS","MBLAC2","MGST3","MRPL15","MRPS5","MTA2","MTPAP","NCOA5","NDUFA6","NDUFB4","NHP2","NNT","NUP188","OCIAD1","OLFML3","P4HA2","PGAM5","PIP","POSTN.1","PPA1","PPIA","PPP1CA","PPP3CA","PRDX3","PRDX5","PRR12","PSMC1","PTGS1","PTPRA","QTRT1","RALBP1","RANBP1","RBM22","RBM27","RBM4","ROR1","SDCBP.1","SERPINA1.1","SERPINF1","SERPINF2","SFPQ","SIGLEC16","SLC12A4","SLC25A3","SLC34A2","SLC35E1","SLC4A1","SMAD2","SNRPD3","SNRPG","SNTB1","SOD1","SORBS1","SQRDL","SRM","SSR1","STK24","STX18","SUMO3","SUPT6H","TBCD","TBCE","THY1","THYN1","TM9SF2","TMEM205","TNC.1","TPM1","TRIM25","TRIM28","TUBGCP2","UBTF","UCHL1","UQCRC1","UTS2","VCAN"],["P01023","P49588","Q9NUJ1","V9GYM3","P62330","P52565","O75915","Q8N2F6","O95671","P08243","O75531","P21810","Q9GZN8","P01024","P0C0L5","P02748","E5RHP7","P00918","C9JQ41","Q76M96","Q08722","P55287","C9J126","Q9UKY7","H0Y2V6","B4E1Z4","P08603","J3QTB2","Q07065","P12532","F8WA11","D6RGG3","P39060","P02452","A0A087WTA8","P02461","P08572","A0A087X0S5","E7ENL6","P12111","P09669","F8WJN3","P50238","P52943","Q5TZA2","P02511","O75390","P01037","Q96CG8","P53634","A0A590UJ50","C9JFR7","O00399","Q9BUQ8","O95786","Q9BUN8","P17661","E5RIA4","Q96DC8","O15371","Q15717","Q9Y6C2","P11171","Q9H6S3","P38117","Q9UNN5","Q8WUP2","B1AHL2","P02671","P02675","P02679","Q14318","P02751","P49354","Q9BYC5","O75323","Q92896","P17900","H0Y6I0","Q96SL4","Q92522","Q9P0M6","P68871","P02042","P69891","A0A024R1R8","Q16777","P52789","Q5T7C4","P26583","E9PES6","P35914","P22626","D6R9P3","Q14103","A0A087WUK2","P31943","Q5T6W2","P37235","P02790","Q0VDF9","P61604","A0A286YEY1","P01857","A0A4W9A917","A0A0B4J1V0","P01780","P01834","A0A5H1ZRS2","P01624","P19827","Q5T985","K7EM02","Q96CX2","Q7Z4H8","P03952","P05783","P07942","Q9Y608","P51884","Q9BS40","Q8WZA0","P21397","P29966","Q68D91","O14880","Q9P015","P82675","O94776","Q9NVV4","Q9HCD5","R4GN43","O95168","Q9NX24","Q13423","Q5SRE5","Q9NX40","Q9NRN5","O15460","Q96HS1","P12273","Q15063","Q15181","P62937","P62136","Q08209","P30048","P30044","Q9ULL5","P62191","A0A087X296","P18433","Q9BXR0","Q15311","P43487","Q9NW64","Q9P2N5","Q9BWF3","A0A0A0MTJ5","O00560","P01009","P36955","P08697","P23246","A6NMB1","I3L4N6","F8VVM2","O95436","Q96K37","P02730","Q15796","P62318","Q49AN9","Q13884","P00441","Q9BX66","Q9Y6N5","P19623","C9J3L8","B4DR80","Q9P2W9","A8MUA9","Q7KZ85","J3KR97","Q15813","E9PIM6","Q9P016","Q99805","K7ELQ9","P24821","A0A0S2Z4G6","Q14258","Q13263","Q9BSJ2","E9PKP7","P09936","P31930","Q5H8X8","P13611"],["3.0e-04","2.8e-03","4.2e-03","7.6e-04","1.1e-03","7.4e-05","3.6e-03","3.9e-03","6.2e-04","3.9e-03","4.2e-03","1.4e-04","3.8e-03","1.1e-03","5.4e-04","9.1e-05","1.9e-04","1.6e-03","1.9e-04","8.1e-07","9.7e-05","8.5e-04","9.8e-04","2.6e-03","2.6e-03","2.1e-04","2.2e-03","2.5e-03","3.4e-03","4.0e-04","4.4e-03","1.1e-04","8.7e-04","1.2e-05","1.7e-06","4.1e-03","6.0e-04","1.4e-03","3.2e-05","2.4e-04","3.9e-03","1.3e-03","4.4e-03","5.7e-04","2.8e-05","1.7e-04","2.9e-03","5.8e-05","2.9e-03","3.6e-03","1.0e-04","4.8e-04","1.8e-03","4.9e-03","5.5e-04","4.5e-03","1.5e-03","2.1e-03","8.4e-04","3.1e-03","1.6e-03","4.8e-03","2.5e-03","5.1e-03","3.6e-03","1.7e-05","2.4e-03","6.6e-04","9.1e-05","5.6e-06","2.9e-03","2.0e-03","2.0e-03","1.5e-04","2.0e-03","3.6e-03","2.1e-03","3.0e-03","2.1e-03","4.7e-03","1.8e-03","7.7e-04","1.6e-03","1.1e-04","2.1e-04","3.6e-03","4.8e-03","3.1e-03","1.3e-03","9.2e-05","6.0e-04","4.6e-03","3.2e-03","9.4e-04","4.1e-03","5.3e-04","7.4e-04","2.0e-04","1.2e-04","2.6e-05","1.4e-03","2.6e-03","3.0e-04","1.1e-03","3.0e-03","2.2e-04","6.8e-04","6.0e-04","1.6e-03","2.6e-03","1.3e-04","2.3e-03","1.6e-03","2.2e-03","9.6e-04","1.2e-03","1.6e-03","4.4e-03","4.8e-05","1.2e-03","4.9e-03","8.9e-04","5.1e-04","3.4e-03","3.2e-03","1.8e-03","2.9e-04","2.9e-03","3.5e-03","5.1e-03","6.0e-04","3.4e-03","1.1e-03","1.2e-03","5.0e-03","1.1e-03","1.3e-04","3.7e-03","1.6e-04","4.3e-03","4.8e-04","7.8e-04","4.5e-03","5.0e-03","5.5e-04","3.1e-03","2.1e-04","5.4e-04","1.2e-04","2.9e-03","1.6e-04","4.8e-03","1.8e-04","1.5e-04","1.3e-03","8.5e-04","1.4e-03","4.6e-03","3.6e-03","3.0e-03","9.2e-04","4.3e-04","4.0e-03","2.0e-03","2.1e-04","4.0e-03","5.0e-04","2.8e-03","1.1e-04","1.3e-03","7.8e-04","8.9e-06","1.9e-04","1.1e-03","7.7e-05","4.1e-03","4.8e-03","4.3e-03","4.2e-03","1.5e-03","6.6e-04","2.4e-03","4.3e-03","7.6e-04","2.0e-03","3.0e-03","1.2e-03","3.9e-03","2.8e-03","1.4e-04","4.7e-03","1.5e-03","3.1e-03","8.7e-09","2.2e-03","8.5e-04","1.9e-04","1.6e-03","1.0e-03"],["1.2e-02","4.0e-02","4.6e-02","2.1e-02","2.4e-02","9.6e-03","4.3e-02","4.5e-02","1.9e-02","4.5e-02","4.6e-02","9.7e-03","4.5e-02","2.3e-02","1.8e-02","9.6e-03","9.7e-03","2.8e-02","9.7e-03","7.8e-04","9.6e-03","2.1e-02","2.3e-02","3.8e-02","3.8e-02","9.7e-03","3.5e-02","3.7e-02","4.3e-02","1.6e-02","4.7e-02","9.6e-03","2.1e-02","3.9e-03","1.1e-03","4.6e-02","1.8e-02","2.7e-02","6.2e-03","1.0e-02","4.5e-02","2.5e-02","4.7e-02","1.8e-02","6.2e-03","9.7e-03","4.0e-02","9.4e-03","4.0e-02","4.3e-02","9.6e-03","1.8e-02","3.0e-02","4.9e-02","1.8e-02","4.7e-02","2.8e-02","3.4e-02","2.1e-02","4.0e-02","2.8e-02","4.9e-02","3.7e-02","5.0e-02","4.3e-02","4.7e-03","3.7e-02","1.9e-02","9.6e-03","2.7e-03","4.0e-02","3.3e-02","3.3e-02","9.7e-03","3.3e-02","4.3e-02","3.3e-02","4.0e-02","3.3e-02","4.8e-02","3.0e-02","2.1e-02","2.8e-02","9.6e-03","9.7e-03","4.3e-02","4.9e-02","4.0e-02","2.5e-02","9.6e-03","1.8e-02","4.7e-02","4.0e-02","2.2e-02","4.6e-02","1.8e-02","2.1e-02","9.7e-03","9.6e-03","6.2e-03","2.7e-02","3.8e-02","1.2e-02","2.3e-02","4.0e-02","9.7e-03","1.9e-02","1.8e-02","2.8e-02","3.8e-02","9.7e-03","3.6e-02","2.8e-02","3.5e-02","2.2e-02","2.5e-02","2.8e-02","4.7e-02","8.4e-03","2.5e-02","4.9e-02","2.2e-02","1.8e-02","4.3e-02","4.1e-02","3.1e-02","1.2e-02","4.0e-02","4.3e-02","5.0e-02","1.8e-02","4.3e-02","2.4e-02","2.4e-02","5.0e-02","2.3e-02","9.7e-03","4.3e-02","9.7e-03","4.7e-02","1.8e-02","2.1e-02","4.7e-02","5.0e-02","1.8e-02","4.0e-02","9.7e-03","1.8e-02","9.6e-03","4.0e-02","9.7e-03","4.9e-02","9.7e-03","9.7e-03","2.5e-02","2.1e-02","2.7e-02","4.7e-02","4.3e-02","4.0e-02","2.2e-02","1.7e-02","4.6e-02","3.3e-02","9.7e-03","4.5e-02","1.8e-02","4.0e-02","9.6e-03","2.5e-02","2.1e-02","3.5e-03","9.7e-03","2.4e-02","9.6e-03","4.6e-02","4.9e-02","4.7e-02","4.6e-02","2.8e-02","1.9e-02","3.6e-02","4.7e-02","2.1e-02","3.3e-02","4.0e-02","2.5e-02","4.5e-02","4.0e-02","9.7e-03","4.8e-02","2.8e-02","4.0e-02","1.7e-05","3.5e-02","2.1e-02","9.7e-03","2.8e-02","2.3e-02"],["True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True"],[-5.74,4.87,2.8,-3.84,3.75,-4.33,3.35,2.77,4.89,-3.99,4.1,-6.09,3.09,-4.23,-4.95,-4.38,-4.69,-3.71,4.02,7.67,4.57,-7.95,3.7,3,4.58,-5.1,-4.09,3.46,-3.33,5.82,3.36,-7.05,-3.3,-5.97,-6.48,-4.94,-3.62,-3.93,-5.45,-4.92,3.52,3.66,4.61,4.34,-5.35,4.45,3.86,5.44,-3.38,4.03,-4.85,4.02,-2.98,2.91,5.36,2.98,4.18,3.54,4.69,2.73,3.78,-2.92,-4.84,3.26,3.58,5.33,-5.07,-4.45,-5.12,-6.18,-3.72,4.94,-3.46,5.12,4.59,3.63,5.67,-3.8,3.59,-3.23,3.4,4.08,-3.11,-4.57,-4.1,2.89,2.93,3.08,3.99,5.19,3.73,-3.95,3.88,4.52,4.1,3.88,4.25,4.02,5.06,-7.04,-4.01,3.79,-4.43,-3.47,-3.22,-4.6,-3.96,-4.02,-4.49,-4.22,-5.13,-3.61,3.81,-3.06,-3.42,-3.23,5.6,-3.12,-4.9,-5.54,-3.08,-6.37,-3.72,-4.04,3.35,3.81,-6.89,3.17,3.33,-2.53,3.37,2.95,3.21,3.54,3.55,4.48,4.36,-2.86,-4.13,3.31,5.34,-5.6,2.83,3.02,3.63,4.03,4.24,4.31,-4.96,3.89,4.82,-2.79,4.51,-4.02,3.11,4.22,3.74,2.89,3.45,3.67,-3.44,-4.09,-4.49,2.94,-5.35,-3.32,4.37,3.02,-4.78,-3.73,-3.84,6.3,7.01,-4.29,4.53,-4.45,-3.55,2.97,2.71,3.6,-4.31,2.96,3.48,8,3.06,3.17,3.29,3.86,3.92,-6.71,-3.17,4.8,3.38,9.78,3.39,3.7,4.82,4.66,-3.65]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>name<\/th>\n      <th>ID<\/th>\n      <th>p.val<\/th>\n      <th>p.adj<\/th>\n      <th>significant<\/th>\n      <th>ratio<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-center","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"pageLength":10,"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="old-plot-unique-2" class="section level5">
<h5>Old plot + Unique</h5>
<pre class="r"><code>unique_vec &lt;- c(&quot;AARS2&quot;, &quot;ABCA8&quot;, &quot;BAZ1B&quot;, &quot;CCDC58&quot;, &quot;COMT&quot;, &quot;FAF1&quot;, &quot;FUT8&quot;, &quot;GATAD2B&quot;, &quot;INTS5&quot;, &quot;MBLAC2&quot;, &quot;MCCC1&quot;, &quot;NDUFA6&quot;, &quot;NDUFV2&quot;, &quot;NELFB&quot;, &quot;PABPN1&quot;, &quot;PLIN1&quot;, &quot;PLSCR1&quot;, &quot;POLR2E&quot;, &quot;PPIF&quot;, &quot;PTGES2&quot;, &quot;PTGES3&quot;, &quot;PTGS1&quot;, &quot;QTRT1&quot;, &quot;RPL18&quot;, &quot;RPL37A&quot;, &quot;SMNDC1&quot;, &quot;SNRPD3&quot;, &quot;TIMM10&quot;, &quot;TPD52&quot;, &quot;XRCC1&quot;, &quot;CAV1&quot;, &quot;EFEMP1&quot;, &quot;PON1&quot;, &quot;PRR12&quot;, &quot;SNX7&quot;, &quot;STX18&quot;, &quot;CCDC58&quot;, &quot;CD47&quot;, &quot;CPSF6&quot;, &quot;CST1&quot;, &quot;CSTF3&quot;, &quot;FKBP8&quot;, &quot;FNTA&quot;, &quot;GTF3C1&quot;, &quot;HK2&quot;, &quot;MBLAC2&quot;, &quot;MUC16&quot;, &quot;PCDHGA10&quot;, &quot;PLD3&quot;, &quot;PLIN1&quot;, &quot;PPIF&quot;, &quot;PTGES3&quot;, &quot;QTRT1&quot;, &quot;RAD21&quot;, &quot;RPL37A&quot;, &quot;S100A1&quot;, &quot;SLC34A2&quot;, &quot;TRIM33&quot;, &quot;TTR&quot;, &quot;TUBGCP2&quot;, &quot;U2SURP&quot;, &quot;UTS2&quot;, &quot;WISP2&quot;, &quot;XPC&quot;)

plot_volcano_3(dep, contrast=test, add_names=T,label_size=4,adjusted = F)</code></pre>
<p><img src="figure/Integration_old_analysis_T1_TME1.Rmd/V43-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="unique-table-2" class="section level5">
<h5>Unique Table</h5>
<pre class="r"><code>results %&gt;%  select(starts_with(&quot;name&quot;) | starts_with(test)) %&gt;% filter(name %in% unique_vec) %&gt;% arrange(across(ends_with(&quot;p.adj&quot;))) %&gt;% mutate(across(2:3, format, scientific = T, digits = 2)) %&gt;% dplyr::rename_with(~ tolower(gsub(paste0(test,&quot;_&quot;), &quot;&quot;, .x)), .cols = starts_with(test)) %&gt;% DT::datatable()</code></pre>
<div id="htmlwidget-731dbbad9816a9db95c6" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-731dbbad9816a9db95c6">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52"],["TUBGCP2","SNRPD3","FAF1","CST1","CD47","PRR12","CCDC58","FNTA","PTGS1","QTRT1","STX18","CPSF6","UTS2","FKBP8","FUT8","SLC34A2","HK2","MBLAC2","NDUFA6","PPIF","XPC","SNX7","PLIN1","CSTF3","PTGES3","WISP2","PTGES2","PON1","MCCC1","INTS5","AARS2","ABCA8","RPL37A","PCDHGA10","TTR","U2SURP","TPD52","RAD21","TIMM10","TRIM33","PLSCR1","PABPN1","S100A1","MUC16","XRCC1","CAV1","PLD3","GTF3C1","POLR2E","BAZ1B","NDUFV2","RPL18"],["8.7e-09","8.9e-06","1.7e-05","5.8e-05","9.7e-05","1.2e-04","1.9e-04","1.5e-04","1.6e-04","1.8e-04","6.6e-04","1.3e-03","1.6e-03","2.0e-03","2.0e-03","2.8e-03","3.1e-03","3.2e-03","3.4e-03","5.4e-03","5.6e-03","6.1e-03","6.2e-03","6.6e-03","6.9e-03","7.7e-03","8.9e-03","1.0e-02","1.1e-02","1.1e-02","1.2e-02","1.4e-02","1.6e-02","1.6e-02","1.8e-02","2.9e-02","3.7e-02","3.9e-02","4.0e-02","4.1e-02","4.6e-02","4.6e-02","5.1e-02","5.4e-02","7.2e-02","8.6e-02","1.1e-01","1.1e-01","1.2e-01","1.5e-01","2.9e-01","4.0e-01"],["1.7e-05","3.5e-03","4.7e-03","9.4e-03","9.6e-03","9.6e-03","9.7e-03","9.7e-03","9.7e-03","9.7e-03","1.9e-02","2.5e-02","2.8e-02","3.3e-02","3.3e-02","4.0e-02","4.0e-02","4.1e-02","4.3e-02","5.2e-02","5.2e-02","5.3e-02","5.3e-02","5.5e-02","5.6e-02","6.1e-02","6.5e-02","7.0e-02","7.2e-02","7.3e-02","7.5e-02","8.2e-02","8.9e-02","9.0e-02","9.3e-02","1.2e-01","1.4e-01","1.4e-01","1.4e-01","1.4e-01","1.5e-01","1.6e-01","1.7e-01","1.7e-01","2.1e-01","2.3e-01","2.7e-01","2.8e-01","2.8e-01","3.3e-01","4.7e-01","5.7e-01"],[true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],[9.78,6.3,5.33,5.44,4.57,-4.96,4.02,5.12,4.82,4.51,-4.31,3.66,4.66,4.94,4.59,3.02,3.08,3.35,2.95,3.33,2.91,-3.05,3.05,3.02,3.59,2.79,3.24,-3.08,2.91,2.56,4.11,3.02,2.49,2.91,2.41,2.56,2.91,2.38,3,2.18,2.32,2.28,2.16,1.92,1.65,-1.58,1.52,1.68,2.27,1.75,1.08,0.783]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>name<\/th>\n      <th>p.val<\/th>\n      <th>p.adj<\/th>\n      <th>significant<\/th>\n      <th>ratio<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":5},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>expdesign &lt;- read.table(here::here(&quot;data/expdesign_old_exp.tsv&quot;), header = T, stringsAsFactors = F)

cond_1&lt;-&quot;T1_P4&quot;
cond_2&lt;-&quot;TME1_P4&quot;

conditions&lt;-c(cond_1,cond_2)

expdesign &lt;- subset(expdesign, condition %in% conditions)

test&lt;- paste(cond_1,cond_2,sep=&quot;_vs_&quot;)

columns&lt;-match(paste(&quot;LFQ.intensity.&quot;,expdesign$label,sep=&quot;&quot;),colnames(data_unique))

data_se &lt;- make_se(data_unique, columns, expdesign)</code></pre>
</div>
</div>
</div>
<div id="t1_p4_vs_tme1_p4" class="section level3">
<h3>T1_P4_vs_TME1_P4</h3>
<pre class="r"><code>data_filt &lt;- filter_missval(data_se, thr = 1)

set.seed(20210520)
data_imp_man &lt;- impute(data_filt, fun = &quot;man&quot;, shift = 1.8, scale = 0.3)

data_diff &lt;- test_diff_BH(data_imp_man, type = &quot;all&quot;)

dep &lt;- add_rejections(data_diff, alpha = 0.05, lfc = 1)

results&lt;-get_results(dep)</code></pre>
<p>There are <strong>204</strong> significant proteins in <strong>T1_P4_vs_TME1_P4</strong> comparison.</p>
<div id="visualization-of-the-results-3" class="section level4 tabset">
<h4><strong>Visualization of the results:</strong></h4>
<hr />
<div id="volcano-plot-3" class="section level5">
<h5>Volcano Plot</h5>
<pre class="r"><code>plot_volcano(dep, contrast=test, add_names=T,label_size=4,adjusted = F)</code></pre>
<p><img src="figure/Integration_old_analysis_T1_TME1.Rmd/V33-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="results-table-3" class="section level5">
<h5>Results Table</h5>
<pre class="r"><code>results %&gt;% 
  filter(significant) %&gt;%
  select(-significant, -ends_with(&quot;centered&quot;)) %&gt;% 
  pivot_longer(
   cols = ends_with(c(&quot;p.val&quot;, &quot;p.adj&quot;, &quot;significant&quot;, &quot;ratio&quot;)),
   names_to = c(&quot;comparison&quot;, &quot;.value&quot;),
   names_pattern = &quot;(.*_.*_.*_.*_.*)_(.*)&quot;) %&gt;% 
  arrange(name, p.adj) %&gt;% 
  relocate(comparison) %&gt;% 
  select(-comparison) %&gt;% 
  mutate(across(ends_with(c(&quot;p.val&quot;, &quot;p.adj&quot;)), format, scientific = T, digits = 2)) %&gt;% 
  mutate(significant = str_to_title(significant)) %&gt;% 
  DT::datatable(options = list(
  columnDefs = list(list(className = &#39;dt-center&#39;, targets = 1:6)),
  pageLength = 10))</code></pre>
<div id="htmlwidget-875e4a54f2c9023ea392" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-875e4a54f2c9023ea392">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204"],["AARS2","ABCA8","ABLIM1","ACAA1","ACLY","AGK","ALDH6A1","ANP32A","ANXA3","ATAD3A","ATP1A1","ATP5H","ATP5O","B2M","BANF1","BTF3L4","C16orf46","C1QBP","CALR","CANX","CBX3","CCDC58","CD47","CD74","CDC2","CDC37L1","CIRBP","COPG2","CORO1A","CORO7-PAM16","COX4I1","CPSF6","CRABP2","CRAT","CRMP1","CS","CSTF2","CTBP1","CYC1","CYCS","DDAH1","DDX17","DDX21","DNAJC8","DSG2","EEF1B2","EEF1D.1","EIF4H","ELAVL1","F11R","FAM162A","FH","FKBP3","FTH1","FTL","FUBP1","FUNDC2","FUS","G3BP2","GBAS","GFPT1","GLG1","GORASP2","GPI","GSTK1","GYS1","H1FX","H2AFY","H2AFY2","HADHB","HDGF","HIST1H1B","HIST1H2BN","HIST2H2AC","HLA-A","HLA-C","HLA-DRA","HLA-DRB1","HLA-G","HMGB1","HNRNPA0","HNRNPA1","HNRNPA2B1","HNRNPAB","HNRNPC","HNRNPD.1","HNRNPDL.1","HNRNPH1","HNRNPH3","HNRNPK","HNRNPL","HNRNPM","HNRNPU","HSP90B1","HSPA1B","HSPA5","HSPA9","HSPD1","HSPE1","ILF3","KHDRBS1","KIAA1217","KRT18","LDHB","LMAN2","LMNB1","LRRFIP2","LUC7L2","MDH1.1","MDH2","MEX3A","MT-CO2","MYO6","NARS","NCF2","NCL","NDUFB10","NME1-NME2","NMT1","NONO","NOP58","NPM1","NUCKS1","OXA1L","P0DP04","P4HB","PABPN1","PAK1","PDIA3","PDIA3.1","PDIA6","PDXDC1","PFDN2","PHB","PLD3","PPA1","PPIB","PPIF","PPIG","PPP1CA","PRDX3","PRDX5","PRKCSH","PSME1","PSME2","PTGES3","PTMS","PUF60","QTRT1","RAB6A","RBMX","RCC2","RPL12","RPL13","RPL17","RPL18","RPL22","RPL4","RPL9","RPLP0","RPLP2","RPN1","RPS14","RPS15A.1","RPS19","RPS6.1","RPSA","RUVBL2","S100A1","S100A13","SARNP","SDHA","SEPT10","SERPINH1","SET","SFPQ","SLC1A5","SLC27A1","SNRPD1","SNRPD3","SRP9","SRSF1","SRSF6","SRSF9","SSB","SSBP1","SSR4","STIP1","STMN1","STX18","TACO1","TAF5","TBCD","TIMM10","TIMM13","TMPO.1","TOR4A","TUBGCP2","TUFM","USP14","UTS2","VARS","VPS13C","YWHAE"],["Q5JTZ9","A0A0A0MSU4","A0A3B3IS55","P09110","P53396","E9PC15","Q02252","P39687","P12429","Q9NVI7","P05023","O75947","P48047","P61769","O75531","Q96K17","Q6P387","Q07021","P27797","P27824","Q13185","C9JQ41","Q08722","H0YBZ2","A0A024QZP7","B1AL69","Q14011","Q9UBF2","P31146","A0A0A6YYL4","P13073","F8WJN3","P29373","P43155","Q14194","O75390","E7EWR4","Q13363","P08574","C9JFR7","O94760","A0A1X7SBZ2","Q9NR30","O75937","Q14126","P24534","E9PRY8","Q15056","Q15717","Q9Y624","Q96A26","P07954","Q00688","P02794","P02792","Q96AE4","Q9BWH2","P35637","Q9UN86","O75323","Q06210","Q92896","Q9H8Y8","P06744","Q9Y2Q3","P13807","Q92522","O75367","Q9P0M6","P55084","P51858","P16401","U3KQK0","Q16777","A0A0G2JI36","A0A1W2PRZ0","A0A0G2JMH6","A2BFX2","Q8MH48","Q5T7C4","Q13151","F8W6I7","P22626","D6R9P3","G3V4C1","Q14103","A0A087WUK2","P31943","P31942","P61978","P14866","P52272","Q00839","P14625","A0A0G2JIW1","P11021","P38646","P10809","P61604","Q12906","Q07666","Q5T5P2","P05783","A0A5F9ZHM4","Q12907","P20700","Q9Y608","A0A0A6YYJ8","P40925","P40926","A1L020","P00403","A0A0A0MRM8","O43776","P19878","P19338","O96000","Q32Q12","P30419","Q15233","Q9Y2X3","P06748","Q9H1E3","A0A087X0L7","P0DP04","P07237","Q86U42","Q13153","H7BZJ3","P30101","Q15084","Q6P996","Q9UHV9","P35232","Q8IV08","Q15181","P23284","P30405","C9JN15","P62136","P30048","P30044","K7ELL7","Q06323","A0A087X1Z3","Q15185","P20962","A0A0J9YYL3","Q9BXR0","P20340","P38159","Q9P258","P30050","P26373","A0A087WXM6","G3V203","P35268","P36578","D6RAN4","P05388","P05387","P04843","P62263","P62244","P39019","P62753","C9J9K3","Q9Y230","P23297","Q99584","P82979","P31040","E7EW69","P50454","A0A0C4DFV9","P23246","Q15758","Q6PCB7","P62314","P62318","P49458","J3KTL2","A0A590UJK4","Q13242","P05455","Q04837","P51571","P31948","P16949","Q9P2W9","Q9BSH4","Q15542","J3KR97","P62072","Q9Y5L4","P42167","Q9NXH8","Q9BSJ2","P49411","P54578","Q5H8X8","P26640","Q709C8","P62258"],["1.5e-04","1.4e-03","3.5e-03","2.5e-03","2.6e-03","1.2e-03","5.9e-03","4.6e-04","2.4e-03","4.5e-03","5.7e-03","3.6e-03","4.0e-03","3.1e-05","2.3e-03","4.3e-03","3.0e-03","3.2e-04","5.7e-04","1.3e-03","1.0e-03","6.8e-04","3.5e-03","8.0e-05","5.8e-04","2.4e-03","6.4e-03","1.3e-03","2.4e-03","3.8e-03","1.8e-03","5.3e-03","6.3e-03","3.3e-03","5.4e-03","7.3e-04","4.8e-03","2.0e-03","5.2e-03","3.6e-03","2.8e-03","6.2e-03","2.9e-03","5.2e-03","5.4e-03","3.1e-03","2.1e-03","2.1e-03","3.4e-03","1.6e-03","6.3e-04","3.4e-04","1.4e-03","2.3e-03","8.3e-04","4.9e-04","3.9e-03","1.8e-03","3.0e-03","1.6e-03","3.3e-03","2.7e-03","5.0e-03","5.9e-03","6.1e-03","4.4e-03","5.8e-03","4.2e-03","4.3e-04","6.3e-03","2.4e-03","4.9e-04","4.7e-03","5.6e-03","6.6e-04","4.2e-03","3.3e-04","1.9e-04","2.1e-03","2.6e-03","3.6e-03","1.7e-03","7.1e-04","1.4e-03","1.4e-04","1.3e-03","3.2e-04","3.4e-03","4.4e-04","2.2e-03","3.9e-03","2.9e-03","5.3e-04","3.7e-03","2.3e-03","3.7e-03","1.5e-03","1.9e-04","7.7e-05","1.6e-04","1.9e-03","4.7e-04","6.6e-06","1.0e-03","5.1e-03","1.9e-03","3.4e-03","6.9e-05","8.4e-04","2.7e-04","1.7e-03","2.3e-03","7.4e-04","4.2e-03","2.2e-05","9.9e-05","1.8e-03","8.5e-04","4.7e-03","3.1e-03","6.3e-03","1.3e-04","7.6e-04","1.4e-04","6.4e-03","4.3e-03","1.1e-04","2.8e-03","2.5e-03","2.8e-03","8.0e-04","4.4e-03","5.7e-03","3.7e-03","6.2e-03","1.2e-03","5.2e-03","1.0e-03","5.1e-03","2.8e-04","1.9e-03","3.5e-05","9.0e-04","3.0e-04","1.2e-03","3.8e-04","2.5e-03","3.0e-03","7.3e-04","5.7e-03","2.0e-03","1.8e-03","5.0e-03","3.3e-05","3.4e-03","5.7e-04","5.4e-03","1.9e-03","4.4e-03","5.2e-03","4.5e-03","3.6e-03","5.0e-04","2.6e-03","4.4e-04","1.8e-03","3.5e-04","1.9e-03","6.4e-05","5.7e-03","1.7e-03","4.6e-03","2.6e-03","2.0e-03","1.3e-04","2.7e-03","2.4e-03","3.4e-03","3.5e-05","1.9e-03","1.1e-04","1.7e-03","5.8e-03","6.1e-03","2.1e-03","3.0e-03","2.8e-03","3.7e-04","7.7e-04","5.8e-03","1.4e-03","1.7e-03","9.6e-04","1.4e-03","8.2e-04","3.9e-03","4.1e-03","1.7e-03","1.1e-03","5.5e-03","3.5e-03","3.8e-04","4.7e-03","3.2e-03"],["1.3e-02","3.0e-02","3.9e-02","3.4e-02","3.5e-02","2.8e-02","4.8e-02","2.0e-02","3.4e-02","4.3e-02","4.8e-02","3.9e-02","4.1e-02","9.3e-03","3.4e-02","4.2e-02","3.7e-02","1.9e-02","2.1e-02","2.9e-02","2.6e-02","2.3e-02","3.9e-02","1.3e-02","2.1e-02","3.4e-02","5.0e-02","3.0e-02","3.4e-02","4.0e-02","3.2e-02","4.7e-02","5.0e-02","3.8e-02","4.7e-02","2.3e-02","4.4e-02","3.2e-02","4.6e-02","3.9e-02","3.6e-02","5.0e-02","3.6e-02","4.6e-02","4.7e-02","3.7e-02","3.3e-02","3.3e-02","3.8e-02","3.2e-02","2.2e-02","1.9e-02","3.0e-02","3.4e-02","2.3e-02","2.0e-02","4.0e-02","3.2e-02","3.6e-02","3.2e-02","3.8e-02","3.5e-02","4.5e-02","4.8e-02","4.9e-02","4.2e-02","4.8e-02","4.2e-02","2.0e-02","5.0e-02","3.4e-02","2.0e-02","4.4e-02","4.7e-02","2.3e-02","4.2e-02","1.9e-02","1.4e-02","3.3e-02","3.5e-02","3.9e-02","3.2e-02","2.3e-02","3.0e-02","1.3e-02","2.9e-02","1.9e-02","3.8e-02","2.0e-02","3.4e-02","4.0e-02","3.6e-02","2.0e-02","3.9e-02","3.4e-02","3.9e-02","3.0e-02","1.4e-02","1.3e-02","1.3e-02","3.2e-02","2.0e-02","9.3e-03","2.6e-02","4.6e-02","3.2e-02","3.8e-02","1.3e-02","2.3e-02","1.9e-02","3.2e-02","3.4e-02","2.3e-02","4.2e-02","9.3e-03","1.3e-02","3.2e-02","2.3e-02","4.4e-02","3.7e-02","5.0e-02","1.3e-02","2.3e-02","1.3e-02","5.0e-02","4.2e-02","1.3e-02","3.6e-02","3.5e-02","3.6e-02","2.3e-02","4.2e-02","4.8e-02","3.9e-02","5.0e-02","2.9e-02","4.6e-02","2.6e-02","4.6e-02","1.9e-02","3.2e-02","9.3e-03","2.4e-02","1.9e-02","2.8e-02","1.9e-02","3.4e-02","3.7e-02","2.3e-02","4.8e-02","3.2e-02","3.2e-02","4.6e-02","9.3e-03","3.8e-02","2.1e-02","4.7e-02","3.2e-02","4.2e-02","4.6e-02","4.3e-02","3.9e-02","2.0e-02","3.5e-02","2.0e-02","3.2e-02","1.9e-02","3.2e-02","1.3e-02","4.8e-02","3.2e-02","4.4e-02","3.5e-02","3.2e-02","1.3e-02","3.6e-02","3.4e-02","3.8e-02","9.3e-03","3.2e-02","1.3e-02","3.2e-02","4.8e-02","4.9e-02","3.3e-02","3.6e-02","3.6e-02","1.9e-02","2.3e-02","4.8e-02","3.0e-02","3.2e-02","2.5e-02","3.0e-02","2.3e-02","4.0e-02","4.1e-02","3.2e-02","2.8e-02","4.7e-02","3.9e-02","1.9e-02","4.4e-02","3.8e-02"],["True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True"],[5.88,4.2,3.85,-4.39,3.92,4.17,3.76,5.09,4.52,4.05,4.15,3.92,3.99,6.66,4.8,3.78,-4.94,5.18,5.79,5.69,4.26,4.43,4.34,5.47,4.79,3.67,3.66,3.99,4.16,-5.01,5.34,3.75,3.77,3.53,4.16,5.47,3.42,3.92,4.75,4.19,3.79,4.45,3.97,3.26,3.33,3.99,4.7,3.62,3.78,4.15,4.79,6.25,4.09,4.58,4.15,4.84,-4.7,4.78,3.47,4.64,4.33,5.35,3.63,4.91,3.39,3.36,3.47,4.59,4.81,4.26,4.29,5.82,5.55,4.94,4.34,3.9,5.24,5.24,4.11,4.97,3.81,5.21,5.4,4.33,6.27,4.81,4.6,5.32,5.64,4.59,3.84,3.95,6.68,5.37,5.65,4.73,5.34,6.7,7.67,5.72,4.1,4.43,9.11,6.27,3.75,4.79,-4.66,5.28,5.33,7.1,5.59,5.15,4.65,4.18,6.12,6.73,3.84,6.37,3.79,4.06,3.38,5.77,4.31,5.87,-3.47,4.55,5.57,3.51,5.28,5.3,5.63,3.31,3.1,5,3.2,4.53,4.09,4.06,-3.75,4.87,5.16,6.66,4.43,5.66,4.35,4.66,5.2,3.79,4.7,3.75,4.7,4.26,4.38,6.1,4.1,5.7,4.59,4.76,4.13,4.63,4.99,4.55,4.83,4.48,5.39,4.93,5.45,4.28,5.6,4.04,4.08,3.96,4.17,5.86,5.85,5.32,4.57,3.87,5.77,5.01,5.38,5.39,3.5,3.35,4.59,4.06,4.12,5.42,4.37,-3.38,5.21,-5.4,6.11,3.9,4.27,3.78,-4.09,6.9,6.23,3.83,4.38,5.41,3.96,4.66]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>name<\/th>\n      <th>ID<\/th>\n      <th>p.val<\/th>\n      <th>p.adj<\/th>\n      <th>significant<\/th>\n      <th>ratio<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-center","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"pageLength":10,"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="old-plot-unique-3" class="section level5">
<h5>Old plot + Unique</h5>
<pre class="r"><code>unique_vec &lt;- c(&quot;AARS2&quot;, &quot;ABCA8&quot;, &quot;BAZ1B&quot;, &quot;CCDC58&quot;, &quot;COMT&quot;, &quot;FAF1&quot;, &quot;FUT8&quot;, &quot;GATAD2B&quot;, &quot;INTS5&quot;, &quot;MBLAC2&quot;, &quot;MCCC1&quot;, &quot;NDUFA6&quot;, &quot;NDUFV2&quot;, &quot;NELFB&quot;, &quot;PABPN1&quot;, &quot;PLIN1&quot;, &quot;PLSCR1&quot;, &quot;POLR2E&quot;, &quot;PPIF&quot;, &quot;PTGES2&quot;, &quot;PTGES3&quot;, &quot;PTGS1&quot;, &quot;QTRT1&quot;, &quot;RPL18&quot;, &quot;RPL37A&quot;, &quot;SMNDC1&quot;, &quot;SNRPD3&quot;, &quot;TIMM10&quot;, &quot;TPD52&quot;, &quot;XRCC1&quot;, &quot;CAV1&quot;, &quot;EFEMP1&quot;, &quot;PON1&quot;, &quot;PRR12&quot;, &quot;SNX7&quot;, &quot;STX18&quot;)

plot_volcano_3(dep, contrast=test, add_names=T,label_size=4,adjusted = F)</code></pre>
<p><img src="figure/Integration_old_analysis_T1_TME1.Rmd/V44-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="unique-table-3" class="section level5">
<h5>Unique Table</h5>
<pre class="r"><code>results %&gt;%  select(starts_with(&quot;name&quot;) | starts_with(test)) %&gt;% filter(name %in% unique_vec) %&gt;% arrange(across(ends_with(&quot;p.adj&quot;))) %&gt;% mutate(across(2:3, format, scientific = T, digits = 2)) %&gt;% dplyr::rename_with(~ tolower(gsub(paste0(test,&quot;_&quot;), &quot;&quot;, .x)), .cols = starts_with(test)) %&gt;% DT::datatable()</code></pre>
<div id="htmlwidget-8256c90a5b4deed805c1" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-8256c90a5b4deed805c1">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25"],["PABPN1","AARS2","PTGES3","RPL18","CCDC58","QTRT1","PPIF","ABCA8","TIMM10","SNRPD3","STX18","BAZ1B","RPL37A","MBLAC2","PTGS1","TPD52","FAF1","GATAD2B","PLSCR1","PLIN1","NDUFV2","MCCC1","XRCC1","SMNDC1","COMT"],["1.1e-04","1.5e-04","3.8e-04","5.7e-04","6.8e-04","7.3e-04","1.0e-03","1.4e-03","1.4e-03","1.9e-03","5.8e-03","7.0e-03","1.4e-02","1.6e-02","1.9e-02","3.3e-02","3.6e-02","4.0e-02","5.1e-02","5.4e-02","1.7e-01","1.9e-01","2.1e-01","2.8e-01","6.8e-01"],["1.3e-02","1.3e-02","1.9e-02","2.1e-02","2.3e-02","2.3e-02","2.6e-02","3.0e-02","3.0e-02","3.2e-02","4.8e-02","5.2e-02","6.8e-02","7.3e-02","7.9e-02","1.1e-01","1.1e-01","1.2e-01","1.3e-01","1.4e-01","2.9e-01","3.0e-01","3.3e-01","4.0e-01","7.5e-01"],[true,true,true,true,true,true,true,true,true,true,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false],[5.57,5.88,4.66,5.7,4.43,4.7,4.06,4.2,3.9,5.01,-3.38,3.56,3.03,3.07,3.17,2.58,3.12,2.34,2.28,2.33,1.41,1.73,1.29,1.15,0.454]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>name<\/th>\n      <th>p.val<\/th>\n      <th>p.adj<\/th>\n      <th>significant<\/th>\n      <th>ratio<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":5},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>expdesign &lt;- read.table(here::here(&quot;data/expdesign_old_exp.tsv&quot;), header = T, stringsAsFactors = F)

cond_1&lt;-&quot;T1_P5&quot;
cond_2&lt;-&quot;TME1_P5&quot;

conditions&lt;-c(cond_1,cond_2)

expdesign &lt;- subset(expdesign, condition %in% conditions)

test&lt;- paste(cond_1,cond_2,sep=&quot;_vs_&quot;)

columns&lt;-match(paste(&quot;LFQ.intensity.&quot;,expdesign$label,sep=&quot;&quot;),colnames(data_unique))

data_se &lt;- make_se(data_unique, columns, expdesign)</code></pre>
</div>
</div>
</div>
<div id="t1_p5_vs_tme1_p5" class="section level3">
<h3>T1_P5_vs_TME1_P5</h3>
<pre class="r"><code>data_filt &lt;- filter_missval(data_se, thr = 1)

set.seed(20210520)
data_imp_man &lt;- impute(data_filt, fun = &quot;man&quot;, shift = 1.8, scale = 0.3)

data_diff &lt;- test_diff_BH(data_imp_man, type = &quot;all&quot;)

dep &lt;- add_rejections(data_diff, alpha = 0.05, lfc = 1)

results&lt;-get_results(dep)</code></pre>
<p>There are <strong>94</strong> significant proteins in <strong>T1_P5_vs_TME1_P5</strong> comparison.</p>
<div id="visualization-of-the-results-4" class="section level4 tabset">
<h4><strong>Visualization of the results:</strong></h4>
<hr />
<div id="volcano-plot-4" class="section level5">
<h5>Volcano Plot</h5>
<pre class="r"><code>plot_volcano(dep, contrast=test, add_names=T,label_size=4,adjusted = F)</code></pre>
<p><img src="figure/Integration_old_analysis_T1_TME1.Rmd/V34-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="results-table-4" class="section level5">
<h5>Results Table</h5>
<pre class="r"><code>results %&gt;% 
  filter(significant) %&gt;%
  select(-significant, -ends_with(&quot;centered&quot;)) %&gt;% 
  pivot_longer(
   cols = ends_with(c(&quot;p.val&quot;, &quot;p.adj&quot;, &quot;significant&quot;, &quot;ratio&quot;)),
   names_to = c(&quot;comparison&quot;, &quot;.value&quot;),
   names_pattern = &quot;(.*_.*_.*_.*_.*)_(.*)&quot;) %&gt;% 
  arrange(name, p.adj) %&gt;% 
  relocate(comparison) %&gt;% 
  select(-comparison) %&gt;% 
  mutate(across(ends_with(c(&quot;p.val&quot;, &quot;p.adj&quot;)), format, scientific = T, digits = 2)) %&gt;% 
  mutate(significant = str_to_title(significant)) %&gt;% 
  DT::datatable(options = list(
  columnDefs = list(list(className = &#39;dt-center&#39;, targets = 1:6)),
  pageLength = 10))</code></pre>
<div id="htmlwidget-431172203e1cb75c84d4" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-431172203e1cb75c84d4">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94"],["AARS2","ABCF2","ADD3","AGAP1","AGT","AKR1C3","ALDH3A2","ANP32E","ARIH2","ASNS","BOLA2","C16orf46","CLDN3","COX5B","CPB2","CPNE1","CPSF6","DNAJC9","DPY30","DSC1","ECI2","EIF3I","EIF4G2","ELAVL1","G3BP1","G3BP2","GALNT5","GLG1","GPNMB","HIST1H1C","HIST2H2AC","HLA-DRA","HMGB2","HPX","IGHA1","IL16","ITIH1","KATNAL2","KCNAB2","LACTB2","LRBA","LUC7L2","MIF","MMP11","MTPAP","MVD","NAP1L1","NDUFAF5","NDUFS4","NIF3L1","NLN","NUCKS1","OLA1","PCCA","PDCD10","PDIA3","POLR2E","PPIH","PRPF4B","PRPSAP2","PTPRA","RAB7A","RAB8B","RAC1","RB1CC1","ROR1","RPL18","RPL29","S100A4","SASH3","SEPT6","SETD3","SF3B4","SLC35E1","SMAD2","SNRPD1","SPTBN2","STAM2","STK24","STX18","TCOF1.1","TIMM10","TNPO2","TOR4A","TPM1","TSFM","TUBA4A","U2AF1","UBE2I","UMPS","UPF2","USMG5","USP14","VCAN"],["Q5JTZ9","Q9UG63","Q9UEY8","Q9UPQ3","P01019","P42330","P51648","Q9BTT0","O95376","P08243","Q9H3K6","Q6P387","O15551","P10606","A0A087WSY5","Q99829","F8WJN3","Q8WXX5","Q9C005","Q08554","A0A0C4DGA2","Q13347","P78344","Q15717","Q13283","Q9UN86","Q7Z7M9","Q92896","A0A3B3ISS6","P16403","Q16777","A0A0G2JMH6","P26583","P02790","A0A286YEY1","H0YLH9","P19827","K7EM02","A0A590UJB6","Q53H82","A0A494C1L5","A0A0A6YYJ8","P14174","P24347","Q9NVV4","P53602","F8VY35","B3KR61","O43181","Q9GZT8","Q9BYT8","Q9H1E3","J3KQ32","P05165","Q9BUL8","H7BZJ3","A0A0A0MQR7","C9JQD4","Q13523","O60256","P18433","P51149","Q92930","P63000","Q8TDY2","A0A0A0MTJ5","G3V203","A0A3B3ITT5","P26447","O75995","B1AMS2","Q86TU7","Q15427","Q96K37","Q15796","P62314","O15020","O75886","B4DR80","Q9P2W9","Q13428","P62072","O14787","Q9NXH8","A0A0S2Z4G6","P43897","P68366","P0DN76","H3BQQ9","P11172","Q9HAU5","Q96IX5","P54578","P13611"],["2.0e-04","1.1e-03","1.3e-03","5.3e-04","4.1e-04","1.6e-03","1.3e-03","2.2e-03","2.3e-03","1.1e-04","2.0e-04","1.9e-03","2.1e-03","3.7e-06","2.3e-03","2.9e-04","4.3e-04","3.4e-05","4.5e-05","8.2e-04","5.1e-04","5.2e-04","2.1e-03","1.1e-03","5.8e-04","2.1e-03","5.5e-07","6.9e-05","1.9e-05","6.3e-04","2.2e-03","2.3e-03","1.2e-03","1.3e-04","1.2e-03","9.0e-04","3.7e-04","1.6e-03","9.6e-04","4.7e-05","5.7e-06","9.3e-04","8.0e-04","4.4e-04","4.4e-04","1.7e-04","1.5e-04","1.2e-04","5.6e-04","3.9e-04","2.0e-03","5.2e-04","1.0e-03","1.0e-03","2.1e-03","2.0e-05","2.3e-03","5.2e-04","1.6e-04","1.7e-03","3.2e-05","1.5e-04","5.3e-04","8.2e-04","1.3e-06","4.8e-04","4.6e-04","7.4e-04","1.0e-03","8.5e-04","8.4e-04","9.7e-04","1.8e-04","1.3e-03","5.1e-04","1.1e-03","1.0e-03","1.6e-03","9.8e-04","6.1e-05","2.1e-03","2.3e-04","9.6e-04","1.1e-03","3.9e-05","3.7e-04","2.3e-03","2.0e-03","3.2e-04","3.9e-04","1.4e-03","7.8e-04","5.2e-04","1.8e-03"],["1.7e-02","3.3e-02","3.4e-02","2.4e-02","2.4e-02","4.1e-02","3.4e-02","4.9e-02","4.9e-02","1.6e-02","1.7e-02","4.6e-02","4.9e-02","2.4e-03","4.9e-02","2.3e-02","2.4e-02","8.3e-03","8.5e-03","3.1e-02","2.4e-02","2.4e-02","4.9e-02","3.1e-02","2.5e-02","4.9e-02","1.1e-03","1.0e-02","6.5e-03","2.7e-02","4.9e-02","4.9e-02","3.4e-02","1.6e-02","3.4e-02","3.1e-02","2.4e-02","4.2e-02","3.1e-02","8.5e-03","2.8e-03","3.1e-02","3.1e-02","2.4e-02","2.4e-02","1.7e-02","1.7e-02","1.6e-02","2.5e-02","2.4e-02","4.8e-02","2.4e-02","3.1e-02","3.1e-02","4.9e-02","6.5e-03","4.9e-02","2.4e-02","1.7e-02","4.3e-02","8.3e-03","1.7e-02","2.4e-02","3.1e-02","1.3e-03","2.4e-02","2.4e-02","3.1e-02","3.1e-02","3.1e-02","3.1e-02","3.1e-02","1.7e-02","3.4e-02","2.4e-02","3.1e-02","3.1e-02","4.2e-02","3.1e-02","1.0e-02","4.9e-02","1.9e-02","3.1e-02","3.1e-02","8.5e-03","2.4e-02","4.9e-02","4.9e-02","2.4e-02","2.4e-02","3.7e-02","3.1e-02","2.4e-02","4.4e-02"],["True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True"],[4.89,3.99,3.01,-4.58,-3.94,4.82,-6.59,4.11,-2.61,-4.19,3.49,-6.34,-5.74,6.42,3.37,3.57,3.55,-8.33,-4.18,-4.42,-4.33,4.65,3.43,3.84,3.84,3.5,-9.02,4.6,-4.66,3.43,3.41,2.66,2.9,-4.81,-4.22,3.21,-4.18,4.54,3.29,-6.14,6.28,3.89,5.08,4.71,-3.19,3.77,4.08,-3.74,3.84,3.26,2.75,3.7,3.24,-4.63,2.69,4.8,3.5,3.11,4.24,3.15,-4.45,4.84,3.44,4.26,-7.55,3.9,5.31,3.59,6.48,3.66,5,-3.18,3.67,-4.52,-4.38,4.49,3.82,-3.54,3.76,-4.67,3.04,3.78,2.79,-4.11,-4.76,-3.45,3.51,3.11,4.42,4.09,-3.55,3.36,3.43,-2.69]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>name<\/th>\n      <th>ID<\/th>\n      <th>p.val<\/th>\n      <th>p.adj<\/th>\n      <th>significant<\/th>\n      <th>ratio<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-center","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"pageLength":10,"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="old-plot-unique-4" class="section level5">
<h5>Old plot + Unique</h5>
<pre class="r"><code>unique_vec &lt;- c(&quot;AARS2&quot;, &quot;ABCA8&quot;, &quot;BAZ1B&quot;, &quot;CCDC58&quot;, &quot;COMT&quot;, &quot;FAF1&quot;, &quot;FUT8&quot;, &quot;GATAD2B&quot;, &quot;INTS5&quot;, &quot;MBLAC2&quot;, &quot;MCCC1&quot;, &quot;NDUFA6&quot;, &quot;NDUFV2&quot;, &quot;NELFB&quot;, &quot;PABPN1&quot;, &quot;PLIN1&quot;, &quot;PLSCR1&quot;, &quot;POLR2E&quot;, &quot;PPIF&quot;, &quot;PTGES2&quot;, &quot;PTGES3&quot;, &quot;PTGS1&quot;, &quot;QTRT1&quot;, &quot;RPL18&quot;, &quot;RPL37A&quot;, &quot;SMNDC1&quot;, &quot;SNRPD3&quot;, &quot;TIMM10&quot;, &quot;TPD52&quot;, &quot;XRCC1&quot;, &quot;CAV1&quot;, &quot;EFEMP1&quot;, &quot;PON1&quot;, &quot;PRR12&quot;, &quot;SNX7&quot;, &quot;STX18&quot;, &quot;AKR1C3&quot;, &quot;ARL6IP4&quot;, &quot;ATAD1&quot;, &quot;C8orf82&quot;, &quot;CIAO1&quot;, &quot;CIRH1A&quot;, &quot;DDX56&quot;, &quot;EYA4&quot;, &quot;FAM177A1&quot;, &quot;HEXIM1&quot;, &quot;HINT2&quot;, &quot;HMGA1&quot;, &quot;IRGQ&quot;, &quot;KATNAL2&quot;, &quot;LAMTOR1&quot;, &quot;LLPH&quot;, &quot;MBNL1&quot;, &quot;MCM5&quot;, &quot;MVD&quot;, &quot;NDUFA6&quot;, &quot;NFKB1&quot;, &quot;NFS1&quot;, &quot;NFX1&quot;, &quot;NUCKS1&quot;, &quot;OPTN&quot;, &quot;POLR2L&quot;, &quot;PON2&quot;, &quot;RAD21&quot;, &quot;RBM28&quot;, &quot;RPL29&quot;, &quot;SEC61B&quot;, &quot;SF3B4&quot;, &quot;SLC9A3R2&quot;, &quot;SUPT6H&quot;, &quot;TIMMDC1&quot;, &quot;TRA2A&quot;)

plot_volcano_3(dep, contrast=test, add_names=T,label_size=4,adjusted = F)</code></pre>
<p><img src="figure/Integration_old_analysis_T1_TME1.Rmd/V45-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="unique-table-4" class="section level5">
<h5>Unique Table</h5>
<pre class="r"><code>results %&gt;%  select(starts_with(&quot;name&quot;) | starts_with(test)) %&gt;% filter(name %in% unique_vec) %&gt;% arrange(across(ends_with(&quot;p.adj&quot;))) %&gt;% mutate(across(2:3, format, scientific = T, digits = 2)) %&gt;% dplyr::rename_with(~ tolower(gsub(paste0(test,&quot;_&quot;), &quot;&quot;, .x)), .cols = starts_with(test)) %&gt;% DT::datatable()</code></pre>
<div id="htmlwidget-d0279b66aa2c8b2f9403" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-d0279b66aa2c8b2f9403">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60"],["STX18","MVD","SF3B4","AARS2","TIMM10","NUCKS1","RPL18","RPL29","AKR1C3","KATNAL2","POLR2E","FUT8","TRA2A","EYA4","SNRPD3","PABPN1","DDX56","TPD52","HMGA1","NDUFA6","OPTN","ABCA8","SEC61B","LAMTOR1","PLIN1","NDUFV2","ATAD1","PON1","BAZ1B","SMNDC1","CIAO1","PON2","PPIF","SLC9A3R2","PLSCR1","HEXIM1","CIRH1A","MBNL1","POLR2L","TIMMDC1","MCM5","FAM177A1","HINT2","PTGES3","RBM28","SUPT6H","IRGQ","NFX1","RPL37A","NFKB1","CCDC58","NFS1","RAD21","C8orf82","LLPH","CAV1","ARL6IP4","EFEMP1","PTGES2","PTGS1"],["6.1e-05","1.7e-04","1.8e-04","2.0e-04","2.3e-04","5.2e-04","4.6e-04","7.4e-04","1.6e-03","1.6e-03","2.3e-03","4.1e-03","5.9e-03","7.2e-03","8.0e-03","8.3e-03","9.0e-03","9.5e-03","9.9e-03","1.0e-02","1.0e-02","1.4e-02","1.4e-02","1.4e-02","1.6e-02","1.6e-02","1.7e-02","2.1e-02","3.2e-02","3.4e-02","3.7e-02","4.4e-02","4.4e-02","5.3e-02","5.6e-02","5.7e-02","6.2e-02","6.3e-02","8.0e-02","9.0e-02","9.1e-02","9.7e-02","9.6e-02","9.7e-02","1.1e-01","1.1e-01","2.1e-01","2.2e-01","2.4e-01","2.4e-01","2.7e-01","2.7e-01","3.1e-01","3.1e-01","3.9e-01","4.2e-01","5.1e-01","5.1e-01","5.9e-01","9.6e-01"],["1.0e-02","1.7e-02","1.7e-02","1.7e-02","1.9e-02","2.4e-02","2.4e-02","3.1e-02","4.1e-02","4.2e-02","4.9e-02","6.3e-02","7.8e-02","8.4e-02","8.8e-02","8.9e-02","9.1e-02","9.4e-02","9.7e-02","9.7e-02","9.8e-02","1.1e-01","1.1e-01","1.1e-01","1.2e-01","1.2e-01","1.2e-01","1.4e-01","1.6e-01","1.7e-01","1.8e-01","1.9e-01","1.9e-01","2.0e-01","2.1e-01","2.1e-01","2.2e-01","2.2e-01","2.5e-01","2.6e-01","2.6e-01","2.7e-01","2.7e-01","2.7e-01","2.9e-01","2.9e-01","4.0e-01","4.2e-01","4.4e-01","4.4e-01","4.7e-01","4.7e-01","5.1e-01","5.1e-01","5.7e-01","5.9e-01","6.8e-01","6.8e-01","7.4e-01","9.8e-01"],[true,true,true,true,true,true,true,true,true,true,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],[-4.67,3.77,3.67,4.89,3.78,3.7,5.31,3.59,4.82,4.54,3.5,4.56,2.19,3.49,4.46,2.95,2.26,2.61,4.26,2.19,2.29,2.65,3.33,2.28,2.17,2.31,2.33,-2.39,2.49,1.87,2.41,2.5,2.32,1.89,1.67,1.63,2.03,2.09,1.27,1.39,1.91,1.57,1.46,2.17,1.46,1.58,1.36,1.06,1.3,0.938,1.43,1.01,1.19,1.24,0.81,1.21,0.583,1.12,0.829,0.0447]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>name<\/th>\n      <th>p.val<\/th>\n      <th>p.adj<\/th>\n      <th>significant<\/th>\n      <th>ratio<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":5},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>expdesign &lt;- read.table(here::here(&quot;data/expdesign_old_exp.tsv&quot;), header = T, stringsAsFactors = F)

cond_1&lt;-&quot;T1_P7&quot;
cond_2&lt;-&quot;TME1_P7&quot;

conditions&lt;-c(cond_1,cond_2)

expdesign &lt;- subset(expdesign, condition %in% conditions)

test&lt;- paste(cond_1,cond_2,sep=&quot;_vs_&quot;)

columns&lt;-match(paste(&quot;LFQ.intensity.&quot;,expdesign$label,sep=&quot;&quot;),colnames(data_unique))

data_se &lt;- make_se(data_unique, columns, expdesign)</code></pre>
</div>
</div>
</div>
<div id="t1_p7_vs_tme1_p7" class="section level3">
<h3>T1_P7_vs_TME1_P7</h3>
<pre class="r"><code>data_filt &lt;- filter_missval(data_se, thr = 1)

set.seed(20210520)
data_imp_man &lt;- impute(data_filt, fun = &quot;man&quot;, shift = 1.8, scale = 0.3)

data_diff &lt;- test_diff_BH(data_imp_man, type = &quot;all&quot;)

dep &lt;- add_rejections(data_diff, alpha = 0.05, lfc = 1)

results&lt;-get_results(dep)</code></pre>
<p>There are <strong>160</strong> significant proteins in <strong>T1_P7_vs_TME1_P7</strong> comparison.</p>
<div id="visualization-of-the-results-5" class="section level4 tabset">
<h4><strong>Visualization of the results:</strong></h4>
<hr />
<div id="volcano-plot-5" class="section level5">
<h5>Volcano Plot</h5>
<pre class="r"><code>plot_volcano(dep, contrast=test, add_names=T,label_size=4,adjusted = F)</code></pre>
<p><img src="figure/Integration_old_analysis_T1_TME1.Rmd/V35-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="results-table-5" class="section level5">
<h5>Results Table</h5>
<pre class="r"><code>results %&gt;% 
  filter(significant) %&gt;%
  select(-significant, -ends_with(&quot;centered&quot;)) %&gt;% 
  pivot_longer(
   cols = ends_with(c(&quot;p.val&quot;, &quot;p.adj&quot;, &quot;significant&quot;, &quot;ratio&quot;)),
   names_to = c(&quot;comparison&quot;, &quot;.value&quot;),
   names_pattern = &quot;(.*_.*_.*_.*_.*)_(.*)&quot;) %&gt;% 
  arrange(name, p.adj) %&gt;% 
  relocate(comparison) %&gt;% 
  select(-comparison) %&gt;% 
  mutate(across(ends_with(c(&quot;p.val&quot;, &quot;p.adj&quot;)), format, scientific = T, digits = 2)) %&gt;% 
  mutate(significant = str_to_title(significant)) %&gt;% 
  DT::datatable(options = list(
  columnDefs = list(list(className = &#39;dt-center&#39;, targets = 1:6)),
  pageLength = 10))</code></pre>
<div id="htmlwidget-4c8ee3b12d7e6c81de8f" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4c8ee3b12d7e6c81de8f">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160"],["AARS2","ABCD3","ACE","ACOX1","AGAP1","APOA2","APOH","ATAD1","B4DLN1","BANF1","BAZ1B","BGN.1","BNIP1","BSG","BTF3","C16orf46","CALD1.2","CDH11","CIAO1","CNOT1","CNPY4","COL18A1","COL1A1","COL1A2","COL3A1","COL6A3.1","CPB2","CPNE1","CSK","CSRP1","CSTF3","CTHRC1","DAB2","DCN","DDT","DNAJC9","DSG2","ECHDC3","ECI1","ECI2","EIF3A","ERO1L","ESRP1","EYA4","FAM114A1","FAM177A1","FBLN1","FBLN2","FBXO22","FCN3","FGB","FHL3","FKBP9","FN1","FTL","FUT8","GALNT5","GBAS","GPX8","GRIPAP1","GTF2I","GYS1","HEXIM1","HLA-A.2","HLA-DRB1","HMGB2","HNRNPC","HP","HPX","HSDL2","IGHA1","IGHG2","IGLC6","INTS5","IRGQ","ISLR","ITGAM","KATNAL2","KRT18","LGALS1","LIMA1","LLGL2","LRRFIP2","LSM8","LUM","LXN","MAN2A1","MARCKS","MCCC1","MDH2","METTL14","MEX3A","MMP14","MUC1","MYL9","NELFB","NIPSNAP1","NIPSNAP3A","NPM1","NUDCD2","NUP98","OLA1","ORM2","P01721","P0DP04","PAFAH1B2","PCBD1","PCBP1","PDCD6","PDLIM3","PDLIM5","PEAK1","PKP2","PLP2","PRDX3","PRRC1","PTGES2","RALBP1","RB1CC1","RCC1","RDX","RMND1","RPL29","RPSA","S100A10","S100A14","SART1","SDHB","SEC61B","SERPINF2","SH3BGRL","SLC20A2","SLC25A22","SLC25A6","SLC35E1","SLK","SNAP23","SNRPA","SNRPD3","SNRPG","SNX7","SPTLC2","SRSF3","STAU1","TAF5","TAGLN","THBS3","THG1L","TMED9","TOR1AIP1.1","TP53I3","TPM1","TPM4","TRIM23","TUBG1","TUBGCP2","TXNDC5","UPF2","VIM","VPS13C"],["Q5JTZ9","A0A3B3ITW3","F6X3S4","Q15067","Q9UPQ3","V9GYM3","P02749","Q8NBU5","B4DLN1","O75531","Q9UIG0","P21810","Q12981","P35613","P20290","Q6P387","Q05682","P55287","O76071","A5YKK6","A0A0J9YYJ0","P39060","P02452","A0A087WTA8","P02461","P12111","A0A087WSY5","Q99829","P41240","P21291","E9PLP8","Q96CG8","P98082","P07585","P30046","Q8WXX5","Q14126","Q96DC8","P42126","A0A0C4DGA2","Q14152","Q96HE7","A0A2U3TZN9","E7ESD5","Q8IWE2","G3V583","B1AHL2","P98095","Q8NEZ5","O75636","P02675","Q13643","O95302","P02751","P02792","Q9BYC5","Q7Z7M9","O75323","Q8TED1","A0A087WT45","P78347","P13807","O94992","A0A140T975","A2BFX2","P26583","G3V4C1","P00738","P02790","Q6YN16","A0A286YEY1","P01859","P0DOY3","Q6P9B9","Q8WZA9","O14498","P11215","K7EM02","P05783","P09382","Q9UHB6","J3QRV5","Q9Y608","O95777","P51884","Q9BS40","Q16706","P29966","E9PHF7","P40926","Q9HCE5","A1L020","P50281","A0A087WWM1","P24844","A0A5H1ZRP4","Q9BPW8","Q9UFN0","P06748","E5RFP0","P52948","J3KQ32","P19652","P01721","P0DP04","P68402","P61457","Q15365","O75340","A0A087WYF8","Q96HC4","Q9H792","Q99959","Q04941","P30048","Q96M27","Q9H7Z7","Q15311","Q8TDY2","P18754","A0A2R8Y5S7","A0A2R8Y4P5","A0A3B3ITT5","C9J9K3","P60903","Q9HCY8","O43290","P21912","P60468","P08697","O75368","Q08357","Q9H936","P12236","Q96K37","Q9H2G2","O00161","P09012","P62318","Q49AN9","Q9UNH6","O15270","P84103","A0A087X1A5","Q15542","Q01995","P49746","Q9NWX6","Q9BVK6","Q5JTV8","Q53FA7","A0A0S2Z4G6","P67936","P36406","P23258","Q9BSJ2","Q8NBS9","Q9HAU5","P08670","Q709C8"],["1.2e-04","3.2e-04","5.7e-06","5.5e-04","1.3e-04","3.3e-04","9.7e-04","1.2e-03","1.8e-04","1.8e-03","1.4e-04","1.5e-03","2.7e-03","2.7e-03","3.4e-03","2.1e-07","1.6e-04","1.5e-03","6.9e-04","3.2e-03","9.7e-04","1.6e-03","8.0e-06","1.5e-05","2.0e-04","4.6e-04","9.7e-04","2.1e-04","3.5e-04","4.8e-06","3.3e-03","1.1e-03","3.0e-03","6.3e-04","2.8e-04","2.2e-05","1.3e-03","3.0e-03","5.4e-04","2.9e-03","3.3e-03","1.9e-03","1.6e-03","2.6e-03","1.1e-03","8.2e-04","2.0e-04","3.4e-03","2.6e-03","2.1e-04","2.4e-03","3.4e-03","1.9e-03","3.1e-03","1.2e-03","7.8e-06","1.3e-08","7.8e-04","4.9e-04","3.3e-03","3.8e-04","2.4e-03","1.1e-03","1.6e-04","2.4e-05","3.1e-03","1.7e-03","2.3e-03","1.3e-03","2.3e-03","6.2e-04","1.1e-03","4.7e-04","8.0e-05","2.4e-03","7.9e-04","3.2e-03","3.2e-05","2.2e-04","4.1e-04","5.5e-04","2.8e-03","1.1e-03","2.2e-03","2.1e-05","3.1e-03","2.0e-03","2.3e-04","6.6e-04","1.7e-03","9.4e-05","2.4e-03","1.8e-03","4.9e-04","4.0e-04","3.4e-03","1.0e-03","1.1e-03","1.6e-03","1.8e-03","2.4e-04","2.6e-03","3.1e-03","4.5e-05","5.0e-05","2.1e-04","2.0e-03","2.2e-03","5.7e-04","3.3e-04","3.0e-03","2.7e-04","5.1e-04","4.5e-04","1.0e-03","2.5e-03","2.3e-05","8.7e-04","4.8e-04","1.4e-05","2.4e-03","3.0e-09","1.9e-03","1.6e-03","3.1e-03","7.7e-05","1.2e-03","5.1e-04","3.4e-05","9.4e-04","3.3e-04","7.0e-06","2.7e-03","2.2e-03","1.1e-04","3.0e-03","2.9e-03","2.6e-05","1.8e-03","2.7e-03","6.9e-05","3.0e-03","1.5e-03","1.6e-03","2.3e-03","1.0e-04","6.1e-05","3.1e-03","1.8e-05","7.0e-05","6.6e-05","8.3e-05","8.6e-04","7.8e-04","3.4e-03","1.0e-03","3.4e-03","1.6e-04","2.0e-03","3.7e-04"],["9.2e-03","1.5e-02","2.3e-03","1.9e-02","9.2e-03","1.5e-02","2.7e-02","2.9e-02","1.1e-02","3.8e-02","9.9e-03","3.5e-02","4.7e-02","4.7e-02","5.0e-02","1.6e-04","1.0e-02","3.5e-02","2.2e-02","4.9e-02","2.7e-02","3.6e-02","2.3e-03","3.5e-03","1.2e-02","1.8e-02","2.7e-02","1.2e-02","1.5e-02","2.3e-03","5.0e-02","2.9e-02","4.9e-02","2.1e-02","1.4e-02","3.7e-03","3.1e-02","4.9e-02","1.9e-02","4.9e-02","5.0e-02","4.0e-02","3.6e-02","4.6e-02","2.9e-02","2.5e-02","1.2e-02","5.0e-02","4.6e-02","1.2e-02","4.5e-02","5.0e-02","3.9e-02","4.9e-02","3.1e-02","2.3e-03","1.5e-05","2.4e-02","1.8e-02","5.0e-02","1.6e-02","4.5e-02","2.8e-02","1.0e-02","3.7e-03","4.9e-02","3.8e-02","4.4e-02","3.2e-02","4.4e-02","2.1e-02","2.9e-02","1.8e-02","7.1e-03","4.5e-02","2.4e-02","4.9e-02","4.3e-03","1.2e-02","1.7e-02","1.9e-02","4.8e-02","2.9e-02","4.3e-02","3.7e-03","4.9e-02","4.1e-02","1.2e-02","2.1e-02","3.8e-02","7.8e-03","4.5e-02","3.9e-02","1.8e-02","1.7e-02","5.0e-02","2.8e-02","2.8e-02","3.6e-02","3.9e-02","1.2e-02","4.6e-02","4.9e-02","5.5e-03","5.8e-03","1.2e-02","4.1e-02","4.3e-02","1.9e-02","1.5e-02","4.9e-02","1.3e-02","1.8e-02","1.8e-02","2.8e-02","4.5e-02","3.7e-03","2.6e-02","1.8e-02","3.5e-03","4.5e-02","7.0e-06","4.0e-02","3.6e-02","4.9e-02","7.1e-03","3.0e-02","1.8e-02","4.3e-03","2.7e-02","1.5e-02","2.3e-03","4.7e-02","4.3e-02","8.2e-03","4.9e-02","4.8e-02","3.8e-03","3.9e-02","4.7e-02","6.7e-03","4.9e-02","3.6e-02","3.6e-02","4.4e-02","8.2e-03","6.7e-03","4.9e-02","3.7e-03","6.7e-03","6.7e-03","7.1e-03","2.6e-02","2.4e-02","5.0e-02","2.8e-02","5.0e-02","1.0e-02","4.0e-02","1.6e-02"],["True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True"],[5.06,3.91,-5.04,2.93,-4.06,-3.26,-3.17,3.14,3.75,2.81,4.11,-4.84,-4.33,2.89,3.35,-7.79,-4.17,-6.99,3.2,3.08,-3.72,-3.19,-7.73,-7.55,-5.42,-5.86,3.2,3.48,3.45,6.04,2.83,-3.47,-2.64,-4.08,4.13,-8.12,3.76,3.95,3.76,-3.23,2.46,2.99,3.36,3.85,-3.24,2.84,-4.37,-2.56,2.62,3.35,-4.4,-3,-2.85,-4.12,-3.9,6.5,-9.89,3.35,-3.62,3.56,4.08,2.69,2.77,-3.65,4.54,2.27,2.76,-2.67,-4.4,3.18,-3.78,-4.63,-6.4,4.02,2.39,-3.95,2.85,4.5,6.63,-5.26,-3.12,-3.82,-3.87,3.57,-5.4,-3.26,2.86,-4.72,3.25,2.64,-4.03,4.5,-3.02,3.63,-4.02,2.23,4.18,3.33,3.28,3.17,4.03,2.89,-3.07,-4.81,-4.09,4.17,-3.85,2.48,3.82,-3.18,-2.76,-3.65,3.93,-3.68,2.91,2.93,4.71,-3.36,-5.85,4.71,3.51,13.4,5.53,2.64,5.13,-4.54,2.67,3.25,4.74,-5.27,-4.02,-7.11,3.45,5.97,-4.13,3.5,3.32,4.68,5.39,-3.95,-4.24,6.08,3.44,3.19,-6.2,-4.11,4.79,-3.29,-4.92,5.37,-4.45,-5.19,-2.96,3.62,2.83,6.01,-3.46,-4.14,-3.36,5.26]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>name<\/th>\n      <th>ID<\/th>\n      <th>p.val<\/th>\n      <th>p.adj<\/th>\n      <th>significant<\/th>\n      <th>ratio<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-center","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"pageLength":10,"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="old-plot-unique-5" class="section level5">
<h5>Old plot + Unique</h5>
<pre class="r"><code>unique_vec &lt;- c(&quot;AARS2&quot;, &quot;ABCA8&quot;, &quot;BAZ1B&quot;, &quot;CCDC58&quot;, &quot;COMT&quot;, &quot;FAF1&quot;, &quot;FUT8&quot;, &quot;GATAD2B&quot;, &quot;INTS5&quot;, &quot;MBLAC2&quot;, &quot;MCCC1&quot;, &quot;NDUFA6&quot;, &quot;NDUFV2&quot;, &quot;NELFB&quot;, &quot;PABPN1&quot;, &quot;PLIN1&quot;, &quot;PLSCR1&quot;, &quot;POLR2E&quot;, &quot;PPIF&quot;, &quot;PTGES2&quot;, &quot;PTGES3&quot;, &quot;PTGS1&quot;, &quot;QTRT1&quot;, &quot;RPL18&quot;, &quot;RPL37A&quot;, &quot;SMNDC1&quot;, &quot;SNRPD3&quot;, &quot;TIMM10&quot;, &quot;TPD52&quot;, &quot;XRCC1&quot;, &quot;CAV1&quot;, &quot;EFEMP1&quot;, &quot;PON1&quot;, &quot;PRR12&quot;, &quot;SNX7&quot;, &quot;STX18&quot;, &quot;BNIP1&quot;, &quot;C16orf46&quot;, &quot;COX7C&quot;, &quot;GALNT5&quot;, &quot;GLA&quot;, &quot;NRAS&quot;, &quot;OLFML3&quot;, &quot;PCBD1&quot;, &quot;PLP2&quot;, &quot;S100A14&quot;, &quot;THG1L&quot;)

plot_volcano_3(dep, contrast=test, add_names=T,label_size=4,adjusted = F)</code></pre>
<p><img src="figure/Integration_old_analysis_T1_TME1.Rmd/V47-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="unique-table-5" class="section level5">
<h5>Unique Table</h5>
<pre class="r"><code>results %&gt;%  select(starts_with(&quot;name&quot;) | starts_with(test)) %&gt;% filter(name %in% unique_vec) %&gt;% arrange(across(ends_with(&quot;p.adj&quot;))) %&gt;% mutate(across(2:3, format, scientific = T, digits = 2)) %&gt;% dplyr::rename_with(~ tolower(gsub(paste0(test,&quot;_&quot;), &quot;&quot;, .x)), .cols = starts_with(test)) %&gt;% DT::datatable()</code></pre>
<div id="htmlwidget-1cd6c7bc4e04ef3272b5" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1cd6c7bc4e04ef3272b5">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40"],["GALNT5","C16orf46","FUT8","PTGES2","SNX7","INTS5","S100A14","AARS2","BAZ1B","PLP2","MCCC1","SNRPD3","PCBD1","BNIP1","THG1L","NELFB","PRR12","STX18","PON1","PLIN1","QTRT1","TIMM10","NDUFA6","CCDC58","NRAS","FAF1","COX7C","POLR2E","GLA","OLFML3","MBLAC2","XRCC1","PTGES3","CAV1","RPL18","PTGS1","GATAD2B","PABPN1","EFEMP1","COMT"],["1.3e-08","2.1e-07","7.8e-06","2.3e-05","6.9e-05","8.0e-05","7.7e-05","1.2e-04","1.4e-04","4.5e-04","6.6e-04","1.8e-03","2.0e-03","2.7e-03","3.1e-03","3.4e-03","3.6e-03","4.3e-03","5.3e-03","8.3e-03","8.7e-03","8.6e-03","1.3e-02","1.3e-02","1.5e-02","1.7e-02","1.9e-02","2.0e-02","2.4e-02","2.6e-02","4.5e-02","5.1e-02","5.5e-02","8.5e-02","1.1e-01","1.5e-01","1.9e-01","3.0e-01","3.2e-01","6.4e-01"],["1.5e-05","1.6e-04","2.3e-03","3.7e-03","6.7e-03","7.1e-03","7.1e-03","9.2e-03","9.9e-03","1.8e-02","2.1e-02","3.9e-02","4.1e-02","4.7e-02","4.9e-02","5.0e-02","5.1e-02","5.4e-02","6.0e-02","7.5e-02","7.6e-02","7.6e-02","8.8e-02","8.8e-02","9.3e-02","1.0e-01","1.1e-01","1.1e-01","1.2e-01","1.2e-01","1.6e-01","1.8e-01","1.8e-01","2.3e-01","2.7e-01","3.2e-01","3.7e-01","4.8e-01","5.0e-01","7.6e-01"],[true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],[-9.89,-7.79,6.5,4.71,-4.24,4.02,-4.54,5.06,4.11,-3.68,3.25,5.39,-3.85,-4.33,-3.29,2.23,-3.45,-3.6,-2.86,2.54,3.7,2.92,2.21,2.28,-2.42,2.96,-2.64,2.04,-1.78,-2.21,2.53,1.52,2.34,-1.71,2.81,1.66,1.09,1.36,1.44,0.374]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>name<\/th>\n      <th>p.val<\/th>\n      <th>p.adj<\/th>\n      <th>significant<\/th>\n      <th>ratio<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":5},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</div>
</div>
<div id="section" class="section level2 unnumbered">
<h2></h2>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=Italian_Italy.1252  LC_CTYPE=Italian_Italy.1252   
[3] LC_MONETARY=Italian_Italy.1252 LC_NUMERIC=C                  
[5] LC_TIME=Italian_Italy.1252    

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] naniar_0.6.1                visdat_0.5.3               
 [3] datapasta_3.1.0             here_1.0.1                 
 [5] patchwork_1.1.1             stringr_1.4.0              
 [7] DT_0.18                     enrichR_3.0                
 [9] ggplot2_3.3.3               tibble_3.1.0               
[11] preprocessCore_1.52.1       SummarizedExperiment_1.20.0
[13] Biobase_2.50.0              GenomicRanges_1.42.0       
[15] GenomeInfoDb_1.26.2         IRanges_2.24.1             
[17] S4Vectors_0.28.1            BiocGenerics_0.36.0        
[19] MatrixGenerics_1.2.1        matrixStats_0.58.0         
[21] DEP_1.12.0                  tidyr_1.1.3                
[23] dplyr_1.0.5                 MBQN_2.2.0                 
[25] workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] circlize_0.4.12        BiocFileCache_1.14.0   plyr_1.8.6            
  [4] gmm_1.6-6              shinydashboard_0.7.1   crosstalk_1.1.1       
  [7] BiocParallel_1.24.1    digest_0.6.27          foreach_1.5.1         
 [10] htmltools_0.5.1.1      fansi_0.4.2            magrittr_2.0.1        
 [13] memoise_2.0.0          cluster_2.1.0          doParallel_1.0.16     
 [16] limma_3.46.0           ComplexHeatmap_2.6.2   readr_1.4.0           
 [19] imputeLCMD_2.0         sandwich_3.0-0         colorspace_2.0-0      
 [22] blob_1.2.1             rappdirs_0.3.3         ggrepel_0.9.1         
 [25] xfun_0.23              crayon_1.4.1           RCurl_1.98-1.2        
 [28] jsonlite_1.7.2         impute_1.64.0          zoo_1.8-9             
 [31] iterators_1.0.13       glue_1.4.2             gtable_0.3.0          
 [34] zlibbioc_1.36.0        XVector_0.30.0         GetoptLong_1.0.5      
 [37] DelayedArray_0.16.2    shape_1.4.5            scales_1.1.1          
 [40] vsn_3.58.0             mvtnorm_1.1-1          DBI_1.1.1             
 [43] Rcpp_1.0.6             mzR_2.24.1             xtable_1.8-4          
 [46] clue_0.3-58            bit_4.0.4              htmlwidgets_1.5.3     
 [49] httr_1.4.2             RColorBrewer_1.1-2     ellipsis_0.3.1        
 [52] farver_2.1.0           pkgconfig_2.0.3        XML_3.99-0.6          
 [55] sass_0.3.1             dbplyr_2.1.1           utf8_1.2.1            
 [58] labeling_0.4.2         tidyselect_1.1.1       rlang_0.4.10          
 [61] later_1.1.0.1          munsell_0.5.0          tools_4.0.3           
 [64] cachem_1.0.4           generics_0.1.0         RSQLite_2.2.5         
 [67] evaluate_0.14          fastmap_1.1.0          mzID_1.28.0           
 [70] yaml_2.2.1             knitr_1.31             bit64_4.0.5           
 [73] fs_1.5.0               purrr_0.3.4            ncdf4_1.17            
 [76] whisker_0.4            mime_0.10              compiler_4.0.3        
 [79] curl_4.3               png_0.1-7              affyio_1.60.0         
 [82] bslib_0.2.4            stringi_1.5.3          highr_0.8             
 [85] MSnbase_2.15.7         lattice_0.20-41        ProtGenerics_1.22.0   
 [88] Matrix_1.2-18          tmvtnorm_1.4-10        vctrs_0.3.7           
 [91] pillar_1.6.0           norm_1.0-9.5           lifecycle_1.0.0       
 [94] BiocManager_1.30.10    jquerylib_0.1.3        MALDIquant_1.19.3     
 [97] GlobalOptions_0.1.2    bitops_1.0-6           httpuv_1.5.5          
[100] R6_2.5.0               pcaMethods_1.82.0      affy_1.68.0           
[103] promises_1.2.0.1       codetools_0.2-16       MASS_7.3-53           
[106] assertthat_0.2.1       rprojroot_2.0.2        rjson_0.2.20          
[109] withr_2.4.2            GenomeInfoDbData_1.2.4 hms_1.0.0             
[112] grid_4.0.3             rmarkdown_2.7          Cairo_1.5-12.2        
[115] git2r_0.28.0           shiny_1.6.0           </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
