<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Fabio Bedin | MS-Unit" />


<title>TME1 vs TME2 populations</title>

<script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.18/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<script src="site_libs/plotly-binding-4.9.4.1/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-1.57.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotly-main-1.57.1/plotly-latest.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">LN OvCa dataset</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/FabioBedin/OC_Nezi">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">TME1 vs TME2 populations</h1>
<h4 class="author">Fabio Bedin | MS-Unit</h4>
<h4 class="date">13 luglio 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-07-23
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>OC_Nezi/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210520code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210520)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210520code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210520)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomFabioBedinOCNezitree66a229c24654952ca2a5d583d329ede22dac2cd5targetblank66a229ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/FabioBedin/OC_Nezi/tree/66a229c24654952ca2a5d583d329ede22dac2cd5" target="_blank">66a229c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomFabioBedinOCNezitree66a229c24654952ca2a5d583d329ede22dac2cd5targetblank66a229ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/FabioBedin/OC_Nezi/tree/66a229c24654952ca2a5d583d329ede22dac2cd5" target="_blank">66a229c</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.RData

Untracked files:
    Untracked:  Rplot.png
    Untracked:  Rplot01.png
    Untracked:  T1_vs_T2.tex
    Untracked:  UP_p1_t1_t2.txt
    Untracked:  analysis/T2_R_vs_T2_S_gene_name.txt
    Untracked:  analysis/test.cys
    Untracked:  code/custom_functions.R
    Untracked:  data/anova_table_t1_t2.txt
    Untracked:  data/anova_table_t1_t2_filt_1500.txt
    Untracked:  data/anova_table_t1_tme1.txt
    Untracked:  data/anova_table_t1_tme1_filt_1500.txt
    Untracked:  data/anova_table_t2_tme2.txt
    Untracked:  data/anova_table_t2_tme2_filt_1500.txt
    Untracked:  data/expdesign.tsv
    Untracked:  data/expdesign_SI_S_R.tsv
    Untracked:  data/expdesign_old_exp.tsv
    Untracked:  data/proteinGroups.txt
    Untracked:  gene_name_BP_anova.txt
    Untracked:  output/GO_preliminary_LN.pptx
    Untracked:  output/results_T1_vs_TME1_patients.xlsx
    Untracked:  output/results_T1_vs_TME1_population.xlsx
    Untracked:  output/results_T1_vs_TME1_sub-populations.xlsx
    Untracked:  output/results_T2_vs_T1_patients.xlsx
    Untracked:  output/results_T2_vs_T1_population.xlsx
    Untracked:  output/results_T2_vs_T1_sub-populations.xlsx
    Untracked:  output/results_T2_vs_TME2_patients.xlsx
    Untracked:  output/results_T2_vs_TME2_population.xlsx
    Untracked:  output/results_T2_vs_TME2_sub-populations.xlsx
    Untracked:  output/results_TME2_vs_TME1_population.xlsx
    Untracked:  site_libs/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/TME2_vs_TME1_population.Rmd</code>) and HTML (<code>docs/TME2_vs_TME1_population.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/FabioBedin/OC_Nezi/blob/66a229c24654952ca2a5d583d329ede22dac2cd5/analysis/TME2_vs_TME1_population.Rmd" target="_blank">66a229c</a>
</td>
<td>
FabioBedin
</td>
<td>
2021-07-23
</td>
<td>
Final update 23-07-2021
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="ovarian-cancer-dataset" class="section level1">
<h1>Ovarian cancer dataset</h1>
<hr />
<pre class="r"><code>knitr::opts_chunk$set(fig.align = &quot;center&quot;)
library(&quot;MBQN&quot;)
library(&quot;dplyr&quot;)
library(&quot;tidyr&quot;)
library(&quot;DEP&quot;)
library(&quot;SummarizedExperiment&quot;)
library(&quot;preprocessCore&quot;)
library(&quot;tibble&quot;)
library(&quot;ggplot2&quot;)
library(&quot;enrichR&quot;)
library(&quot;DT&quot;)
library(&quot;stringr&quot;)
library(&quot;patchwork&quot;)
library(&quot;here&quot;)
library(&quot;datapasta&quot;)
library(&quot;gprofiler2&quot;)
library(&quot;org.Hs.eg.db&quot;)
library(&quot;rrvgo&quot;)
library(&quot;openxlsx&quot;)
select &lt;- dplyr::select</code></pre>
<pre class="r"><code>source(here::here(&quot;code/custom_functions.R&quot;))</code></pre>
<pre class="r"><code>header_style &lt;- createStyle(
  fontSize = 12,
  fontColour = &quot;#0f0f0f&quot;,
  fgFill = &quot;#faf2ca&quot;,
  halign = &quot;center&quot;,
  border = &quot;TopBottomLeftRight&quot;
)

body_style &lt;- createStyle(
  halign = &quot;center&quot;,
  border = &quot;TopBottomLeftRight&quot;
)

excel &lt;- createWorkbook()</code></pre>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<hr />
<p>For this analysis i will analyze <strong>TME2</strong> and <strong>TME1</strong> populations, to indentify interesting HITs .</p>
<pre class="r"><code>data &lt;- read.csv(here::here(&quot;data/proteinGroups.txt&quot;), header = TRUE,stringsAsFactors = FALSE, sep = &quot;\t&quot;)

expdesign &lt;- read.table(here::here(&quot;data/expdesign.tsv&quot;), header = T, stringsAsFactors = F)

cond_1&lt;-&quot;TME2&quot;

cond_2&lt;-&quot;TME1&quot;

conditions&lt;-c(cond_1,cond_2)

test&lt;- paste(cond_1,cond_2,sep=&quot;_vs_&quot;)

expdesign &lt;- subset(expdesign, condition %in% conditions)</code></pre>
</div>
<div id="data-wrangling" class="section level2">
<h2>Data wrangling</h2>
<hr />
<p>First, we need to wrangling the original dataset. From the <strong>proteinGroups</strong> obtained by <strong>MaxQuant</strong> software, remove proteins that are marked form the software as <em>potential contaminant</em>, <em>only identify by site</em> and <em>reverse</em>.</p>
<pre class="r"><code>data &lt;- data[data$Reverse != &quot;+&quot; &amp; data$Potential.contaminant != &quot;+&quot; &amp; data$Only.identified.by.site != &quot;+&quot;,]

data_unique &lt;- make_unique(data, &quot;Gene.names&quot;, &quot;Protein.IDs&quot;, delim = &quot;;&quot;)

columns&lt;-match(paste(&quot;LFQ.intensity.&quot;,expdesign$label,sep=&quot;&quot;),colnames(data_unique))

data_se &lt;- make_se(data_unique, columns, expdesign)</code></pre>
<p>To understand the structure of the dataset and define a strategy, I use different quality control polts:</p>
<div id="quality-control-plots" class="section level3 tabset tabset-fade tabset-pills">
<h3>Quality control Plots</h3>
<p>All this plot can also hepl to decide the strategy to filter the missing data.</p>
<div id="frequency-plot" class="section level4">
<h4>Frequency plot</h4>
<pre class="r"><code>plot_frequency(data_se)</code></pre>
<p><img src="figure/TME2_vs_TME1_population.Rmd/plot_freq-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="number-of-protein-per-sample" class="section level4">
<h4>Number of protein per sample</h4>
<pre class="r"><code>plot_numbers(data_se)</code></pre>
<p><img src="figure/TME2_vs_TME1_population.Rmd/plot%20numbers-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="distibution-of-each-sample" class="section level4">
<h4>Distibution of each sample</h4>
<pre class="r"><code>plot_normalization(data_se)</code></pre>
<p><img src="figure/TME2_vs_TME1_population.Rmd/plot_norm-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="the-problem-of-missing-data" class="section level2">
<h2>The problem of missing data</h2>
<hr />
<p>In the original dataset, the percentage of missing data is <strong>73.58%</strong> and can be visualized with the following heatmap.</p>
<pre class="r"><code>plot_missval(data_se)</code></pre>
<p><img src="figure/TME2_vs_TME1_population.Rmd/plot_mv-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="filter-missing-data" class="section level2">
<h2>Filter missing data</h2>
<hr />
<p>To reduce the percentage of missing data I have decided to filter samples with less then <strong>1500</strong> identify proteins. Afrer that, I also use a second strong filter to remove all the proteins that are not identify in <strong>75%</strong> in each replicates in at least 1 groups.</p>
<pre class="r"><code>data_filt &lt;- filter_bad_samples(data_se, thr = 1500)

data_filt &lt;- filter_missval(data_filt, thr = 4)</code></pre>
<p>These plot show the effect of the filters. Now we reduce the percetage of missing data in the dataset down to: <strong>16.66%</strong>.</p>
<div id="plots-with-filters" class="section level3 tabset tabset-fade tabset-pills">
<h3>Plots with filters</h3>
<div id="protein-coverage" class="section level4">
<h4>protein coverage</h4>
<pre class="r"><code>plot_coverage(data_filt)</code></pre>
<p><img src="figure/TME2_vs_TME1_population.Rmd/plot_coverage-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="number-of-protein-per-sample-1" class="section level4">
<h4>Number of protein per sample</h4>
<pre class="r"><code>plot_numbers(data_filt)</code></pre>
<p><img src="figure/TME2_vs_TME1_population.Rmd/plot_numbers_filt-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="distibution" class="section level4">
<h4>Distibution</h4>
<pre class="r"><code>plot_normalization(data_filt)</code></pre>
<p><img src="figure/TME2_vs_TME1_population.Rmd/plot_norm_filt-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="plot-missing-data" class="section level4">
<h4>Plot missing data</h4>
<pre class="r"><code>plot_missval(data_filt)</code></pre>
<p><img src="figure/TME2_vs_TME1_population.Rmd/plot_mv_filt-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="unique" class="section level4">
<h4>Unique</h4>
<p>Proteins unique for each conditions after filters. This Interesting proteins will also be imputed for use in differential analysis.</p>
<pre class="r"><code>unique_pair(data_filt, conditions)</code></pre>
<pre><code>Joining, by = &quot;name&quot;</code></pre>
<div id="htmlwidget-dd20578417a55988ff21" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-dd20578417a55988ff21">{"x":{"filter":"none","data":[["1","2","3","4","5"],["APOBEC3C","GMFB","NTMT1","PLCB3","SACS"],[true,true,true,true,true],[false,false,false,false,false]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>name<\/th>\n      <th>unique_TME2<\/th>\n      <th>unique_TME1<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-center","targets":[1,2,3]},{"orderable":false,"targets":0}],"pageLength":10,"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'color':value == true ? \"green\" : value == false ? \"red\" : null});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'color':value == true ? \"green\" : value == false ? \"red\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
</div>
</div>
<div id="imputation" class="section level2">
<h2>Imputation</h2>
<hr />
<p>An idealized version of a label-free discovery mass spectrometry proteomics experiment would provide absolute abundance measurements for a whole proteome, across varying conditions. Unfortunately, this ideal is not realized. Measurements are made on peptides requiring an inferential step to obtain protein level estimates. The inference is complicated by experimental factors that necessitate relative abundance estimation and result in widespread <strong>non-ignorable missing data</strong>. Relative abundance on the log scale takes the form of parameter contrasts. In a complete-case analysis, contrast estimates may be biased by missing data and a substantial amount of useful information will often go unused.</p>
<p>To avoid this problem, we impute missing data using random draws from a manually defined left-shifted Gaussian distribution.</p>
<pre class="r"><code>set.seed(20210520)
data_imp_man &lt;- impute(data_filt, fun = &quot;man&quot;, shift = 1.8, scale = 0.3)</code></pre>
<p>This plot show the effect of imputation, as expected a new population of low intensity values as emerged.</p>
<pre class="r"><code>plot_imputation(data_filt, data_imp_man)</code></pre>
<p><img src="figure/TME2_vs_TME1_population.Rmd/plot_imputation-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="differential-enrichment-analysis" class="section level2">
<h2>Differential enrichment analysis</h2>
<hr />
<p>Protein-wise linear models combined with empirical Bayes statistics are used for the differential enrichment analysis (or differential expression analysis).</p>
<pre class="r"><code>data_diff &lt;- test_diff_BH(data_imp_man, type = &quot;manual&quot;, test = test)</code></pre>
<pre><code>Tested contrasts: TME2_vs_TME1</code></pre>
<p>I define sifnificant protein with <strong>FDR = 0.05</strong> and <strong>FC = 1</strong></p>
<pre class="r"><code>dep &lt;- add_rejections(data_diff, alpha = 0.05, lfc = 1)

results&lt;-get_results(dep)</code></pre>
<p>There are <strong>105</strong> significant proteins within all comparisons. The following table shows the results:</p>
<pre class="r"><code>results %&gt;% 
  select(starts_with(&quot;name&quot;) | starts_with(test)) %&gt;% 
  filter(across(ends_with(&quot;significant&quot;))) %&gt;% 
  arrange(across(ends_with(&quot;p.adj&quot;))) %&gt;%
  mutate(across(2:3, format, scientific = T, digits = 2)) %&gt;%
  dplyr::rename_with(~ tolower(gsub(paste0(test,&quot;_&quot;), &quot;&quot;, .x)), .cols = starts_with(test)) %&gt;% 
  mutate(significant = str_to_title(significant)) %&gt;%
  dplyr::rename(FC = ratio) %&gt;% 
  DT::datatable(options = list(
  columnDefs = list(list(className = &#39;dt-center&#39;, targets = 1:5)),
  pageLength = 10)) %&gt;%
  formatStyle(&#39;FC&#39;, backgroundColor = styleInterval(c(-1, 1), c(&#39;lightblue&#39;, &#39;white&#39;, &#39;tomato&#39;))) %&gt;% 
  formatStyle(&#39;significant&#39;, color = styleEqual(c(&quot;True&quot;, &quot;False&quot;), c(&#39;green&#39;, &#39;red&#39;)))</code></pre>
<div id="htmlwidget-c78f59155089cd61dbe3" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c78f59155089cd61dbe3">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105"],["SACS","APOBEC3C","ATP5J2-PTCD1","OSGEP","GMFB","NTMT1","C9orf142","PLCB3","PPP2CB","IGKV2-24","MAOA","STK4","DMXL1","RAB43","NACA","SETD3","HMGCL","MXRA7","SCP2","TTR","SURF4","UFM1","VAC14","MTR","ETFDH","ISYNA1","PFDN6","AFM","SMAD2","CDH2","PLIN4","TARS2","CUTA","MTPAP","PRELP","HBD","PACSIN2","BCLAF1","C14orf166","APPL2","SNTB1","STXBP2","MARS","IDH3G","LRPAP1","NHP2L1","TUBB2A","HIST1H1B","IGHV3-49","TPP2","RAB5C","G6PD","EWSR1","NDUFS8","STX1A","EIF3G","URGCP","HBB","ARHGEF12","SRI","FNDC3B","PDLIM1","MAPRE1","PSMB6","NCF2","RPL14","AGAP1","HNRNPM","COL6A2","DDAH2","PSMD6","PDCD6","UBE2C","CRP","SMARCC1","SERPINB6","GM2A","ZNF207","SUCLG1","SRSF9","ALDH1A1","HRG","IGLV3-10","USP4","SNRPB2","TPD52L2","PCCA","AP2M1","GPNMB","SULT1A4","COL6A1","SART3","CD276","DDB1","PTPRA","AHCYL1","ARPC3","KNG1","LIMS1","RBM38","OLFML3","PLCG1","DNAJC5","HBA1","ITPA"],["1.7e-09","4.6e-09","4.5e-08","4.4e-08","7.5e-08","1.8e-07","2.7e-07","4.0e-07","9.8e-07","1.4e-06","1.7e-06","1.8e-06","3.5e-06","3.6e-06","5.2e-06","7.4e-06","7.9e-06","8.5e-06","1.0e-05","1.1e-05","1.5e-05","1.5e-05","1.6e-05","2.0e-05","2.4e-05","2.6e-05","3.0e-05","3.3e-05","3.8e-05","4.0e-05","4.5e-05","5.7e-05","6.0e-05","7.2e-05","7.2e-05","7.9e-05","1.2e-04","1.3e-04","1.3e-04","1.4e-04","1.4e-04","1.4e-04","1.6e-04","2.1e-04","2.2e-04","2.2e-04","2.2e-04","2.5e-04","2.7e-04","2.7e-04","3.1e-04","3.3e-04","3.6e-04","4.0e-04","3.9e-04","4.2e-04","4.4e-04","4.8e-04","5.0e-04","5.4e-04","6.0e-04","6.0e-04","6.9e-04","7.3e-04","7.6e-04","8.4e-04","9.1e-04","9.4e-04","1.0e-03","1.0e-03","1.0e-03","1.1e-03","1.1e-03","1.1e-03","1.2e-03","1.3e-03","1.4e-03","1.4e-03","1.4e-03","1.5e-03","1.6e-03","1.6e-03","1.5e-03","1.6e-03","1.8e-03","1.8e-03","1.8e-03","1.9e-03","1.9e-03","2.0e-03","2.1e-03","2.0e-03","2.1e-03","2.2e-03","2.3e-03","2.4e-03","2.5e-03","2.6e-03","2.6e-03","2.6e-03","2.7e-03","2.7e-03","2.8e-03","2.8e-03","2.8e-03"],["3.1e-06","4.2e-06","2.0e-05","2.0e-05","2.8e-05","5.6e-05","7.0e-05","9.3e-05","2.0e-04","2.6e-04","2.8e-04","2.8e-04","4.7e-04","4.7e-04","6.4e-04","8.5e-04","8.5e-04","8.7e-04","9.7e-04","1.0e-03","1.2e-03","1.2e-03","1.2e-03","1.5e-03","1.8e-03","1.8e-03","2.1e-03","2.2e-03","2.4e-03","2.5e-03","2.7e-03","3.3e-03","3.4e-03","3.8e-03","3.8e-03","4.0e-03","6.2e-03","6.3e-03","6.3e-03","6.3e-03","6.3e-03","6.3e-03","7.0e-03","8.6e-03","8.7e-03","8.7e-03","8.7e-03","9.4e-03","9.9e-03","9.9e-03","1.1e-02","1.2e-02","1.2e-02","1.3e-02","1.3e-02","1.4e-02","1.4e-02","1.5e-02","1.6e-02","1.6e-02","1.8e-02","1.8e-02","2.0e-02","2.1e-02","2.2e-02","2.3e-02","2.5e-02","2.5e-02","2.7e-02","2.7e-02","2.7e-02","2.7e-02","2.7e-02","2.8e-02","2.9e-02","3.1e-02","3.2e-02","3.2e-02","3.3e-02","3.4e-02","3.5e-02","3.5e-02","3.5e-02","3.5e-02","3.8e-02","3.8e-02","3.9e-02","3.9e-02","4.0e-02","4.1e-02","4.1e-02","4.1e-02","4.2e-02","4.3e-02","4.4e-02","4.6e-02","4.8e-02","4.8e-02","4.8e-02","4.8e-02","4.9e-02","5.0e-02","5.0e-02","5.0e-02","5.0e-02"],["True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True","True"],[4.14,3.36,-2.93,3.75,2.7,3.44,-2.8,4.88,-3.36,3.31,-3.22,5.1,3.75,-7.53,-2.84,-2.75,-3.13,-4.09,-1.95,3.08,2.5,1.9,3.49,3.08,-3.18,-1.95,-2.64,2.43,-3.47,2.25,2.32,-4.37,2.61,-1.95,4.64,-4.49,-2.05,-1.97,-1.97,-1.91,-3.13,2.18,-2.66,-1.72,-2,-2.19,-2.92,-3.34,1.65,2.32,-2.96,-1.82,-2.75,-1.8,8.23,-2,1.55,-3.34,-3,-2.29,-2.29,-2.34,-2,-2.49,2.77,-2.52,-2.43,-1.28,2.62,1.9,-2.02,1.77,-1.71,-2.73,1.92,1.97,-2.62,-2.41,-1.95,-1.84,1.88,1.29,1.48,1.69,-1.92,-1.25,-2.29,1.47,2.86,-1.85,2.06,-1.55,1.73,-1.4,-1.96,-1.71,-3.69,1.11,1.78,-2.05,2.3,2.34,-1.67,-2.75,1.36]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>name<\/th>\n      <th>p.val<\/th>\n      <th>p.adj<\/th>\n      <th>significant<\/th>\n      <th>FC<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-center","targets":[1,2,3,4,5]},{"orderable":false,"targets":0}],"pageLength":10,"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'background-color':isNaN(parseFloat(value)) ? '' : value <= -1 ? \"lightblue\" : value <= 1 ? \"white\" : \"tomato\"});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'color':value == \"True\" ? \"green\" : value == \"False\" ? \"red\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
<div id="visualization-of-the-results" class="section level3 tabset tabset-fade tabset-pills">
<h3>Visualization of the results</h3>
<hr />
<p>The results from the analysis can be easily visualized by a number of plot functions.</p>
<div id="pca" class="section level4">
<h4>PCA</h4>
<pre class="r"><code>plot_pca(dep, x = 1, y = 2, n = 500, point_size = 4, indicate = &quot;condition&quot;)</code></pre>
<p><img src="figure/TME2_vs_TME1_population.Rmd/PCA-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="correlation-matrix" class="section level4">
<h4>Correlation matrix</h4>
<pre class="r"><code>plot_cor(dep, significant = TRUE, lower = 0, upper = 1, pal = &quot;Reds&quot;)</code></pre>
<p><img src="figure/TME2_vs_TME1_population.Rmd/plot_cor-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="volcano-plot" class="section level4">
<h4>Volcano plot</h4>
<pre class="r"><code>plot_volcano(dep, contrast=test, add_names=T,label_size=5,adjusted = F)</code></pre>
<p><img src="figure/TME2_vs_TME1_population.Rmd/all_volcano-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="cluster-heatmap" class="section level4">
<h4>Cluster Heatmap</h4>
<pre class="r"><code>plot_heatmap(dep, type = &quot;centered&quot;, kmeans = TRUE, k = 2, show_row_names = F, indicate = &quot;condition&quot;,col_limit = 5)</code></pre>
<p><img src="figure/TME2_vs_TME1_population.Rmd/heatmap-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="gene-ontology" class="section level3 tabset">
<h3>Gene Ontology</h3>
<div id="tme2" class="section level4">
<h4>TME2</h4>
<pre class="r"><code>Gname &lt;- results %&gt;% filter(significant &amp; get(paste0(test, &quot;_ratio&quot;)) &gt; 0) %&gt;% pull(name)

GO &lt;- gost(query = Gname, organism = &quot;hsapiens&quot;, sources = c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), ordered_query = T)

gostplot(GO, capped = TRUE, interactive = TRUE)</code></pre>
<div id="htmlwidget-17e1383e8f4ef8b1e0bf" style="width:960px;height:672px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-17e1383e8f4ef8b1e0bf">{"x":{"data":[{"x":[64.2746180187713,55.4462778402309,69.6488993741428,62.1358335723345,64.4307336717959,61.7221270918193,64.5244030636107,54.0802658762658,55.415054709626,54.9037759459705,58.4983388568616,58.0885352676721,59.4896732535677,61.2537801327456],"y":[2.92786238107424,2.60496193199295,2.33696196781597,2.33233342108304,2.33233342108304,2.30838389602403,2.29028957929912,2.28177163924923,2.28177163924923,2.20084253881387,2.11584179970733,2.11091723890851,1.49963060806549,1.47219370820948],"text":["GO:0062023 (421) <br> collagen-containing extracellular matrix <br> 1.181e-03","GO:0005773 (818) <br> vacuole <br> 2.483e-03","GO:1903561 (2263) <br> extracellular vesicle <br> 4.603e-03","GO:0043230 (2265) <br> extracellular organelle <br> 4.652e-03","GO:0065010 (2265) <br> extracellular membrane-bounded organelle <br> 4.652e-03","GO:0042383 (141) <br> sarcolemma <br> 4.916e-03","GO:0070062 (2178) <br> extracellular exosome <br> 5.125e-03","GO:0000323 (723) <br> lytic vacuole <br> 5.227e-03","GO:0005764 (723) <br> lysosome <br> 5.227e-03","GO:0005615 (3595) <br> extracellular space <br> 6.297e-03","GO:0031012 (562) <br> extracellular matrix <br> 7.659e-03","GO:0030312 (563) <br> external encapsulating structure <br> 7.746e-03","GO:0031982 (4058) <br> vesicle <br> 3.165e-02","GO:0036019 (25) <br> endolysosome <br> 3.371e-02"],"key":["GO:0062023","GO:0005773","GO:1903561","GO:0043230","GO:0065010","GO:0042383","GO:0070062","GO:0000323","GO:0005764","GO:0005615","GO:0031012","GO:0030312","GO:0031982","GO:0036019"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(16,150,24,1)","opacity":0.8,"size":[17.855572629986,19.4237645452457,21.5599427543828,21.561685742383,21.561685742383,14.7968354628896,21.4842405045742,19.144415938611,19.144415938611,22.4509020241687,18.558031099635,18.5622508728292,22.6771653543307,3.77952755905512],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(16,150,24,1)"}},"hoveron":"points","set":"SharedData80a1fcd6","name":"GO:CC","legendgroup":"GO:CC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[2,49.9206590071537],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[53.8226750487752,71.2217645783655],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[75.123780619987,193.971385215695],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[197.873401257316,200.003902016042],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(221,68,119,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[180],"y":[16.2],"text":"values above this threshold are capped","hovertext":"","textfont":{"size":7.55905511811024,"color":"rgba(190,190,190,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,210],"y":[16,16],"text":"","type":"scatter","mode":"lines","line":{"width":0.755905511811024,"color":"rgba(190,190,190,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.2835201328352,"r":6.6417600664176,"b":56.6372244321251,"l":61.0377750103778},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP","KEGG"],"tickvals":[25.9603295035768,62.5222198135703,134.547582917841,198.938651636679],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP","KEGG"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.6268161062682},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.132835201328352,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1,18.5],"tickmode":"array","ticktext":["0","2","4","6","8","10","12","14",">16"],"tickvals":[0,2,4,6,8,10,12,14,16],"categoryorder":"array","categoryarray":["0","2","4","6","8","10","12","14",">16"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"-log10(p-adj)","x":-0.0108706932337069,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis"},{"text":"query_1","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.9730178497302,"yanchor":1,"ysizemode":"pixel"}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"34305dd95184":{"x":{},"y":{},"text":{},"colour":{},"size":{},"alpha":{},"type":"scatter"},"343078e5452d":{"x":{},"xend":{},"y":{},"yend":{}},"34301463490":{"x":{},"xend":{},"y":{},"yend":{}},"34306db24f73":{"x":{},"xend":{},"y":{},"yend":{}},"34306f58478f":{"x":{},"xend":{},"y":{},"yend":{}},"3430b1b1e51":{"x":{},"y":{}},"34302a00518b":{"yintercept":{}}},"cur_data":"34305dd95184","visdat":{"34305dd95184":["function (y) ","x"],"343078e5452d":["function (y) ","x"],"34301463490":["function (y) ","x"],"34306db24f73":["function (y) ","x"],"34306f58478f":["function (y) ","x"],"3430b1b1e51":["function (y) ","x"],"34302a00518b":["function (y) ","x"]},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":false,"dynamic":false,"color":null,"selectize":false,"defaultValues":null,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedData80a1fcd6"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div id="table-tme2" class="section level4">
<h4>Table TME2</h4>
<pre class="r"><code>GO$result %&gt;% 
  as.data.frame() %&gt;% 
  select(starts_with(c(&quot;source&quot;, &quot;term_name&quot;, &quot;p_value&quot;, &quot;term_size&quot;, &quot;intersection_size&quot;))) %&gt;% 
  arrange(p_value) %&gt;%  
  mutate(p_value = format(p_value, scientific = T, digits = 2)) %&gt;% 
  DT::datatable(options = list(
  columnDefs = list(list(className = &#39;dt-center&#39;, targets = 1:6)),
  pageLength = 10)) %&gt;%
  formatStyle(&#39;source&#39;, backgroundColor = styleEqual(c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), c(&#39;orange&#39;, &#39;tomato&#39;, &#39;lightgreen&#39;, &#39;violet&#39;)))</code></pre>
<div id="htmlwidget-7f5b14b850b92b7d1598" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7f5b14b850b92b7d1598">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14"],["GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:CC"],[2679,417,4056,2131,2719,2025,2743,67,409,278,1199,1094,1453,1905],["collagen-containing extracellular matrix","vacuole","extracellular vesicle","extracellular organelle","extracellular membrane-bounded organelle","sarcolemma","extracellular exosome","lytic vacuole","lysosome","extracellular space","extracellular matrix","external encapsulating structure","vesicle","endolysosome"],["1.2e-03","2.5e-03","4.6e-03","4.7e-03","4.7e-03","4.9e-03","5.1e-03","5.2e-03","5.2e-03","6.3e-03","7.7e-03","7.7e-03","3.2e-02","3.4e-02"],[421,818,2263,2265,2265,141,2178,723,723,3595,562,563,4058,25],[7,10,15,15,15,3,7,9,9,12,7,7,19,2]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>source<\/th>\n      <th>source_order<\/th>\n      <th>term_name<\/th>\n      <th>p_value<\/th>\n      <th>term_size<\/th>\n      <th>intersection_size<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-center","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"pageLength":10,"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'background-color':value == \"GO:BP\" ? \"orange\" : value == \"GO:MF\" ? \"tomato\" : value == \"GO:CC\" ? \"lightgreen\" : value == \"KEGG\" ? \"violet\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
<div id="tme1" class="section level4">
<h4>TME1</h4>
<pre class="r"><code>Gname &lt;- results %&gt;% filter(significant &amp; get(paste0(test, &quot;_ratio&quot;)) &lt; 0) %&gt;% pull(name)

GO &lt;- gost(query = Gname, organism = &quot;hsapiens&quot;, sources = c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), ordered_query = T)

gostplot(GO, capped = TRUE, interactive = TRUE)</code></pre>
<div id="htmlwidget-f665a4417ef8a18c6182" style="width:960px;height:672px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-f665a4417ef8a18c6182">{"x":{"data":[{"x":[97.3152744419586,124.801977885245,114.19204792183,84.4889265972565,84.4850244530994,86.5180415589983,84.5084373180425],"y":[2.20335051340252,2.10902067674909,1.7802449155393,1.63543982001215,1.59448994515208,1.39433515159171,1.3071609052055],"text":["GO:0016071 (880) <br> mRNA metabolic process <br> 6.261e-03","GO:0044248 (2381) <br> cellular catabolic process <br> 7.780e-03","GO:0034622 (1150) <br> cellular protein-containing complex assembly <br> 1.659e-02","GO:0006397 (532) <br> mRNA processing <br> 2.315e-02","GO:0006396 (1010) <br> RNA processing <br> 2.544e-02","GO:0006979 (457) <br> response to oxidative stress <br> 4.033e-02","GO:0006402 (388) <br> mRNA catabolic process <br> 4.930e-02"],"key":["GO:0016071","GO:0044248","GO:0034622","GO:0006397","GO:0006396","GO:0006979","GO:0006402"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,153,0,1)","opacity":0.8,"size":[19.1861535689619,20.6038659170467,19.5798048876813,18.4171454758136,19.3900827643146,18.1768796144593,17.9135182170874],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(255,153,0,1)"}},"hoveron":"points","set":"SharedData2a0dd1ba","name":"GO:BP","legendgroup":"GO:BP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[55.3213853178112,64.5244030636107,69.6488993741428,62.1358335723345,64.4307336717959,59.3413633831944,55.6648397544653],"y":[3.47332535350041,2.68174313487827,2.46763928355909,2.46272590780059,2.46272590780059,1.90015367440094,1.74891036133067],"text":["GO:0005737 (11951) <br> cytoplasm <br> 3.363e-04","GO:0070062 (2178) <br> extracellular exosome <br> 2.081e-03","GO:1903561 (2263) <br> extracellular vesicle <br> 3.407e-03","GO:0043230 (2265) <br> extracellular organelle <br> 3.446e-03","GO:0065010 (2265) <br> extracellular membrane-bounded organelle <br> 3.446e-03","GO:0031838 (11) <br> haptoglobin-hemoglobin complex <br> 1.258e-02","GO:0005833 (13) <br> hemoglobin complex <br> 1.783e-02"],"key":["GO:0005737","GO:0070062","GO:1903561","GO:0043230","GO:0065010","GO:0031838","GO:0005833"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(16,150,24,1)","opacity":0.8,"size":[22.6771653543307,20.4818441334493,20.5343751667814,20.5355853505457,20.5355853505457,9.79584921339076,10.4026058367767],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(16,150,24,1)"}},"hoveron":"points","set":"SharedData2a0dd1ba","name":"GO:CC","legendgroup":"GO:CC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[22.057995708206,3.70531986857705,48.6250841870906,42.2096474275009,20.9692445792976,22.0540933744106,21.2111892746106],"y":[2.30321951668482,2.18319057390671,2.16445426495473,2.02579581240066,1.74790928332279,1.65124731769981,1.35793128892008],"text":["GO:0031721 (5) <br> hemoglobin alpha binding <br> 4.975e-03","GO:0003723 (1951) <br> RNA binding <br> 6.559e-03","GO:1901363 (6210) <br> heterocyclic compound binding <br> 6.848e-03","GO:0097159 (6291) <br> organic cyclic compound binding <br> 9.423e-03","GO:0030492 (9) <br> hemoglobin binding <br> 1.787e-02","GO:0031720 (10) <br> haptoglobin binding <br> 2.232e-02","GO:0030621 (6) <br> U4 snRNA binding <br> 4.386e-02"],"key":["GO:0031721","GO:0003723","GO:1901363","GO:0097159","GO:0030492","GO:0031720","GO:0030621"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(220,57,18,1)","opacity":0.8,"size":[3.77952755905512,20.3298921737568,21.8645259375893,21.8809664138151,8.9741239064159,9.42050901214848,6.67261094673741],"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(220,57,18,1)"}},"hoveron":"points","set":"SharedData2a0dd1ba","name":"GO:MF","legendgroup":"GO:MF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[2,49.9206590071537],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[53.8226750487752,71.2217645783655],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[75.123780619987,193.971385215695],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[197.873401257316,200.003902016042],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.3385826771654,"color":"rgba(221,68,119,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[180],"y":[16.2],"text":"values above this threshold are capped","hovertext":"","textfont":{"size":7.55905511811024,"color":"rgba(190,190,190,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,210],"y":[16,16],"text":"","type":"scatter","mode":"lines","line":{"width":0.755905511811024,"color":"rgba(190,190,190,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.2835201328352,"r":6.6417600664176,"b":56.6372244321251,"l":61.0377750103778},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP","KEGG"],"tickvals":[25.9603295035768,62.5222198135703,134.547582917841,198.938651636679],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP","KEGG"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.65296803652968,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.6268161062682},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.132835201328352,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1,18.5],"tickmode":"array","ticktext":["0","2","4","6","8","10","12","14",">16"],"tickvals":[0,2,4,6,8,10,12,14,16],"categoryorder":"array","categoryarray":["0","2","4","6","8","10","12","14",">16"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"-log10(p-adj)","x":-0.0108706932337069,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis"},{"text":"query_1","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.2835201328352},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.9730178497302,"yanchor":1,"ysizemode":"pixel"}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"343019d7315b":{"x":{},"y":{},"text":{},"colour":{},"size":{},"alpha":{},"type":"scatter"},"34302a883345":{"x":{},"xend":{},"y":{},"yend":{}},"34304fcc58c":{"x":{},"xend":{},"y":{},"yend":{}},"34306f534613":{"x":{},"xend":{},"y":{},"yend":{}},"3430688d6dc4":{"x":{},"xend":{},"y":{},"yend":{}},"3430656227d0":{"x":{},"y":{}},"343045905c0e":{"yintercept":{}}},"cur_data":"343019d7315b","visdat":{"343019d7315b":["function (y) ","x"],"34302a883345":["function (y) ","x"],"34304fcc58c":["function (y) ","x"],"34306f534613":["function (y) ","x"],"3430688d6dc4":["function (y) ","x"],"3430656227d0":["function (y) ","x"],"343045905c0e":["function (y) ","x"]},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":false,"dynamic":false,"color":null,"selectize":false,"defaultValues":null,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedData2a0dd1ba"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div id="table-tme1" class="section level4">
<h4>Table TME1</h4>
<pre class="r"><code>GO$result %&gt;% 
  as.data.frame() %&gt;% 
  select(starts_with(c(&quot;source&quot;, &quot;term_name&quot;, &quot;p_value&quot;, &quot;term_size&quot;, &quot;intersection_size&quot;))) %&gt;% 
  arrange(p_value) %&gt;%  
  mutate(p_value = format(p_value, scientific = T, digits = 2)) %&gt;% 
  DT::datatable(options = list(
  columnDefs = list(list(className = &#39;dt-center&#39;, targets = 1:6)),
  pageLength = 10)) %&gt;%
  formatStyle(&#39;source&#39;, backgroundColor = styleEqual(c(&quot;GO:BP&quot;, &quot;GO:MF&quot;, &quot;GO:CC&quot;, &quot;KEGG&quot;), c(&#39;orange&#39;, &#39;tomato&#39;, &#39;lightgreen&#39;, &#39;violet&#39;)))</code></pre>
<div id="htmlwidget-ba3b0b0989c6b20e8f5b" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ba3b0b0989c6b20e8f5b">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21"],["GO:CC","GO:CC","GO:CC","GO:CC","GO:CC","GO:MF","GO:BP","GO:MF","GO:MF","GO:BP","GO:MF","GO:CC","GO:BP","GO:CC","GO:MF","GO:MF","GO:BP","GO:BP","GO:BP","GO:MF","GO:BP"],[385,2743,4056,2131,2719,5141,5688,438,11949,12732,10305,1415,10013,473,4862,5140,2401,2400,2921,4924,2406],["cytoplasm","extracellular exosome","extracellular vesicle","extracellular organelle","extracellular membrane-bounded organelle","hemoglobin alpha binding","mRNA metabolic process","RNA binding","heterocyclic compound binding","cellular catabolic process","organic cyclic compound binding","haptoglobin-hemoglobin complex","cellular protein-containing complex assembly","hemoglobin complex","hemoglobin binding","haptoglobin binding","mRNA processing","RNA processing","response to oxidative stress","U4 snRNA binding","mRNA catabolic process"],["3.4e-04","2.1e-03","3.4e-03","3.4e-03","3.4e-03","5.0e-03","6.3e-03","6.6e-03","6.8e-03","7.8e-03","9.4e-03","1.3e-02","1.7e-02","1.8e-02","1.8e-02","2.2e-02","2.3e-02","2.5e-02","4.0e-02","4.4e-02","4.9e-02"],[11951,2178,2263,2265,2265,5,880,1951,6210,2381,6291,11,1150,13,9,10,532,1010,457,6,388],[52,16,16,16,16,2,12,18,35,18,35,2,14,2,2,2,9,12,7,2,7]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>source<\/th>\n      <th>source_order<\/th>\n      <th>term_name<\/th>\n      <th>p_value<\/th>\n      <th>term_size<\/th>\n      <th>intersection_size<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-center","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0}],"pageLength":10,"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'background-color':value == \"GO:BP\" ? \"orange\" : value == \"GO:MF\" ? \"tomato\" : value == \"GO:CC\" ? \"lightgreen\" : value == \"KEGG\" ? \"violet\" : null});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
</div>
<div id="treemap-tme1" class="section level4">
<h4>Treemap TME1</h4>
<pre class="r"><code>simMatrix &lt;- GO$result %&gt;% 
  as.data.frame() %&gt;%
  filter(source == &quot;GO:BP&quot;) %&gt;% 
  pull(term_id) %&gt;% 
  calculateSimMatrix(orgdb=&quot;org.Hs.eg.db&quot;, ont=&quot;BP&quot;, method=&quot;Rel&quot;)

scores &lt;- setNames(-log10(GO$result$p_value), GO$result$term_id)

reducedTerms &lt;- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb=&quot;org.Hs.eg.db&quot;)

treemapPlot(reducedTerms)</code></pre>
<p><img src="figure/TME2_vs_TME1_population.Rmd/treemap_1-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="scatterplot-tme1" class="section level4">
<h4>ScatterPlot TME1</h4>
<pre class="r"><code>scatterPlot(simMatrix, reducedTerms, size = &quot;score&quot;, labelSize = 5)</code></pre>
<p><img src="figure/TME2_vs_TME1_population.Rmd/scatter_2-1.png" width="1248" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="section" class="section level2 unnumbered">
<h2></h2>
<pre class="r"><code>a &lt;- get_df_wide(dep) %&gt;% 
  select(c(name, Protein.IDs, Protein.names, starts_with(conditions), -ends_with(c(&quot;CI.R&quot;, &quot;CI.L&quot;)))) %&gt;% 
  dplyr::rename_with(., ~ gsub(pattern = paste0(test, &quot;_&quot;), replacement = &quot;&quot;, .), starts_with(test)) %&gt;% 
  dplyr::rename_with(., ~ gsub(pattern = &quot;^&quot;, replacement = &quot;LFQ_intensity_&quot;, .), starts_with(conditions)) %&gt;%
  dplyr::rename(FC = diff) %&gt;% 
  mutate(significant = if_else(significant, &quot;+&quot;, &quot;&quot;)) %&gt;% 
  arrange(desc(significant), p.val) %&gt;% 
  mutate(across(c(p.adj, p.val), format, scientific = T, digits = 2)) %&gt;% 
  mutate(across(starts_with(c(&quot;LFQ&quot;, &quot;FC&quot;)), .fns = ~ round(.x, digits = 2))) %&gt;% 
  relocate(significant) %&gt;%
  relocate(starts_with(&quot;LFQ&quot;), .after = p.val)
  

addWorksheet(excel, sheetName = test, gridLines = F)

writeDataTable(excel, sheet = test, x = a, keepNA = T, na.string = &quot;NaN&quot;)

n_row &lt;- a %&gt;% nrow() + 1

n_col &lt;- a %&gt;% ncol()

setColWidths(excel, sheet = test, cols = 1:n_col, widths = 21)

addStyle(excel, sheet = test, style = header_style, rows = 1, cols = 1:n_col, gridExpand = T)

addStyle(excel, sheet = test, style = body_style, rows = 2:n_row, cols = 1:n_col, gridExpand = T)</code></pre>
<pre class="r"><code>a &lt;- unique_pair(data_filt, conditions = conditions, table = T)</code></pre>
<pre><code>Joining, by = &quot;name&quot;</code></pre>
<pre class="r"><code>unique_name &lt;- paste0(&quot;Unique_&quot;, test)

addWorksheet(excel, sheetName = unique_name, gridLines = F)

writeDataTable(excel, sheet = unique_name, x = a, keepNA = T, na.string = &quot;NaN&quot;)

n_row &lt;- a %&gt;% nrow() + 1

n_col &lt;- a %&gt;% ncol()

setColWidths(excel, sheet = unique_name, cols = 1:n_col, widths = 21)

addStyle(excel, sheet = unique_name, style = header_style, rows = 1, cols = 1:n_col, gridExpand = T)

addStyle(excel, sheet = unique_name, style = body_style, rows = 2:n_row, cols = 1:n_col, gridExpand = T)</code></pre>
<pre class="r"><code>saveWorkbook(excel, here::here(&quot;output/results_TME2_vs_TME1_population.xlsx&quot;), overwrite = T)</code></pre>
<hr />
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=Italian_Italy.1252  LC_CTYPE=Italian_Italy.1252   
[3] LC_MONETARY=Italian_Italy.1252 LC_NUMERIC=C                  
[5] LC_TIME=Italian_Italy.1252    

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] openxlsx_4.2.4              rrvgo_1.2.0                
 [3] org.Hs.eg.db_3.12.0         AnnotationDbi_1.52.0       
 [5] gprofiler2_0.2.0            datapasta_3.1.0            
 [7] here_1.0.1                  patchwork_1.1.1            
 [9] stringr_1.4.0               DT_0.18                    
[11] enrichR_3.0                 ggplot2_3.3.5              
[13] tibble_3.1.2                preprocessCore_1.52.1      
[15] SummarizedExperiment_1.20.0 Biobase_2.50.0             
[17] GenomicRanges_1.42.0        GenomeInfoDb_1.26.2        
[19] IRanges_2.24.1              S4Vectors_0.28.1           
[21] BiocGenerics_0.36.0         MatrixGenerics_1.2.1       
[23] matrixStats_0.59.0          DEP_1.12.0                 
[25] tidyr_1.1.3                 dplyr_1.0.7                
[27] MBQN_2.2.0                  workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] circlize_0.4.13        BiocFileCache_1.14.0   igraph_1.2.6          
  [4] plyr_1.8.6             gmm_1.6-6              lazyeval_0.2.2        
  [7] shinydashboard_0.7.1   crosstalk_1.1.1        BiocParallel_1.24.1   
 [10] gridBase_0.4-7         digest_0.6.27          GOSemSim_2.16.1       
 [13] foreach_1.5.1          htmltools_0.5.1.1      GO.db_3.12.1          
 [16] fansi_0.5.0            magrittr_2.0.1         memoise_2.0.0         
 [19] tm_0.7-8               cluster_2.1.0          doParallel_1.0.16     
 [22] limma_3.46.0           ComplexHeatmap_2.6.2   readr_1.4.0           
 [25] wordcloud_2.6          imputeLCMD_2.0         sandwich_3.0-1        
 [28] colorspace_2.0-2       treemap_2.4-2          ggrepel_0.9.1         
 [31] blob_1.2.1             rappdirs_0.3.3         xfun_0.24             
 [34] crayon_1.4.1           RCurl_1.98-1.3         jsonlite_1.7.2        
 [37] impute_1.64.0          zoo_1.8-9              iterators_1.0.13      
 [40] glue_1.4.2             gtable_0.3.0           zlibbioc_1.36.0       
 [43] XVector_0.30.0         GetoptLong_1.0.5       DelayedArray_0.16.2   
 [46] shape_1.4.6            scales_1.1.1           pheatmap_1.0.12       
 [49] vsn_3.58.0             mvtnorm_1.1-2          DBI_1.1.1             
 [52] Rcpp_1.0.6             mzR_2.24.1             viridisLite_0.4.0     
 [55] xtable_1.8-4           clue_0.3-59            bit_4.0.4             
 [58] htmlwidgets_1.5.3      httr_1.4.2             RColorBrewer_1.1-2    
 [61] ellipsis_0.3.2         farver_2.1.0           pkgconfig_2.0.3       
 [64] XML_3.99-0.6           sass_0.4.0             dbplyr_2.1.1          
 [67] utf8_1.2.1             labeling_0.4.2         tidyselect_1.1.1      
 [70] rlang_0.4.11           later_1.2.0            munsell_0.5.0         
 [73] tools_4.0.3            cachem_1.0.5           generics_0.1.0        
 [76] RSQLite_2.2.7          evaluate_0.14          fastmap_1.1.0         
 [79] mzID_1.28.0            yaml_2.2.1             knitr_1.33            
 [82] bit64_4.0.5            fs_1.5.0               zip_2.2.0             
 [85] purrr_0.3.4            ncdf4_1.17             whisker_0.4           
 [88] mime_0.11              slam_0.1-48            xml2_1.3.2            
 [91] compiler_4.0.3         plotly_4.9.4.1         curl_4.3.2            
 [94] png_0.1-7              affyio_1.60.0          bslib_0.2.5.1         
 [97] stringi_1.6.2          highr_0.9              MSnbase_2.15.7        
[100] lattice_0.20-41        ProtGenerics_1.22.0    Matrix_1.2-18         
[103] tmvtnorm_1.4-10        vctrs_0.3.8            pillar_1.6.1          
[106] norm_1.0-9.5           lifecycle_1.0.0        BiocManager_1.30.16   
[109] jquerylib_0.1.4        MALDIquant_1.19.3      GlobalOptions_0.1.2   
[112] data.table_1.14.0      bitops_1.0-7           httpuv_1.6.1          
[115] R6_2.5.0               pcaMethods_1.82.0      affy_1.68.0           
[118] promises_1.2.0.1       codetools_0.2-16       MASS_7.3-53           
[121] assertthat_0.2.1       rprojroot_2.0.2        rjson_0.2.20          
[124] withr_2.4.2            GenomeInfoDbData_1.2.4 hms_1.1.0             
[127] grid_4.0.3             rmarkdown_2.9          Cairo_1.5-12.2        
[130] git2r_0.28.0           NLP_0.2-1              shiny_1.6.0           </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
