---
title: "T1 vs TME1 Sub-populations anaysis"
author: "Fabio Bedin | MS-Unit"
output: html_document
---

# Ovarian cancer dataset

***

```{r librerie, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(fig.align = "center")
library("MBQN")
library("dplyr")
library("tidyr")
library("DEP")
library("SummarizedExperiment")
library("preprocessCore")
library("tibble")
library("ggplot2")
library("enrichR")
library("DT")
library("stringr")
library("patchwork")
library("openxlsx")
```

```{r Custom Functions}

source(here::here("code/custom_functions.R"))

```

```{r excel_define_wb}
header_style <- createStyle(
  fontSize = 12,
  fontColour = "#0f0f0f",
  fgFill = "#faf2ca",
  halign = "center",
  border = "TopBottomLeftRight"
)

body_style <- createStyle(
  halign = "center",
  border = "TopBottomLeftRight"
)

excel <- createWorkbook()
```

## Introduction

***

For this analysis i will focus in all the possible combination between **T1** and **TME1** populations, where different sub-populations can be identify as sensivie **(S)**, intermediate sensitive **(IS)** and resistant **(R)**.

```{r input_Data}
data <- read.csv(here::here("data/proteinGroups.txt"), header = TRUE,stringsAsFactors = FALSE, sep = "\t")

expdesign <- read.table(here::here("data/expdesign_SI_S_R.tsv"), header = T, stringsAsFactors = F)

anova_table <- read.table(here::here("data/anova_table_t1_tme1_filt_1500.txt"), header = T, stringsAsFactors = F, sep = "\t")

conditions<-c("T1_IS","T1_S","T1_R","TME1_IS","TME1_S","TME1_R")

expdesign <- subset(expdesign, condition %in% conditions)
```

## Data wrangling

***

First, we need to wrangling the original dataset. From the **proteinGroups** obtained by **MaxQuant** software, I remove proteins that are marked form the software as *potential contaminant*, *only identify by site* and *reverse*.

```{r Data wrangling}
data <- data[data$Reverse != "+" & data$Potential.contaminant != "+" & data$Only.identified.by.site != "+",]

data_unique <- make_unique(data, "Gene.names", "Protein.IDs", delim = ";")

columns<-match(paste("LFQ.intensity.",expdesign$label,sep=""),colnames(data_unique))

data_se <- make_se(data_unique, columns, expdesign)
```

To understand the structure of the dataset and define a strategy, I use different quality control polts:

### Quality control Plots {.tabset .tabset-fade .tabset-pills}

All this plot can also hepl to decide the strategy to filter the missing data.

#### Frequency plot

```{r plot_freq, fig.height = 7, fig.width = 12}
plot_frequency(data_se)
```

#### Number of protein per sample

```{r plot numbers, fig.height = 7, fig.width = 12}
plot_numbers(data_se)
```

#### Distibution of each sample

```{r plot_norm, fig.height = 10, fig.width = 10}
plot_normalization(data_se)
```

## The problem of missing data

***

In the original dataset, the percentage of missing data is **`r round((sum(is.na(assay(data_se)))/prod(dim(assay(data_se))))*100, digits = 2)`%** and can be visualized with the following heatmap.

```{r plot_mv, message=FALSE, fig.height = 7, fig.width = 12}
plot_missval(data_se)
```

## Filter missing data

***

To reduce the percentage of missing data I have decided to filter samples with less then **1500** identify proteins.
Afrer that, I also use a second strong filter to remove all the proteins that are not identify in all replicates in at least 1 groups.

```{r data_filt}
data_filt <- filter_bad_samples(data_se, thr = 1500)

data_filt <- filter_missval(data_filt, thr = 0)
```


These plot show the effect of the filters. Now we reduce the percetage of missing data in the dataset down to: **`r round((sum(is.na(assay(data_filt)))/prod(dim(assay(data_filt))))*100, digits = 2)`%**.

### Plots with filters {.tabset .tabset-fade .tabset-pills}

#### protein coverage

```{r plot_coverage}
plot_coverage(data_filt)
```

#### Number of protein per sample

```{r plot_numbers_filt}
plot_numbers(data_filt)
```

#### Distibution

```{r plot_norm_filt, fig.height = 8, fig.width = 8}
plot_normalization(data_filt)
```

#### Plot missing data

```{r plot_mv_filt, message=FALSE}
plot_missval(data_filt)
```

## Imputation

***

An idealized version of a label-free discovery mass spectrometry proteomics experiment would provide absolute abundance measurements for a whole proteome, across varying conditions. Unfortunately, this ideal is not realized. Measurements are made on peptides requiring an inferential step to obtain protein level estimates. The inference is complicated by experimental factors that necessitate relative abundance estimation and result in widespread **non-ignorable missing data**. Relative abundance on the log scale takes the form of parameter contrasts. In a complete-case analysis, contrast estimates may be biased by missing data and a substantial amount of useful information will often go unused.

To avoid this problem, we impute missing data using random draws from manually defined left-shifted Gaussian distribution.


```{r mixed_imputation, message=FALSE, warning=FALSE, results='hide'}
#set.seed(20210520)
data_imp_man <- impute(data_filt, fun = "man", shift = 1.8, scale = 0.3)
```

This plot show the effect of imputation, as expected a new population of low intensity values as emerged.

```{r plot_imputation}
plot_imputation(data_filt, data_imp_man)
```

## Differential enrichment analysis

***

Protein-wise linear models combined with empirical Bayes statistics are used for the differential enrichment analysis (or differential expression analysis). I test all possible combination between T1 S, IS and R vs T2 S, IS, and R.

```{r test_diff}
data_diff <- test_diff(data_imp_man, type = "all")
```

I define sifnificant protein with **FDR = 0.05** and **FC = 1**

```{r Diff_prot}
dep <- add_rejections(data_diff, alpha = 0.05, lfc = 1)

results<-get_results(dep)
```

```{r excel_table_res}
a <- get_df_wide(dep) %>% 
  select(c(name, Protein.IDs, Protein.names, starts_with(conditions), -ends_with(c("CI.R", "CI.L")))) %>% 
  relocate(contains("_vs_"), .after = where(is.character)) %>% 
  dplyr::rename_with(., ~ gsub(pattern = "^", replacement = "LFQ_intensity_", .), !contains(c("_vs_", "name", "Protein."))) %>%
  dplyr::rename_with(., ~ gsub(pattern = "....$", replacement = "FC", .), ends_with("_diff")) %>%
  relocate(contains("_significant"), .before = starts_with("name")) %>% 
  mutate(across(ends_with("_significant"), ~ if_else(.x, "+", ""))) %>% 
  mutate(across(contains(c("_p.adj", "_p.val")), format, scientific = T, digits = 2)) %>% 
  mutate(across(contains(c("LFQ", "FC")), .fns = ~ round(.x, digits = 2))) 

test <- "Multiple_comparisons_results"

addWorksheet(excel, sheetName = test, gridLines = F)

writeDataTable(excel, sheet = test, x = a, keepNA = T, na.string = "NaN")

n_row <- a %>% nrow() + 1

n_col <- a %>% ncol()

setColWidths(excel, sheet = test, cols = 1:n_col, widths = 21)

addStyle(excel, sheet = test, style = header_style, rows = 1, cols = 1:n_col, gridExpand = T)

addStyle(excel, sheet = test, style = body_style, rows = 2:n_row, cols = 1:n_col, gridExpand = T)
```

There are **`r results %>% filter(significant) %>% nrow()`** significant proteins within all comparisons.
The following table shows the results:

```{r significant_table}
results %>% 
  filter(significant) %>%
  select(-significant, -ends_with("centered")) %>% 
  pivot_longer(
   cols = ends_with(c("p.val", "p.adj", "significant", "ratio")),
   names_to = c("comparison", ".value"),
   names_pattern = "(.*_.*_.*_.*_.*)_(.*)") %>% 
  arrange(name, p.adj) %>% 
  relocate(comparison) %>% 
  mutate(across(ends_with(c("p.val", "p.adj")), format, scientific = T, digits = 2)) %>% 
  mutate(significant = str_to_title(significant)) %>% 
  dplyr::rename(FC = ratio) %>%
  DT::datatable(options = list(
  columnDefs = list(list(className = 'dt-center', targets = 1:7)),
  pageLength = 15)) %>% 
  formatStyle('FC', backgroundColor = styleInterval(c(-1, 1), c('lightblue', 'white', 'tomato'))) %>% 
  formatStyle('significant', color = styleEqual(c("True", "False"), c('green', 'red')))
```


### Visualization of the results {.tabset .tabset-fade .tabset-pills}

***

The results from the analysis can be easily visualized by a number of plot functions.

#### PCA

```{r PCA, message=FALSE, warning=FALSE, fig.height = 8, fig.width = 8}
plot_pca(dep, x = 1, y = 2, n = 500, point_size = 4, indicate = "condition")
```



#### Correlation matrix

```{r plot_cor, fig.height = 8}
plot_cor(dep, significant = TRUE, lower = 0, upper = 1, pal = "Reds")
```


#### Overview all volcano plot

```{r all_volcano, warning=FALSE, fig.height = 8, fig.width = 8}
cont <- "T1_IS_vs_T1_R"
p1 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T1_IS_vs_T1_S"
p2 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T1_IS_vs_TME1_IS"
p3 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T1_IS_vs_TME1_R"
p4 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T1_IS_vs_TME1_S"
p5 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T1_R_vs_T1_S"
p6 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T1_R_vs_TME1_IS"
p7 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T1_R_vs_TME1_R"
p8 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T1_R_vs_TME1_S"
p9 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T1_S_vs_TME1_IS"
p10 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T1_S_vs_TME1_R"
p11 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T1_S_vs_TME1_S"
p12 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "TME1_IS_vs_TME1_R"
p13 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "TME1_IS_vs_TME1_S"
p14 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "TME1_R_vs_TME1_S"
p15 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
```

```{r plot_v_all, fig.height = 12, fig.width = 18}
(p1 | p2 | p3 | p4 | p5) /
  (p6 | p7 | p8 | p9 | p10) /
  (p11 | p12 | p13 | p14 | p15) 
```

#### Cluster Heatmap

```{r heatmap, fig.height = 15, fig.width = 10}
plot_heatmap(dep, type = "centered", kmeans = TRUE, k = 7, show_row_names = T, indicate = "condition",col_limit = 5)
```

```{r}
#cluster_table <- plot_heatmap(dep, type = "centered", kmeans = TRUE, k = 8, show_row_names = T, indicate = "condition",col_limit = 5, plot = F) %>% dplyr::rename(cluster = k) 

#write.table(cluster_table ,file = "cluster_table.txt", sep="\t", quote=FALSE, row.names=FALSE, col.names = T)
```


### Volcano Plots T1_R vs TME1_R {.tabset}

#### Significant

```{r V1,warning=FALSE, fig.height = 8, fig.width = 8}
cont <- "T1_R_vs_TME1_R"
plot_volcano(dep, contrast=cont, add_names=T,label_size=4,adjusted = F)
```

```{r T1}
results %>%  select(starts_with("name") | starts_with(cont)) %>% filter(across(ends_with("significant"))) %>% arrange(across(ends_with("p.adj"))) %>% mutate(across(2:3, format, scientific = T, digits = 2)) %>% dplyr::rename_with(~ tolower(gsub(paste0(cont,"_"), "", .x)), .cols = starts_with(cont)) %>%  DT::datatable()
```

#### Unique

```{r V2,warning=FALSE, fig.height = 8, fig.width = 8}
unique_vec <- c("AARS2","ABCA8","BAZ1B","CCDC58","COMT","FAF1","FUT8","MBLAC2","MCCC1","NDUFA6","NDUFV2","PABPN1","PLIN1","POLR2E","PPIF","PTGES2","PTGES3","PTGS1","QTRT1","RPL18","RPL37A","SNRPD3","TIMM10","TPD52","XRCC1","CAV1","EFEMP1","PON1","PRR12","SNX7","STX18","ATAD1","C8orf82","CIRH1A","DDX56","FAM177A1","IRGQ","KATNAL2","LAMTOR1","MBNL1","MCM5","NDUFA6","NFKB1","NUCKS1","OPTN","POLR2L","PON2","RPL29","SEC61B","SF3B4","SLC9A3R2","SUPT6H","TRA2A","C16orf46","GLA","NRAS","PCBD1","PLP2","BNIP1","COX7C","GALNT5","OLFML3","S100A14","THG1L")
plot_volcano_2(dep, contrast=cont, add_names=T,label_size=4,adjusted = F)
```

```{r T2}
results %>%  select(starts_with("name") | starts_with(cont)) %>% filter(name %in% unique_vec) %>% arrange(across(ends_with("p.adj"))) %>% mutate(across(2:3, format, scientific = T, digits = 2)) %>% dplyr::rename_with(~ tolower(gsub(paste0(cont,"_"), "", .x)), .cols = starts_with(cont)) %>% dplyr::rename(FC = ratio) %>%
  DT::datatable(options = list(
  columnDefs = list(list(className = 'dt-center', targets = 1:5)),
  pageLength = 10)) %>% 
  formatStyle('FC', backgroundColor = styleInterval(c(-1, 1), c('lightblue', 'white', 'tomato'))) %>% 
  formatStyle('significant', color = styleEqual(c("True", "False"), c('green', 'red')))
```

### Volcano Plots T1_S vs TME1_S {.tabset}

#### Significant

```{r V5,warning=FALSE, fig.height = 8, fig.width = 8}
cont <- "T1_S_vs_TME1_S"
plot_volcano(dep, contrast=cont, add_names=T,label_size=4,adjusted = F)
```

```{r T5}
results %>%  select(starts_with("name") | starts_with(cont)) %>% filter(across(ends_with("significant"))) %>% arrange(across(ends_with("p.adj"))) %>% mutate(across(2:3, format, scientific = T, digits = 2)) %>% dplyr::rename_with(~ tolower(gsub(paste0(cont,"_"), "", .x)), .cols = starts_with(cont)) %>%  DT::datatable()
```

#### Unique

```{r V6,warning=FALSE, fig.height = 8, fig.width = 8}
unique_vec <- unique_vec <- c("AARS2","ABCA8","BAZ1B","CCDC58","COMT","FAF1","FUT8","MBLAC2","MCCC1","NDUFA6","NDUFV2","PABPN1","PLIN1","POLR2E","PPIF","PTGES2","PTGES3","PTGS1","QTRT1","RPL18","RPL37A","SNRPD3","TIMM10","TPD52","XRCC1","CAV1","EFEMP1","PON1","PRR12","SNX7","STX18","CCDC58","CD47","CPSF6","CST1","CSTF3","FKBP8","FNTA","HK2","MBLAC2","MUC16","PCDHGA10","PLD3","PLIN1","PPIF","PTGES3","QTRT1","RPL37A","S100A1","SLC34A2","TRIM33","TUBGCP2","U2SURP","UTS2","WISP2","XPC")
plot_volcano_2(dep, contrast=cont, add_names=T,label_size=4,adjusted = F)
```

```{r T6}
results %>%  select(starts_with("name") | starts_with(cont)) %>% filter(name %in% unique_vec) %>% arrange(across(ends_with("p.adj"))) %>% mutate(across(2:3, format, scientific = T, digits = 2)) %>% dplyr::rename_with(~ tolower(gsub(paste0(cont,"_"), "", .x)), .cols = starts_with(cont)) %>% dplyr::rename(FC = ratio) %>%
  DT::datatable(options = list(
  columnDefs = list(list(className = 'dt-center', targets = 1:5)),
  pageLength = 10)) %>% 
  formatStyle('FC', backgroundColor = styleInterval(c(-1, 1), c('lightblue', 'white', 'tomato'))) %>% 
  formatStyle('significant', color = styleEqual(c("True", "False"), c('green', 'red')))
```

### Volcano Plots T1_IS vs TME1_IS {.tabset}

#### Significant

```{r V3,warning=FALSE, fig.height = 8, fig.width = 8}
cont <- "T1_IS_vs_TME1_IS"
plot_volcano(dep, contrast=cont, add_names=T,label_size=4,adjusted = F)
```

```{r T3}
results %>%  select(starts_with("name") | starts_with(cont)) %>% filter(across(ends_with("significant"))) %>% arrange(across(ends_with("p.adj"))) %>% mutate(across(2:3, format, scientific = T, digits = 2)) %>% dplyr::rename_with(~ tolower(gsub(paste0(cont,"_"), "", .x)), .cols = starts_with(cont)) %>%  DT::datatable()
```

#### Unique

```{r V4,warning=FALSE, fig.height = 8, fig.width = 8}
unique_vec <- c("AARS2","ABCA8","BAZ1B","CCDC58","COMT","FAF1","FUT8","MBLAC2","MCCC1","NDUFA6","NDUFV2","PABPN1","PLIN1","POLR2E","PPIF","PTGES2","PTGES3","PTGS1","QTRT1","RPL18","RPL37A","SNRPD3","TIMM10","TPD52","XRCC1","CAV1","EFEMP1","PON1","PRR12","SNX7","STX18","ACAD8","AGK","BRD4","CACNA2D1","CAPS","CCDC58","CIRH1A","IRGQ","METTL14","MRPS24","NOL6","PPIF","PRKCD","PRKCI","PTGES3","RAB18","RPL37A","SNCG","THY1","TM9SF3","UQCC2","ZNF185","IGHV3-15")
plot_volcano_2(dep, contrast=cont, add_names=T,label_size=4,adjusted = F)
```

```{r T4}
results %>%  select(starts_with("name") | starts_with(cont)) %>% filter(name %in% unique_vec) %>% arrange(across(ends_with("p.adj"))) %>% mutate(across(2:3, format, scientific = T, digits = 2)) %>% dplyr::rename_with(~ tolower(gsub(paste0(cont,"_"), "", .x)), .cols = starts_with(cont)) %>% dplyr::rename(FC = ratio) %>%
  DT::datatable(options = list(
  columnDefs = list(list(className = 'dt-center', targets = 1:5)),
  pageLength = 10)) %>% 
  formatStyle('FC', backgroundColor = styleInterval(c(-1, 1), c('lightblue', 'white', 'tomato'))) %>% 
  formatStyle('significant', color = styleEqual(c("True", "False"), c('green', 'red')))
```

### Volcano Plots TME1_R vs TME1_S {.tabset}

#### Significant

```{r V55,warning=FALSE, fig.height = 8, fig.width = 8}
cont <- "TME1_R_vs_TME1_S"
plot_volcano(dep, contrast=cont, add_names=T,label_size=4,adjusted = F)
```

```{r T55}
results %>%  select(starts_with("name") | starts_with(cont)) %>% filter(across(ends_with("significant"))) %>% arrange(across(ends_with("p.adj"))) %>% mutate(across(2:3, format, scientific = T, digits = 2)) %>% dplyr::rename_with(~ tolower(gsub(paste0(cont,"_"), "", .x)), .cols = starts_with(cont)) %>%  dplyr::rename(FC = ratio) %>%
  DT::datatable(options = list(
  columnDefs = list(list(className = 'dt-center', targets = 1:5)),
  pageLength = 10)) %>% 
  formatStyle('FC', backgroundColor = styleInterval(c(-1, 1), c('lightblue', 'white', 'tomato'))) %>% 
  formatStyle('significant', color = styleEqual(c("True", "False"), c('green', 'red')))
```

## Multiple sample ANOVA test {.tabset .tabset-fade .tabset-pills}

***

In this analysis, instead of comparing each possible condition individually, I used a multiple sample **ANOVA** test to observe the overall pattern of change.

```{r ANOVA, out.width = "200%"}
knitr::include_graphics("assets/HeatMap_t1_tme1_1500.png", error = F)
```

### Cluster Table

Now i select only the 4 bigger cluster:

```{r ANOVA_table, warning=F}
# anova_table %>%
#   group_by(Cluster) %>%
#   summarise(n = n())

name_cluster_order <- anova_table %>% 
  select(starts_with("T")) %>% colnames()

anova_table %>%
  select(!c(Majority.protein.IDs, ANOVA.Significant, Protein.names, Protein.IDs)) %>% 
  dplyr::rename(Log.ANOVA.p.value = X.Log.ANOVA.p.value) %>% 
  pivot_longer(cols = starts_with("T"), names_to = "Samples", values_to = "Intensity") %>% 
  mutate(across(contains(c(".value", "Intensity")), format, scientific = T, digits = 2)) %>% 
  mutate(Cluster = case_when(Cluster == "Cluster -499" ~ "Cluster 1",
                             Cluster == "Cluster -493" ~ "Cluster 2",
                             Cluster == "Cluster -501" ~ "Cluster 3")) %>% 
  relocate(Cluster, Samples, Intensity) %>% 
  DT::datatable()
  
```


### Cluster Profile {.tabset}

#### Cluster 1

```{r Cluster_1, warning=FALSE, message=FALSE, fig.height = 8, fig.width = 8}
anova_table %>% 
  filter(Cluster == "Cluster -499") %>% 
  select(starts_with("T") | contains("Gene")) %>% 
  pivot_longer(!Gene.names, names_to ="Sample", values_to = "Intensity") %>% 
  ggplot(mapping = aes(x = Sample, y = Intensity, group = Gene.names)) + 
  geom_line(color = "khaki4") + 
  geom_smooth(aes(group = 1)) + 
  scale_x_discrete(limits = name_cluster_order) +
  theme(axis.title = element_text(size = 16, face = "bold"),
        axis.text = element_text(size = 14), 
        axis.text.x = element_text(angle = 90), 
        plot.title = element_text(size = 20, face = "bold", hjust = 0.5),
        panel.background = element_rect(fill = NA)) +
  labs(title = "Cluster 1", x = "Samples", y = "log2 Intensity") + 
  theme(axis.line = element_line(size = 0.4, linetype = "solid"), 
        axis.text.x = element_text(vjust = 0.5))
```

#### Cluster 2

```{r Cluster_2, warning=FALSE, message=FALSE, fig.height = 8, fig.width = 8}
anova_table %>% 
  filter(Cluster == "Cluster -493") %>% 
  select(starts_with("T") | contains("Gene")) %>% 
  pivot_longer(!Gene.names, names_to ="Sample", values_to = "Intensity") %>% 
  ggplot(mapping = aes(x = Sample, y = Intensity, group = Gene.names)) + 
  geom_line(color = "blue") + 
  geom_smooth(aes(group = 1)) + 
  scale_x_discrete(limits = name_cluster_order) +
  theme(axis.title = element_text(size = 16, face = "bold"),
        axis.text = element_text(size = 14), 
        axis.text.x = element_text(angle = 90), 
        plot.title = element_text(size = 20, face = "bold", hjust = 0.5),
        panel.background = element_rect(fill = NA)) +
  labs(title = "Cluster 2", x = "Samples", y = "log2 Intensity") + 
  theme(axis.line = element_line(size = 0.4, linetype = "solid"), 
        axis.text.x = element_text(vjust = 0.5))
```

#### Cluster 3

```{r Cluster_3, warning=FALSE, message=FALSE, fig.height = 8, fig.width = 8}
anova_table %>% 
  filter(Cluster == "Cluster -501") %>% 
  select(starts_with("T") | contains("Gene")) %>% 
  pivot_longer(!Gene.names, names_to ="Sample", values_to = "Intensity") %>% 
  ggplot(mapping = aes(x = Sample, y = Intensity, group = Gene.names)) + 
  geom_line(color = "aquamarine2") + 
  geom_smooth(aes(group = 1)) + 
  scale_x_discrete(limits = name_cluster_order) +
  theme(axis.title = element_text(size = 16, face = "bold"),
        axis.text = element_text(size = 14), 
        axis.text.x = element_text(angle = 90), 
        plot.title = element_text(size = 20, face = "bold", hjust = 0.5),
        panel.background = element_rect(fill = NA)) +
  labs(title = "Cluster 3", x = "Samples", y = "log2 Intensity") + 
  theme(axis.line = element_line(size = 0.4, linetype = "solid"), 
        axis.text.x = element_text(vjust = 0.5))
```

### Gene Ontology {.tabset}

```{r}
b <- anova_table %>% select(Gene.names) %>% mutate(Gene.names = gsub(pattern = ";.*$", replacement = "", Gene.names)) %>% pull(Gene.names)

write.table(b,file = "gene_name_BP_anova.txt", sep="\t", quote=FALSE, row.names=FALSE, col.names = F)
```

#### BP

```{r BP_anova, warning=FALSE, fig.height = 7, fig.width = 10}
library(treemap) 								# treemap package by Martijn Tennekes

# Set the working directory if necessary
# setwd("C:/Users/username/workingdir");

# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","freqInDbPercent","value","uniqueness","dispensability","representative");
revigo.data <- rbind(c("GO:0006397","mRNA processing",0.870,5.983,0.720,0.000,"mRNA processing"),
c("GO:0034641","cellular nitrogen compound metabolic process",26.444,10.133,0.845,0.141,"mRNA processing"),
c("GO:0044237","cellular metabolic process",51.477,3.461,0.904,0.175,"mRNA processing"),
c("GO:0046483","heterocycle metabolic process",21.847,8.006,0.876,0.226,"mRNA processing"),
c("GO:0006725","cellular aromatic compound metabolic process",21.889,6.917,0.876,0.226,"mRNA processing"),
c("GO:0006139","nucleobase-containing compound metabolic process",18.964,9.072,0.731,0.371,"mRNA processing"),
c("GO:0016071","mRNA metabolic process",1.124,5.346,0.748,0.501,"mRNA processing"),
c("GO:0044271","cellular nitrogen compound biosynthetic process",14.482,4.542,0.752,0.544,"mRNA processing"),
c("GO:0034645","cellular macromolecule biosynthetic process",11.134,3.032,0.808,0.547,"mRNA processing"),
c("GO:0016070","RNA metabolic process",8.077,7.100,0.713,0.569,"mRNA processing"),
c("GO:0006396","RNA processing",3.789,4.932,0.709,0.596,"mRNA processing"),
c("GO:0006259","DNA metabolic process",5.371,3.014,0.725,0.627,"mRNA processing"),
c("GO:0008380","RNA splicing",0.610,5.697,0.739,0.640,"mRNA processing"),
c("GO:0090304","nucleic acid metabolic process",13.823,9.606,0.721,0.653,"mRNA processing"),
c("GO:0045934","negative regulation of nucleobase-containing compound metabolic process",0.749,4.857,0.647,0.000,"negative regulation of nucleobase-containing compound metabolic process"),
c("GO:0060333","interferon-gamma-mediated signaling pathway",0.002,4.004,0.869,0.167,"negative regulation of nucleobase-containing compound metabolic process"),
c("GO:0001910","regulation of leukocyte mediated cytotoxicity",0.010,3.078,0.876,0.185,"negative regulation of nucleobase-containing compound metabolic process"),
c("GO:0016241","regulation of macroautophagy",0.028,4.323,0.819,0.276,"negative regulation of nucleobase-containing compound metabolic process"),
c("GO:0044319","wound healing, spreading of cells",0.004,3.234,0.933,0.284,"negative regulation of nucleobase-containing compound metabolic process"),
c("GO:0019219","regulation of nucleobase-containing compound metabolic process",10.210,6.616,0.665,0.509,"negative regulation of nucleobase-containing compound metabolic process"),
c("GO:0007159","leukocyte cell-cell adhesion",0.008,4.262,0.964,0.007,"leukocyte cell-cell adhesion"),
c("GO:0070486","leukocyte aggregation",0.002,3.804,0.965,0.607,"leukocyte cell-cell adhesion"),
c("GO:0051641","cellular localization",2.368,5.354,0.908,0.012,"cellular localization"),
c("GO:0034613","cellular protein localization",1.588,4.759,0.817,0.330,"cellular localization"),
c("GO:0009060","aerobic respiration",0.977,3.023,0.890,0.069,"aerobic respiration"),
c("GO:0042398","cellular modified amino acid biosynthetic process",0.524,3.017,0.910,0.085,"cellular modified amino acid biosynthetic process"),
c("GO:1901360","organic cyclic compound metabolic process",22.607,5.357,0.907,0.086,"organic cyclic compound metabolic process"));

stuff <- data.frame(revigo.data);
names(stuff) <- revigo.names;

stuff$value <- as.numeric( as.character(stuff$value) );
stuff$freqInDbPercent <- as.numeric( as.character(stuff$freqInDbPercent) );
stuff$uniqueness <- as.numeric( as.character(stuff$uniqueness) );
stuff$dispensability <- as.numeric( as.character(stuff$dispensability) );

# check the tmPlot command documentation for all possible parameters - there are a lot more
treemap(
  stuff,
  index = c("representative","description"),
  vSize = "value",
  type = "categorical",
  vColor = "representative",
  fontsize.labels=c(16,14),
  title = "REVIGO TreeMap",
  inflate.labels = FALSE,      # set this to TRUE for space-filling group labels - good for posters
  lowerbound.cex.labels = 0,
  palette = "Set1",
  align.labels=list(
        c("center", "top"), 
        c("center", "bottom")
        ),# try to draw as many labels as possible (still, some small squares may not get a label)
  position.legend = "none"
)

```


#### MF

```{r MF_anova, warning=FALSE, fig.height = 7, fig.width = 10}
library(treemap) 								# treemap package by Martijn Tennekes

# Set the working directory if necessary
# setwd("C:/Users/username/workingdir");

# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.
revigo.names <- c("term_ID","description","freqInDbPercent","value","uniqueness","dispensability","representative");
revigo.data <- rbind(c("GO:0003676","nucleic acid binding",19.181,14.403,0.663,0.000,"nucleic acid binding"),
c("GO:1901363","heterocyclic compound binding",39.169,13.004,0.729,0.212,"nucleic acid binding"),
c("GO:0003723","RNA binding",5.649,12.049,0.661,0.252,"nucleic acid binding"),
c("GO:0097159","organic cyclic compound binding",39.193,12.227,0.729,0.293,"nucleic acid binding"),
c("GO:0035639","purine ribonucleoside triphosphate binding",15.552,3.432,0.677,0.325,"nucleic acid binding"),
c("GO:0003677","DNA binding",11.454,4.007,0.625,0.530,"nucleic acid binding"),
c("GO:0008234","cysteine-type peptidase activity",0.495,3.019,1.000,0.000,"cysteine-type peptidase activity"),
c("GO:0015318","inorganic molecular entity transmembrane transporter activity",2.829,3.005,1.000,0.000,"inorganic molecular entity transmembrane transporter activity"));

stuff <- data.frame(revigo.data);
names(stuff) <- revigo.names;

stuff$value <- as.numeric( as.character(stuff$value) );
stuff$freqInDbPercent <- as.numeric( as.character(stuff$freqInDbPercent) );
stuff$uniqueness <- as.numeric( as.character(stuff$uniqueness) );
stuff$dispensability <- as.numeric( as.character(stuff$dispensability) );

# check the tmPlot command documentation for all possible parameters - there are a lot more
treemap(
  stuff,
  index = c("representative","description"),
  vSize = "value",
  type = "categorical",
  vColor = "representative",
  fontsize.labels=c(16,14),
  title = "REVIGO TreeMap",
  inflate.labels = FALSE,      # set this to TRUE for space-filling group labels - good for posters
  lowerbound.cex.labels = 0,
  palette = "Set1",
  align.labels=list(
        c("center", "top"), 
        c("center", "bottom")
        ),# try to draw as many labels as possible (still, some small squares may not get a label)
  position.legend = "none"
)
```


## {-}

***
```{r excel_anova}
a <- anova_table %>%
  as.data.frame() %>%
  relocate(starts_with("T"), .after = last_col()) %>% 
  select(-Majority.protein.IDs) %>% 
  relocate(c(X.Log.ANOVA.p.value, ANOVA.q.value), .after = Gene.names) %>% 
  relocate(Gene.names, .after = Cluster) %>% 
  dplyr::rename(ANOVA.p.value = X.Log.ANOVA.p.value) %>% 
  dplyr::rename_with(., ~ gsub(pattern = "^", replacement = "Z_Score_", .), starts_with("T")) %>%
  mutate(across(contains(".value"), format, scientific = T, digits = 2)) %>% 
  mutate(across(contains("Z_Score_"), .fns = ~ round(.x, digits = 2))) 

test <- "ANOVA_results"

addWorksheet(excel, sheetName = test, gridLines = F)

writeDataTable(excel, sheet = test, x = a, keepNA = T, na.string = "NaN")

n_row <- a %>% nrow() + 1

n_col <- a %>% ncol()

setColWidths(excel, sheet = test, cols = 1:n_col, widths = 21)

addStyle(excel, sheet = test, style = header_style, rows = 1, cols = 1:n_col, gridExpand = T)

addStyle(excel, sheet = test, style = body_style, rows = 2:n_row, cols = 1:n_col, gridExpand = T)
```


```{r save_excel}
saveWorkbook(excel, here::here("output/results_T1_vs_TME1_sub-populations.xlsx"), overwrite = T)
```
