---
title: "T2 vs TME2 Sub-populations anaysis"
author: "Fabio Bedin | MS-Unit"
output: html_document
---

# Ovarian cancer dataset

***

```{r librerie, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(fig.align = "center")
library("MBQN")
library("dplyr")
library("tidyr")
library("DEP")
library("SummarizedExperiment")
library("preprocessCore")
library("tibble")
library("ggplot2")
library("enrichR")
library("DT")
library("stringr")
library("patchwork")
library("openxlsx")
```

```{r Custom Functions}

source(here::here("code/custom_functions.R"))

```

```{r excel_define_wb}
header_style <- createStyle(
  fontSize = 12,
  fontColour = "#0f0f0f",
  fgFill = "#faf2ca",
  halign = "center",
  border = "TopBottomLeftRight"
)

body_style <- createStyle(
  halign = "center",
  border = "TopBottomLeftRight"
)

excel <- createWorkbook()
```

## Introduction

***

For this analysis i will focus in all the possible combination between **T2** and **TME2** populations, where different sub-populations can be identify as sensivie **(S)**, intermediate sensitive **(IS)** and resistant **(R)**.

```{r input_Data}
data <- read.csv(here::here("data/proteinGroups.txt"), header = TRUE,stringsAsFactors = FALSE, sep = "\t")

expdesign <- read.table(here::here("data/expdesign_SI_S_R.tsv"), header = T, stringsAsFactors = F)

anova_table <- read.table(here::here("data/anova_table_t2_tme2_filt_1500.txt"), header = T, stringsAsFactors = F, sep = "\t")

conditions<-c("T2_IS","T2_S","T2_R","TME2_IS","TME2_S","TME2_R")

expdesign <- subset(expdesign, condition %in% conditions)
```

## Data wrangling

***

First, we need to wrangling the original dataset. From the **proteinGroups** obtained by **MaxQuant** software, I remove proteins that are marked form the software as *potential contaminant*, *only identify by site* and *reverse*.

```{r Data wrangling}
data <- data[data$Reverse != "+" & data$Potential.contaminant != "+" & data$Only.identified.by.site != "+",]

data_unique <- make_unique(data, "Gene.names", "Protein.IDs", delim = ";")

columns<-match(paste("LFQ.intensity.",expdesign$label,sep=""),colnames(data_unique))

data_se <- make_se(data_unique, columns, expdesign)
```

To understand the structure of the dataset and define a strategy, I use different quality control polts:

### Quality control Plots {.tabset .tabset-fade .tabset-pills}

All this plot can also hepl to decide the strategy to filter the missing data.

#### Frequency plot

```{r plot_freq, fig.height = 7, fig.width = 12}
plot_frequency(data_se)
```

#### Number of protein per sample

```{r plot numbers, fig.height = 7, fig.width = 12}
plot_numbers(data_se)
```

#### Distibution of each sample

```{r plot_norm, fig.height = 10, fig.width = 10}
plot_normalization(data_se)
```

## The problem of missing data

***

In the original dataset, the percentage of missing data is **`r round((sum(is.na(assay(data_se)))/prod(dim(assay(data_se))))*100, digits = 2)`%** and can be visualized with the following heatmap.

```{r plot_mv, message=FALSE, fig.height = 7, fig.width = 12}
plot_missval(data_se)
```

## Filter missing data

***

To reduce the percentage of missing data I have decided to filter samples with less then **1500** identify proteins.
Afrer that, I also use a second strong filter to remove all the proteins that are not identify in all replicates in at least 1 groups.

```{r data_filt}
data_filt <- filter_bad_samples(data_se, thr = 1500)

data_filt <- filter_missval(data_filt, thr = 0)
```


These plot show the effect of the filters. Now we reduce the percetage of missing data in the dataset down to: **`r round((sum(is.na(assay(data_filt)))/prod(dim(assay(data_filt))))*100, digits = 2)`%**.

### Plots with filters {.tabset .tabset-fade .tabset-pills}

#### protein coverage

```{r plot_coverage}
plot_coverage(data_filt)
```

#### Number of protein per sample

```{r plot_numbers_filt}
plot_numbers(data_filt)
```

#### Distibution

```{r plot_norm_filt, fig.height = 8, fig.width = 8}
plot_normalization(data_filt)
```

#### Plot missing data

```{r plot_mv_filt, message=FALSE}
plot_missval(data_filt)
```

## Imputation

***

An idealized version of a label-free discovery mass spectrometry proteomics experiment would provide absolute abundance measurements for a whole proteome, across varying conditions. Unfortunately, this ideal is not realized. Measurements are made on peptides requiring an inferential step to obtain protein level estimates. The inference is complicated by experimental factors that necessitate relative abundance estimation and result in widespread **non-ignorable missing data**. Relative abundance on the log scale takes the form of parameter contrasts. In a complete-case analysis, contrast estimates may be biased by missing data and a substantial amount of useful information will often go unused.

To avoid this problem, we impute missing data using random draws from a manually defined left-shifted Gaussian distribution.


```{r mixed_imputation, message=FALSE, warning=FALSE, results='hide'}
#set.seed(20210520)
data_imp_man <- impute(data_filt, fun = "man", shift = 1.8, scale = 0.3)
```

This plot show the effect of imputation, as expected a new population of low intensity values as emerged.

```{r plot_imputation}
plot_imputation(data_filt, data_imp_man)
```

## Differential enrichment analysis

***

Protein-wise linear models combined with empirical Bayes statistics are used for the differential enrichment analysis (or differential expression analysis). I test all possible combination between T1 S, IS and R vs T2 S, IS, and R.

```{r test_diff}
data_diff <- test_diff(data_imp_man, type = "all")
```

I define sifnificant protein with **FDR = 0.05** and **FC = 1**

```{r Diff_prot}
dep <- add_rejections(data_diff, alpha = 0.05, lfc = 1)

results<-get_results(dep)
```
 
There are **`r results %>% filter(significant) %>% nrow()`** significant proteins within all comparisons.
The following table shows the results:

```{r excel_table_res}
a <- get_df_wide(dep) %>% 
  select(c(name, Protein.IDs, Protein.names, starts_with(conditions), -ends_with(c("CI.R", "CI.L")))) %>% 
  relocate(contains("_vs_"), .after = where(is.character)) %>% 
  dplyr::rename_with(., ~ gsub(pattern = "^", replacement = "LFQ_intensity_", .), !contains(c("_vs_", "name", "Protein."))) %>%
  dplyr::rename_with(., ~ gsub(pattern = "....$", replacement = "FC", .), ends_with("_diff")) %>%
  relocate(contains("_significant"), .before = starts_with("name")) %>% 
  mutate(across(ends_with("_significant"), ~ if_else(.x, "+", ""))) %>% 
  mutate(across(contains(c("_p.adj", "_p.val")), format, scientific = T, digits = 2)) %>% 
  mutate(across(contains(c("LFQ", "FC")), .fns = ~ round(.x, digits = 2))) 

test <- "Multiple_comparisons_results"

addWorksheet(excel, sheetName = test, gridLines = F)

writeDataTable(excel, sheet = test, x = a, keepNA = T, na.string = "NaN")

n_row <- a %>% nrow() + 1

n_col <- a %>% ncol()

setColWidths(excel, sheet = test, cols = 1:n_col, widths = 21)

addStyle(excel, sheet = test, style = header_style, rows = 1, cols = 1:n_col, gridExpand = T)

addStyle(excel, sheet = test, style = body_style, rows = 2:n_row, cols = 1:n_col, gridExpand = T)
```

```{r significant_table}
results %>% 
  filter(significant) %>%
  select(-significant, -ends_with("centered")) %>% 
  pivot_longer(
   cols = ends_with(c("p.val", "p.adj", "significant", "ratio")),
   names_to = c("comparison", ".value"),
   names_pattern = "(.*_.*_.*_.*_.*)_(.*)") %>% 
  arrange(name, p.adj) %>% 
  relocate(comparison) %>% 
  mutate(across(ends_with(c("p.val", "p.adj")), format, scientific = T, digits = 2)) %>% 
  mutate(significant = str_to_title(significant)) %>% 
  dplyr::rename(FC = ratio) %>%
  DT::datatable(options = list(
  columnDefs = list(list(className = 'dt-center', targets = 1:7)),
  pageLength = 15)) %>% 
  formatStyle('FC', backgroundColor = styleInterval(c(-1, 1), c('lightblue', 'white', 'tomato'))) %>% 
  formatStyle('significant', color = styleEqual(c("True", "False"), c('green', 'red')))
```


### Visualization of the results {.tabset .tabset-fade .tabset-pills}

***

The results from the analysis can be easily visualized by a number of plot functions.

#### PCA

```{r PCA, message=FALSE, warning=FALSE, fig.height = 8, fig.width = 8}
plot_pca(dep, x = 1, y = 2, n = 500, point_size = 4, indicate = "condition")
```


#### Correlation matrix

```{r plot_cor, fig.height = 8}
plot_cor(dep, significant = TRUE, lower = 0, upper = 1, pal = "Reds")
```


#### Overview all volcano plot

```{r all_volcano, warning=FALSE, fig.height = 8, fig.width = 8}
cont <- "T2_IS_vs_T2_R"
p1 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T2_IS_vs_T2_S"
p2 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T2_IS_vs_TME2_IS"
p3 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T2_IS_vs_TME2_R"
p4 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T2_IS_vs_TME2_S"
p5 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T2_R_vs_T2_S"
p6 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T2_R_vs_TME2_IS"
p7 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T2_R_vs_TME2_R"
p8 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T2_R_vs_TME2_S"
p9 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T2_S_vs_TME2_IS"
p10 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T2_S_vs_TME2_R"
p11 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T2_S_vs_TME2_S"
p12 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "TME2_IS_vs_TME2_R"
p13 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "TME2_IS_vs_TME2_S"
p14 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "TME2_R_vs_TME2_S"
p15 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
```

```{r plot_v_all, fig.height = 12, fig.width = 18}
(p1 | p2 | p3 | p4 | p5) /
  (p6 | p7 | p8 | p9 | p10) /
  (p11 | p12 | p13 | p14 | p15) 
```

#### Cluster Heatmap

```{r heatmap, fig.height = 15, fig.width = 10}
plot_heatmap(dep, type = "centered", kmeans = TRUE, k = 7, show_row_names = T, indicate = "condition",col_limit = 5)
```

```{r}
#cluster_table <- plot_heatmap(dep, type = "centered", kmeans = TRUE, k = 8, show_row_names = T, indicate = "condition",col_limit = 5, plot = F) %>% dplyr::rename(cluster = k) 

#write.table(cluster_table ,file = "cluster_table.txt", sep="\t", quote=FALSE, row.names=FALSE, col.names = T)
```


### Volcano Plots T2_R vs TME2_R {.tabset}

#### Significant

```{r V1,warning=FALSE, fig.height = 8, fig.width = 8}
cont <- "T2_R_vs_TME2_R"
plot_volcano(dep, contrast=cont, add_names=T,label_size=4,adjusted = F)
```

```{r T1}
results %>%  select(starts_with("name") | starts_with(cont)) %>% filter(across(ends_with("significant"))) %>% arrange(across(ends_with("p.adj"))) %>% mutate(across(2:3, format, scientific = T, digits = 2)) %>% dplyr::rename_with(~ tolower(gsub(paste0(cont,"_"), "", .x)), .cols = starts_with(cont)) %>%  dplyr::rename(FC = ratio) %>%
  DT::datatable(options = list(
  columnDefs = list(list(className = 'dt-center', targets = 1:5)),
  pageLength = 10)) %>% 
  formatStyle('FC', backgroundColor = styleInterval(c(-1, 1), c('lightblue', 'white', 'tomato'))) %>% 
  formatStyle('significant', color = styleEqual(c("True", "False"), c('green', 'red')))
```

#### Unique

```{r V2,warning=FALSE, fig.height = 8, fig.width = 8}
unique_vec <- c("AAMDC","AHDC1","ARL8A","CMTR1","DDX21","DR1","EYA4","GIPC1","MRPS36","MTPAP","PABPN1","PGAM5","ROR1","SMAD2","SNRPC","SNRPD3","TRIM56","UFD1L","UTS2","XRCC1","ZNF207","PRR12","SACS","BCKDHB","CBX5","DCP1B","LTBP4","METAP1","NCOR2","NDUFB10","NXF1","POLR2E","RAB43","SETD3","TGM1","TOMM22","VWA5A","APOBEC3C","SERPINA6","STX1A","UPF2")
#unique_vec <- c("AHDC1","UTS2","SACS","BCKDHB","CBX5","DCP1B","LTBP4","METAP1","NCOR2","NDUFB10","NXF1","POLR2E","RAB43","SETD3","TGM1","TOMM22","VWA5A","APOBEC3C","SERPINA6","STX1A","UPF2")
plot_volcano_2(dep, contrast=cont, add_names=T,label_size=4,adjusted = F)
```

```{r T2}
results %>%  select(starts_with("name") | starts_with(cont)) %>% filter(name %in% unique_vec) %>% arrange(across(ends_with("p.adj"))) %>% mutate(across(2:3, format, scientific = T, digits = 2)) %>% dplyr::rename_with(~ tolower(gsub(paste0(cont,"_"), "", .x)), .cols = starts_with(cont)) %>% DT::datatable()
```

### Volcano Plots T2_S vs TME2_S {.tabset}

#### Significant

```{r V5,warning=FALSE, fig.height = 8, fig.width = 8}
cont <- "T2_S_vs_TME2_S"
plot_volcano(dep, contrast=cont, add_names=T,label_size=4,adjusted = F)
```

```{r T5}
results %>%  select(starts_with("name") | starts_with(cont)) %>% filter(across(ends_with("significant"))) %>% arrange(across(ends_with("p.adj"))) %>% mutate(across(2:3, format, scientific = T, digits = 2)) %>% dplyr::rename_with(~ tolower(gsub(paste0(cont,"_"), "", .x)), .cols = starts_with(cont)) %>%  dplyr::rename(FC = ratio) %>%
  DT::datatable(options = list(
  columnDefs = list(list(className = 'dt-center', targets = 1:5)),
  pageLength = 10)) %>% 
  formatStyle('FC', backgroundColor = styleInterval(c(-1, 1), c('lightblue', 'white', 'tomato'))) %>% 
  formatStyle('significant', color = styleEqual(c("True", "False"), c('green', 'red')))
```

#### Unique

```{r V6,warning=FALSE, fig.height = 8, fig.width = 8}
unique_vec <- unique_vec <- c("AAMDC","AHDC1","ARL8A","CMTR1","DDX21","DR1","EYA4","GIPC1","MRPS36","MTPAP","PABPN1","PGAM5","ROR1","SMAD2","SNRPC","SNRPD3","TRIM56","UFD1L","UTS2","XRCC1","ZNF207","PRR12","SACS","BUD31","CDKN2A","CTSL","DCP1B","DENND4C","ESRP1","EWSR1","NCKAP1L","NIPSNAP1","RDH14","SEL1L","SRSF9","SYNGR2","TBCD","TFCP2","TMLHE","APOBEC3C","ECM1","GPNMB","PODN","SGCD","SRPX")
#unique_vec <- unique_vec <- c("MRPS36","SNRPC","BUD31","CDKN2A","CTSL","DCP1B","DENND4C","ESRP1","EWSR1","NCKAP1L","NIPSNAP1","RDH14","SEL1L","SRSF9","SYNGR2","TBCD","TFCP2","TMLHE","APOBEC3C","ECM1","GPNMB","PODN","SGCD","SRPX")
plot_volcano_2(dep, contrast=cont, add_names=T,label_size=4,adjusted = F)
```

```{r T6}
results %>%  select(starts_with("name") | starts_with(cont)) %>% filter(name %in% unique_vec) %>% arrange(across(ends_with("p.adj"))) %>% mutate(across(2:3, format, scientific = T, digits = 2)) %>% dplyr::rename_with(~ tolower(gsub(paste0(cont,"_"), "", .x)), .cols = starts_with(cont)) %>% DT::datatable()
```

### Volcano Plots T2_IS vs TME2_IS {.tabset}

#### Significant

```{r V3,warning=FALSE, fig.height = 8, fig.width = 8}
cont <- "T2_IS_vs_TME2_IS"
plot_volcano(dep, contrast=cont, add_names=T,label_size=4,adjusted = F)
```

```{r T3}
results %>%  select(starts_with("name") | starts_with(cont)) %>% filter(across(ends_with("significant"))) %>% arrange(across(ends_with("p.adj"))) %>% mutate(across(2:3, format, scientific = T, digits = 2)) %>% dplyr::rename_with(~ tolower(gsub(paste0(cont,"_"), "", .x)), .cols = starts_with(cont)) %>%  dplyr::rename(FC = ratio) %>%
  DT::datatable(options = list(
  columnDefs = list(list(className = 'dt-center', targets = 1:5)),
  pageLength = 10)) %>% 
  formatStyle('FC', backgroundColor = styleInterval(c(-1, 1), c('lightblue', 'white', 'tomato'))) %>% 
  formatStyle('significant', color = styleEqual(c("True", "False"), c('green', 'red')))
```

#### Unique

```{r V4,warning=FALSE, fig.height = 8, fig.width = 8}
unique_vec <- c("AAMDC","AHDC1","ARL8A","CMTR1","DDX21","DR1","EYA4","GIPC1","MRPS36","MTPAP","PABPN1","PGAM5","ROR1","SMAD2","SNRPC","SNRPD3","TRIM56","UFD1L","UTS2","XRCC1","ZNF207","PRR12","SACS","ATP5J2-PTCD1","C16orf13","GCA","GNG12","GNL3","MAVS","PARD6B","PDCD11","POLR2A","PTGES2","RTN1","SH3KBP1","SNRPA1","SYNGR2","TMEM201","TXLNA")
#unique_vec <- c("AAMDC","ATP5J2-PTCD1","C16orf13","GCA","GNG12","GNL3","MAVS","PARD6B","PDCD11","POLR2A","PTGES2","RTN1","SH3KBP1","SNRPA1","SYNGR2","TMEM201","TXLNA")
plot_volcano_2(dep, contrast=cont, add_names=T,label_size=4,adjusted = F)
```

```{r T4}
results %>%  select(starts_with("name") | starts_with(cont)) %>% filter(name %in% unique_vec) %>% arrange(across(ends_with("p.adj"))) %>% mutate(across(2:3, format, scientific = T, digits = 2)) %>% dplyr::rename_with(~ tolower(gsub(paste0(cont,"_"), "", .x)), .cols = starts_with(cont)) %>% DT::datatable()
```

### Volcano Plots TME2_R vs TME2_S {.tabset}

#### Significant

```{r V55,warning=FALSE, fig.height = 8, fig.width = 8}
cont <- "TME2_R_vs_TME2_S"
plot_volcano(dep, contrast=cont, add_names=T,label_size=4,adjusted = F)
```

```{r T55}
results %>%  select(starts_with("name") | starts_with(cont)) %>% filter(across(ends_with("significant"))) %>% arrange(across(ends_with("p.adj"))) %>% mutate(across(2:3, format, scientific = T, digits = 2)) %>% dplyr::rename_with(~ tolower(gsub(paste0(cont,"_"), "", .x)), .cols = starts_with(cont)) %>%  dplyr::rename(FC = ratio) %>%
  DT::datatable(options = list(
  columnDefs = list(list(className = 'dt-center', targets = 1:5)),
  pageLength = 10)) %>% 
  formatStyle('FC', backgroundColor = styleInterval(c(-1, 1), c('lightblue', 'white', 'tomato'))) %>% 
  formatStyle('significant', color = styleEqual(c("True", "False"), c('green', 'red')))
```


## Multiple sample ANOVA test {.tabset .tabset-fade .tabset-pills}

***

In this analysis, instead of comparing each possible condition individually, I used a multiple sample **ANOVA** test to observe the overall pattern of change.

```{r ANOVA, out.width = "200%"}
knitr::include_graphics("assets/HeatMap_t2_tme2_1500.png", error = F)
```

### Cluster Table

Now i select only the 4 bigger cluster:

```{r ANOVA_table, warning=F}
# anova_table %>%
#   group_by(Cluster) %>%
#   summarise(n = n())

name_cluster_order <- anova_table %>% 
  select(starts_with("T")) %>% colnames()

anova_table %>%
  select(!c(Majority.protein.IDs, ANOVA.Significant, Protein.names, Protein.IDs)) %>% 
  dplyr::rename(Log.ANOVA.p.value = X.Log.ANOVA.p.value) %>% 
  pivot_longer(cols = starts_with("T"), names_to = "Samples", values_to = "Intensity") %>% 
  mutate(across(contains(c(".value", "Intensity")), format, scientific = T, digits = 2)) %>% 
  mutate(Cluster = case_when(Cluster == "Cluster -893" ~ "Cluster 1",
                             Cluster == "Cluster -892" ~ "Cluster 2")) %>% 
  relocate(Cluster, Samples, Intensity) %>% 
  DT::datatable()
  
```


### Cluster Profile {.tabset}

#### Cluster 1

```{r Cluster_1, warning=FALSE, message=FALSE, fig.height = 8, fig.width = 8}
anova_table %>% 
  filter(Cluster == "Cluster -893") %>% 
  select(starts_with("T") | contains("Gene")) %>% 
  pivot_longer(!Gene.names, names_to ="Sample", values_to = "Intensity") %>% 
  ggplot(mapping = aes(x = Sample, y = Intensity, group = Gene.names)) + 
  geom_line(color = "lightblue2") + 
  geom_smooth(aes(group = 1)) + 
  scale_x_discrete(limits = name_cluster_order) +
  theme(axis.title = element_text(size = 16, face = "bold"),
        axis.text = element_text(size = 14), 
        axis.text.x = element_text(angle = 90), 
        plot.title = element_text(size = 20, face = "bold", hjust = 0.5),
        panel.background = element_rect(fill = NA)) +
  labs(title = "Cluster 1", x = "Samples", y = "log2 Intensity") + 
  theme(axis.line = element_line(size = 0.4, linetype = "solid"), 
        axis.text.x = element_text(vjust = 0.5))
```

#### Cluster 2

```{r Cluster_2, warning=FALSE, message=FALSE, fig.height = 8, fig.width = 8}
anova_table %>% 
  filter(Cluster == "Cluster -892") %>% 
  select(starts_with("T") | contains("Gene")) %>% 
  pivot_longer(!Gene.names, names_to ="Sample", values_to = "Intensity") %>% 
  ggplot(mapping = aes(x = Sample, y = Intensity, group = Gene.names)) + 
  geom_line(color = "orange") + 
  geom_smooth(aes(group = 1)) + 
  scale_x_discrete(limits = name_cluster_order) +
  theme(axis.title = element_text(size = 16, face = "bold"),
        axis.text = element_text(size = 14), 
        axis.text.x = element_text(angle = 90), 
        plot.title = element_text(size = 20, face = "bold", hjust = 0.5),
        panel.background = element_rect(fill = NA)) +
  labs(title = "Cluster 2", x = "Samples", y = "log2 Intensity") + 
  theme(axis.line = element_line(size = 0.4, linetype = "solid"), 
        axis.text.x = element_text(vjust = 0.5))
```

### Gene Ontology {.tabset}

```{r}
b <- anova_table %>% select(Gene.names) %>% mutate(Gene.names = gsub(pattern = ";.*$", replacement = "", Gene.names)) %>% pull(Gene.names)

write.table(b,file = "gene_name_BP_anova.txt", sep="\t", quote=FALSE, row.names=FALSE, col.names = F)
```

#### BP

```{r BP_anova, warning=FALSE, fig.height = 7, fig.width = 10}
library(treemap) 								# treemap package by Martijn Tennekes

# Set the working directory if necessary
# setwd("C:/Users/username/workingdir");

# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","freqInDbPercent","value","uniqueness","dispensability","representative");
revigo.data <- rbind(c("GO:0001503","ossification",0.046,4.458,0.951,0.000,"ossification"),
c("GO:0035633","maintenance of blood-brain barrier",0.002,3.167,0.835,0.385,"ossification"),
c("GO:0071827","plasma lipoprotein particle organization",0.007,4.387,0.887,0.414,"ossification"),
c("GO:0071825","protein-lipid complex subunit organization",0.009,4.387,0.938,0.421,"ossification"),
c("GO:0034367","protein-containing complex remodeling",0.008,4.387,0.938,0.425,"ossification"),
c("GO:0003008","system process",0.752,4.409,0.942,0.567,"ossification"),
c("GO:0003013","circulatory system process",0.091,3.171,0.949,0.600,"ossification"),
c("GO:0002376","immune system process",0.876,6.197,1.000,0.000,"immune system process"),
c("GO:0009888","tissue development",0.456,6.714,0.956,0.000,"tissue development"),
c("GO:0031214","biomineral tissue development",0.017,3.372,0.930,0.542,"tissue development"),
c("GO:0035987","endodermal cell differentiation",0.008,3.171,0.928,0.591,"tissue development"),
c("GO:0009887","animal organ morphogenesis",0.235,4.067,0.917,0.667,"tissue development"),
c("GO:0001568","blood vessel development",0.116,3.131,0.924,0.679,"tissue development"),
c("GO:0022610","biological adhesion",0.553,10.682,1.000,0.000,"biological adhesion"),
c("GO:0030162","regulation of proteolysis",0.401,11.654,0.801,0.000,"regulation of proteolysis"),
c("GO:0110020","regulation of actomyosin structure organization",0.023,4.038,0.847,0.176,"regulation of proteolysis"),
c("GO:0050878","regulation of body fluid levels",0.082,4.595,0.841,0.181,"regulation of proteolysis"),
c("GO:0010810","regulation of cell-substrate adhesion",0.036,4.955,0.841,0.182,"regulation of proteolysis"),
c("GO:0050865","regulation of cell activation",0.097,3.374,0.846,0.196,"regulation of proteolysis"),
c("GO:0051346","negative regulation of hydrolase activity",0.274,10.775,0.799,0.200,"regulation of proteolysis"),
c("GO:0030155","regulation of cell adhesion",0.129,3.433,0.843,0.201,"regulation of proteolysis"),
c("GO:0051241","negative regulation of multicellular organismal process",0.181,4.712,0.780,0.207,"regulation of proteolysis"),
c("GO:0040012","regulation of locomotion",0.199,4.172,0.841,0.208,"regulation of proteolysis"),
c("GO:0002682","regulation of immune system process",0.288,8.312,0.837,0.215,"regulation of proteolysis"),
c("GO:0022603","regulation of anatomical structure morphogenesis",0.882,5.577,0.807,0.239,"regulation of proteolysis"),
c("GO:0051049","regulation of transport",0.482,4.532,0.767,0.243,"regulation of proteolysis"),
c("GO:0051239","regulation of multicellular organismal process",0.553,3.635,0.829,0.246,"regulation of proteolysis"),
c("GO:0032879","regulation of localization",0.712,4.128,0.825,0.253,"regulation of proteolysis"),
c("GO:0048583","regulation of response to stimulus",1.204,5.279,0.818,0.267,"regulation of proteolysis"),
c("GO:0070613","regulation of protein processing",0.012,11.609,0.828,0.489,"regulation of proteolysis"),
c("GO:1903317","regulation of protein maturation",0.013,11.609,0.841,0.491,"regulation of proteolysis"),
c("GO:0014910","regulation of smooth muscle cell migration",0.006,3.042,0.807,0.655,"regulation of proteolysis"),
c("GO:0050818","regulation of coagulation",0.019,3.750,0.827,0.696,"regulation of proteolysis"),
c("GO:0032501","multicellular organismal process",2.570,4.801,1.000,0.000,"multicellular organismal process"),
c("GO:0040011","locomotion",0.960,3.836,1.000,0.000,"locomotion"),
c("GO:0043062","extracellular structure organization",0.064,26.038,0.939,0.000,"extracellular structure organization"),
c("GO:0010324","membrane invagination",0.014,6.243,0.933,0.295,"extracellular structure organization"),
c("GO:0031032","actomyosin structure organization",0.036,8.827,0.922,0.313,"extracellular structure organization"),
c("GO:0030198","extracellular matrix organization",0.062,21.297,0.932,0.325,"extracellular structure organization"),
c("GO:0097435","supramolecular fiber organization",0.313,5.983,0.932,0.365,"extracellular structure organization"),
c("GO:0007010","cytoskeleton organization",0.790,4.190,0.925,0.472,"extracellular structure organization"),
c("GO:0006911","phagocytosis, engulfment",0.010,6.243,0.873,0.570,"extracellular structure organization"),
c("GO:0120009","intermembrane lipid transfer",0.034,4.600,0.888,0.610,"extracellular structure organization"),
c("GO:0045214","sarcomere organization",0.010,4.799,0.886,0.657,"extracellular structure organization"),
c("GO:0030048","actin filament-based movement",0.019,3.688,0.957,0.677,"extracellular structure organization"),
c("GO:0050896","response to stimulus",12.913,4.600,1.000,0.000,"response to stimulus"),
c("GO:0072376","protein activation cascade",0.004,19.983,0.924,0.000,"protein activation cascade"),
c("GO:0030203","glycosaminoglycan metabolic process",1.032,6.783,0.908,0.103,"protein activation cascade"),
c("GO:0043687","post-translational protein modification",0.028,4.943,0.974,0.141,"protein activation cascade"),
c("GO:0042157","lipoprotein metabolic process",0.277,3.396,0.974,0.185,"protein activation cascade"),
c("GO:0006956","complement activation",0.025,14.523,0.627,0.192,"protein activation cascade"),
c("GO:0019538","protein metabolic process",19.817,3.658,0.964,0.246,"protein activation cascade"),
c("GO:0006952","defense response",0.774,7.804,0.900,0.271,"protein activation cascade"),
c("GO:0001101","response to acid chemical",0.048,4.062,0.930,0.285,"protein activation cascade"),
c("GO:0019932","second-messenger-mediated signaling",0.085,3.040,0.789,0.299,"protein activation cascade"),
c("GO:0009617","response to bacterium",0.169,6.110,0.901,0.317,"protein activation cascade"),
c("GO:0009605","response to external stimulus",1.712,3.762,0.912,0.399,"protein activation cascade"),
c("GO:2000257","regulation of protein activation cascade",0.002,13.057,0.781,0.455,"protein activation cascade"),
c("GO:0006954","inflammatory response",0.094,5.003,0.897,0.500,"protein activation cascade"),
c("GO:1901136","carbohydrate derivative catabolic process",0.541,4.082,0.939,0.501,"protein activation cascade"),
c("GO:0009611","response to wounding",0.103,3.484,0.914,0.504,"protein activation cascade"),
c("GO:1901700","response to oxygen-containing compound",0.566,3.441,0.917,0.513,"protein activation cascade"),
c("GO:0007165","signal transduction",7.086,3.427,0.703,0.531,"protein activation cascade"),
c("GO:0042339","keratan sulfate metabolic process",0.001,3.936,0.924,0.532,"protein activation cascade"),
c("GO:0006022","aminoglycan metabolic process",1.120,6.783,0.938,0.541,"protein activation cascade"),
c("GO:0009612","response to mechanical stimulus",0.031,4.530,0.918,0.552,"protein activation cascade"),
c("GO:0050727","regulation of inflammatory response",0.057,11.342,0.748,0.553,"protein activation cascade"),
c("GO:0002526","acute inflammatory response",0.012,5.358,0.906,0.604,"protein activation cascade"),
c("GO:0006959","humoral immune response",0.043,13.048,0.877,0.617,"protein activation cascade"),
c("GO:0002252","immune effector process",0.068,7.854,0.917,0.625,"protein activation cascade"),
c("GO:0042340","keratan sulfate catabolic process",0.000,3.936,0.925,0.626,"protein activation cascade"),
c("GO:0032101","regulation of response to external stimulus",0.211,9.592,0.743,0.640,"protein activation cascade"),
c("GO:1903510","mucopolysaccharide metabolic process",0.025,7.202,0.929,0.690,"protein activation cascade"),
c("GO:0030449","regulation of complement activation",0.005,12.318,0.744,0.692,"protein activation cascade"),
c("GO:0002576","platelet degranulation",0.002,7.740,0.939,0.005,"platelet degranulation"),
c("GO:0016192","vesicle-mediated transport",1.371,6.014,0.954,0.179,"platelet degranulation"),
c("GO:0098657","import into cell",0.055,4.391,0.966,0.229,"platelet degranulation"),
c("GO:0048870","cell motility",0.572,3.231,0.942,0.275,"platelet degranulation"),
c("GO:0046903","secretion",0.631,4.189,0.960,0.288,"platelet degranulation"),
c("GO:0006897","endocytosis",0.264,5.724,0.931,0.506,"platelet degranulation"),
c("GO:0032940","secretion by cell",0.587,4.022,0.937,0.632,"platelet degranulation"),
c("GO:0019835","cytolysis",0.061,4.369,0.996,0.007,"cytolysis"),
c("GO:0007155","cell adhesion",0.540,10.682,0.986,0.008,"cell adhesion"),
c("GO:0030029","actin filament-based process",0.368,5.597,0.996,0.009,"actin filament-based process"),
c("GO:0006928","movement of cell or subcellular component",0.961,3.077,0.996,0.010,"movement of cell or subcellular component"),
c("GO:0044273","sulfur compound catabolic process",0.068,5.611,0.957,0.020,"sulfur compound catabolic process"),
c("GO:0044272","sulfur compound biosynthetic process",1.337,3.322,0.961,0.622,"sulfur compound catabolic process"),
c("GO:1901615","organic hydroxy compound metabolic process",0.975,3.638,0.987,0.036,"organic hydroxy compound metabolic process"),
c("GO:0016101","diterpenoid metabolic process",0.028,3.314,0.965,0.042,"diterpenoid metabolic process"),
c("GO:0008202","steroid metabolic process",0.207,3.119,0.969,0.422,"diterpenoid metabolic process"),
c("GO:0001523","retinoid metabolic process",0.021,3.314,0.966,0.687,"diterpenoid metabolic process"),
c("GO:1902652","secondary alcohol metabolic process",0.097,3.600,0.974,0.046,"secondary alcohol metabolic process"),
c("GO:0016125","sterol metabolic process",0.124,3.600,0.958,0.673,"secondary alcohol metabolic process"));

stuff <- data.frame(revigo.data);
names(stuff) <- revigo.names;

stuff$value <- as.numeric( as.character(stuff$value) );
stuff$freqInDbPercent <- as.numeric( as.character(stuff$freqInDbPercent) );
stuff$uniqueness <- as.numeric( as.character(stuff$uniqueness) );
stuff$dispensability <- as.numeric( as.character(stuff$dispensability) );

# check the tmPlot command documentation for all possible parameters - there are a lot more
treemap(
  stuff,
  index = c("representative","description"),
  vSize = "value",
  type = "categorical",
  vColor = "representative",
  fontsize.labels=c(16,14),
  title = "REVIGO TreeMap",
  inflate.labels = FALSE,      # set this to TRUE for space-filling group labels - good for posters
  lowerbound.cex.labels = 0,
  palette = "Set1",
  align.labels=list(
        c("center", "top"), 
        c("center", "bottom")
        ),# try to draw as many labels as possible (still, some small squares may not get a label)
  position.legend = "none"
)

```


#### MF

```{r MF_anova, warning=FALSE, fig.height = 7, fig.width = 10}
library(treemap) 								# treemap package by Martijn Tennekes

# Set the working directory if necessary
# setwd("C:/Users/username/workingdir");

# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.
revigo.names <- c("term_ID","description","freqInDbPercent","value","uniqueness","dispensability","representative");
revigo.data <- rbind(c("GO:0004866","endopeptidase inhibitor activity",0.152,10.636,0.717,0.000,"endopeptidase inhibitor activity"),
c("GO:0005198","structural molecule activity",2.916,7.259,1.000,0.000,"structural molecule activity"),
c("GO:0005201","extracellular matrix structural constituent",0.035,22.733,0.905,0.000,"extracellular matrix structural constituent"),
c("GO:0097493","structural molecule activity conferring elasticity",0.000,3.622,0.920,0.335,"extracellular matrix structural constituent"),
c("GO:0030020","extracellular matrix structural constituent conferring tensile strength",0.000,10.072,0.879,0.340,"extracellular matrix structural constituent"),
c("GO:0008307","structural constituent of muscle",0.004,4.354,0.912,0.381,"extracellular matrix structural constituent"),
c("GO:0030023","extracellular matrix constituent conferring elasticity",0.000,3.622,0.882,0.634,"extracellular matrix structural constituent"),
c("GO:0030021","extracellular matrix structural constituent conferring compression resistance",0.002,5.680,0.875,0.683,"extracellular matrix structural constituent"),
c("GO:0005319","lipid transporter activity",0.151,4.600,0.851,0.000,"lipid transporter activity"),
c("GO:0015248","sterol transporter activity",0.010,3.777,0.834,0.696,"lipid transporter activity"),
c("GO:0005539","glycosaminoglycan binding",0.113,10.371,0.965,0.000,"glycosaminoglycan binding"),
c("GO:0008201","heparin binding",0.038,8.567,0.967,0.203,"glycosaminoglycan binding"),
c("GO:0008236","serine-type peptidase activity",1.282,4.000,0.932,0.000,"serine-type peptidase activity"),
c("GO:0017171","serine hydrolase activity",1.283,4.000,0.967,0.303,"serine-type peptidase activity"),
c("GO:0098772","molecular function regulator",1.022,5.395,1.000,0.000,"molecular function regulator"),
c("GO:0050997","quaternary ammonium group binding",0.005,3.083,0.978,0.033,"quaternary ammonium group binding"),
c("GO:0003823","antigen binding",0.007,4.979,0.978,0.034,"antigen binding"),
c("GO:0050840","extracellular matrix binding",0.011,3.693,0.977,0.035,"extracellular matrix binding"),
c("GO:0005518","collagen binding",0.014,4.893,0.977,0.035,"collagen binding"),
c("GO:0005102","signaling receptor binding",0.505,6.854,0.914,0.045,"signaling receptor binding"),
c("GO:0048407","platelet-derived growth factor binding",0.001,3.917,0.935,0.350,"signaling receptor binding"),
c("GO:0034987","immunoglobulin receptor binding",0.004,6.243,0.930,0.389,"signaling receptor binding"),
c("GO:0002020","protease binding",0.020,5.370,0.925,0.432,"signaling receptor binding"),
c("GO:0005509","calcium ion binding",1.076,5.234,0.956,0.056,"calcium ion binding"),
c("GO:0070405","ammonium ion binding",0.000,3.083,0.974,0.132,"calcium ion binding"),
c("GO:0031210","phosphatidylcholine binding",0.005,3.083,0.965,0.197,"calcium ion binding"),
c("GO:1901681","sulfur compound binding",0.544,3.467,0.972,0.056,"sulfur compound binding"));

stuff <- data.frame(revigo.data);
names(stuff) <- revigo.names;

stuff$value <- as.numeric( as.character(stuff$value) );
stuff$freqInDbPercent <- as.numeric( as.character(stuff$freqInDbPercent) );
stuff$uniqueness <- as.numeric( as.character(stuff$uniqueness) );
stuff$dispensability <- as.numeric( as.character(stuff$dispensability) );

# check the tmPlot command documentation for all possible parameters - there are a lot more
treemap(
  stuff,
  index = c("representative","description"),
  vSize = "value",
  type = "categorical",
  vColor = "representative",
  fontsize.labels=c(16,14),
  title = "REVIGO TreeMap",
  inflate.labels = FALSE,      # set this to TRUE for space-filling group labels - good for posters
  lowerbound.cex.labels = 0,
  palette = "Set1",
  align.labels=list(
        c("center", "top"), 
        c("center", "bottom")
        ),# try to draw as many labels as possible (still, some small squares may not get a label)
  position.legend = "none"
)
```


## {-}

***
```{r excel_anova}
a <- anova_table %>%
  as.data.frame() %>%
  relocate(starts_with("T"), .after = last_col()) %>% 
  select(-Majority.protein.IDs) %>% 
  relocate(c(X.Log.ANOVA.p.value, ANOVA.q.value), .after = Gene.names) %>% 
  relocate(Gene.names, .after = Cluster) %>% 
  dplyr::rename(ANOVA.p.value = X.Log.ANOVA.p.value) %>% 
  dplyr::rename_with(., ~ gsub(pattern = "^", replacement = "Z_Score_", .), starts_with("T")) %>%
  mutate(across(contains(".value"), format, scientific = T, digits = 2)) %>% 
  mutate(across(contains("Z_Score_"), .fns = ~ round(.x, digits = 2))) 

test <- "ANOVA_results"

addWorksheet(excel, sheetName = test, gridLines = F)

writeDataTable(excel, sheet = test, x = a, keepNA = T, na.string = "NaN")

n_row <- a %>% nrow() + 1

n_col <- a %>% ncol()

setColWidths(excel, sheet = test, cols = 1:n_col, widths = 21)

addStyle(excel, sheet = test, style = header_style, rows = 1, cols = 1:n_col, gridExpand = T)

addStyle(excel, sheet = test, style = body_style, rows = 2:n_row, cols = 1:n_col, gridExpand = T)
```

```{r save_excel}
saveWorkbook(excel, here::here("output/results_T2_vs_TME2_sub-populations.xlsx"), overwrite = T)
```
