---
title: "T2 vs TME2 comparison"
author: "Fabio Bedin | MS-Unit"
output: html_document
---

# Ovarian cancer dataset

***

```{r librerie, message=FALSE, warning=FALSE}
#set.seed(20210520)
knitr::opts_chunk$set(fig.align = "center")
library("MBQN")
library("dplyr")
library("tidyr")
library("DEP")
library("SummarizedExperiment")
library("preprocessCore")
library("tibble")
library("ggplot2")
library("enrichR")
library("DT")
library("stringr")
library("patchwork")
```

```{r Custom Functions}

source(here::here("code/custom_functions.R"))

```

## Introduction

***

For this analysis i will focus in all the possible combination between **T2** and **TME2** populations, where different sub-populations can be identify as sensivie **(S)**, intermediate sensitive **(IS)** and resistant **(R)**.

```{r input_Data}
data <- read.csv(here::here("data/proteinGroups.txt"), header = TRUE,stringsAsFactors = FALSE, sep = "\t")

expdesign <- read.table(here::here("data/expdesign_SI_S_R.tsv"), header = T, stringsAsFactors = F)

anova_table <- read.table(here::here("data/anova_table_t2_tme2.txt"), header = T, stringsAsFactors = F, sep = "\t")

conditions<-c("T2_IS","T2_S","T2_R","TME2_IS","TME2_S","TME2_R")

expdesign <- subset(expdesign, condition %in% conditions)
```

## Data wrangling

***

First, we need to wrangling the original dataset. From the **proteinGroups** obtained by **MaxQuant** software, I remove proteins that are marked form the software as *potential contaminant*, *only identify by site* and *reverse*.

```{r Data wrangling}
data <- data[data$Reverse != "+" & data$Potential.contaminant != "+" & data$Only.identified.by.site != "+",]

data_unique <- make_unique(data, "Gene.names", "Protein.IDs", delim = ";")

columns<-match(paste("LFQ.intensity.",expdesign$label,sep=""),colnames(data_unique))

data_se <- make_se(data_unique, columns, expdesign)
```

To understand the structure of the dataset and define a strategy, I use different quality control polts:

### Quality control Plots {.tabset .tabset-fade .tabset-pills}

All this plot can also hepl to decide the strategy to filter the missing data.

#### Frequency plot

```{r plot_freq, fig.height = 7, fig.width = 12}
plot_frequency(data_se)
```

#### Number of protein per sample

```{r plot numbers, fig.height = 7, fig.width = 12}
plot_numbers(data_se)
```

#### Distibution of each sample

```{r plot_norm, fig.height = 10, fig.width = 10}
plot_normalization(data_se)
```

## The problem of missing data

***

In the original dataset, the percentage of missing data is **`r round((sum(is.na(assay(data_se)))/prod(dim(assay(data_se))))*100, digits = 2)`%** and can be visualized with the following heatmap.

```{r plot_mv, message=FALSE, fig.height = 7, fig.width = 12}
plot_missval(data_se)
```

## Filter missing data

***

To reduce the percentage of missing data I have decided to filter samples with less then **2000** identify proteins.
Afrer that, I also use a second strong filter to remove all the proteins that are not identify in all replicates in at least 1 groups.

```{r data_filt}
data_filt <- filter_bad_samples(data_se, thr = 2000)

data_filt <- filter_missval(data_filt, thr = 0)
```


These plot show the effect of the filters. Now we reduce the percetage of missing data in the dataset down to: **`r round((sum(is.na(assay(data_filt)))/prod(dim(assay(data_filt))))*100, digits = 2)`%**.

### Plots with filters {.tabset .tabset-fade .tabset-pills}

#### protein coverage

```{r plot_coverage}
plot_coverage(data_filt)
```

#### Number of protein per sample

```{r plot_numbers_filt}
plot_numbers(data_filt)
```

#### Distibution

```{r plot_norm_filt, fig.height = 8, fig.width = 8}
plot_normalization(data_filt)
```

#### Plot missing data

```{r plot_mv_filt, message=FALSE}
plot_missval(data_filt)
```

## Imputation

***

An idealized version of a label-free discovery mass spectrometry proteomics experiment would provide absolute abundance measurements for a whole proteome, across varying conditions. Unfortunately, this ideal is not realized. Measurements are made on peptides requiring an inferential step to obtain protein level estimates. The inference is complicated by experimental factors that necessitate relative abundance estimation and result in widespread **non-ignorable missing data**. Relative abundance on the log scale takes the form of parameter contrasts. In a complete-case analysis, contrast estimates may be biased by missing data and a substantial amount of useful information will often go unused.

To avoid this problem, we impute missing data using random draws from a manually defined left-shifted Gaussian distribution.


```{r mixed_imputation, message=FALSE, warning=FALSE, results='hide'}
data_imp_man <- impute(data_filt, fun = "man", shift = 1.8, scale = 0.3)
```

This plot show the effect of imputation, as expected a new population of low intensity values as emerged.

```{r plot_imputation}
plot_imputation(data_filt, data_imp_man)
```

## Differential enrichment analysis

***

Protein-wise linear models combined with empirical Bayes statistics are used for the differential enrichment analysis (or differential expression analysis). I test all possible combination between T1 S, IS and R vs T2 S, IS, and R.

```{r test_diff}
data_diff <- test_diff(data_imp_man, type = "all")
```

I define sifnificant protein with **FDR = 0.01** and **FC = 1**

```{r Diff_prot}
dep <- add_rejections(data_diff, alpha = 0.01, lfc = 1)

results<-get_results(dep)
```
 
There are **`r results %>% filter(significant) %>% nrow()`** significant proteins within all comparisons.
The following table shows the results:

```{r significant_table}
results %>% 
  filter(significant) %>%
  select(-significant, -ends_with("centered")) %>% 
  pivot_longer(
   cols = ends_with(c("p.val", "p.adj", "significant", "ratio")),
   names_to = c("comparison", ".value"),
   names_pattern = "(.*_.*_.*_.*_.*)_(.*)") %>% 
  arrange(name, p.adj) %>% 
  relocate(comparison) %>% 
  mutate(across(ends_with(c("p.val", "p.adj")), format, scientific = T, digits = 2)) %>% 
  mutate(significant = str_to_title(significant)) %>% 
  DT::datatable(options = list(
  columnDefs = list(list(className = 'dt-center', targets = 1:7)),
  pageLength = 15))
```


### Visualization of the results {.tabset .tabset-fade .tabset-pills}

***

The results from the analysis can be easily visualized by a number of plot functions.

#### PCA

```{r PCA, message=FALSE, warning=FALSE, fig.height = 8, fig.width = 8}
plot_pca(dep, x = 1, y = 2, n = 500, point_size = 4, indicate = "condition")
```


#### Correlation matrix

```{r plot_cor, fig.height = 8}
plot_cor(dep, significant = TRUE, lower = 0, upper = 1, pal = "Reds")
```


#### Overview all volcano plot

```{r all_volcano, warning=FALSE, fig.height = 8, fig.width = 8}
cont <- "T2_IS_vs_T2_R"
p1 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T2_IS_vs_T2_S"
p2 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T2_IS_vs_TME2_IS"
p3 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T2_IS_vs_TME2_R"
p4 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T2_IS_vs_TME2_S"
p5 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T2_R_vs_T2_S"
p6 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T2_R_vs_TME2_IS"
p7 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T2_R_vs_TME2_R"
p8 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T2_R_vs_TME2_S"
p9 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T2_S_vs_TME2_IS"
p10 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T2_S_vs_TME2_R"
p11 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "T2_S_vs_TME2_S"
p12 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "TME2_IS_vs_TME2_R"
p13 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "TME2_IS_vs_TME2_S"
p14 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
cont <- "TME2_R_vs_TME2_S"
p15 <- plot_volcano(dep, contrast=cont, add_names=F,label_size=5,adjusted = F)
```

```{r plot_v_all, fig.height = 12, fig.width = 18}
(p1 | p2 | p3 | p4 | p5) /
  (p6 | p7 | p8 | p9 | p10) /
  (p11 | p12 | p13 | p14 | p15) 
```

#### Cluster Heatmap

```{r heatmap, fig.height = 15, fig.width = 10}
plot_heatmap(dep, type = "centered", kmeans = TRUE, k = 7, show_row_names = T, indicate = "condition",col_limit = 5)
```

```{r}
#cluster_table <- plot_heatmap(dep, type = "centered", kmeans = TRUE, k = 8, show_row_names = T, indicate = "condition",col_limit = 5, plot = F) %>% dplyr::rename(cluster = k) 

#write.table(cluster_table ,file = "cluster_table.txt", sep="\t", quote=FALSE, row.names=FALSE, col.names = T)
```


### Volcano Plots T2_R vs TME2_R {.tabset}

#### Significant

```{r V1,warning=FALSE, fig.height = 8, fig.width = 8}
cont <- "T2_R_vs_TME2_R"
plot_volcano(dep, contrast=cont, add_names=T,label_size=4,adjusted = F)
```

```{r T1}
results %>%  select(starts_with("name") | starts_with(cont)) %>% filter(across(ends_with("significant"))) %>% arrange(across(ends_with("p.adj"))) %>% mutate(across(2:3, format, scientific = T, digits = 2)) %>% dplyr::rename_with(~ tolower(gsub(paste0(cont,"_"), "", .x)), .cols = starts_with(cont)) %>%  DT::datatable()
```

#### Unique

```{r V2,warning=FALSE, fig.height = 8, fig.width = 8}
unique_vec <- c("AAMDC","AHDC1","ARL8A","CMTR1","DDX21","DR1","EYA4","GIPC1","MRPS36","MTPAP","PABPN1","PGAM5","ROR1","SMAD2","SNRPC","SNRPD3","TRIM56","UFD1L","UTS2","XRCC1","ZNF207","PRR12","SACS","BCKDHB","CBX5","DCP1B","LTBP4","METAP1","NCOR2","NDUFB10","NXF1","POLR2E","RAB43","SETD3","TGM1","TOMM22","VWA5A","APOBEC3C","SERPINA6","STX1A","UPF2")
#unique_vec <- c("AHDC1","UTS2","SACS","BCKDHB","CBX5","DCP1B","LTBP4","METAP1","NCOR2","NDUFB10","NXF1","POLR2E","RAB43","SETD3","TGM1","TOMM22","VWA5A","APOBEC3C","SERPINA6","STX1A","UPF2")
plot_volcano_2(dep, contrast=cont, add_names=T,label_size=4,adjusted = F)
```

```{r T2}
results %>%  select(starts_with("name") | starts_with(cont)) %>% filter(name %in% unique_vec) %>% arrange(across(ends_with("p.adj"))) %>% mutate(across(2:3, format, scientific = T, digits = 2)) %>% dplyr::rename_with(~ tolower(gsub(paste0(cont,"_"), "", .x)), .cols = starts_with(cont)) %>% DT::datatable()
```

### Volcano Plots T2_S vs TME2_S {.tabset}

#### Significant

```{r V5,warning=FALSE, fig.height = 8, fig.width = 8}
cont <- "T2_S_vs_TME2_S"
plot_volcano(dep, contrast=cont, add_names=T,label_size=4,adjusted = F)
```

```{r T5}
results %>%  select(starts_with("name") | starts_with(cont)) %>% filter(across(ends_with("significant"))) %>% arrange(across(ends_with("p.adj"))) %>% mutate(across(2:3, format, scientific = T, digits = 2)) %>% dplyr::rename_with(~ tolower(gsub(paste0(cont,"_"), "", .x)), .cols = starts_with(cont)) %>%  DT::datatable()
```

#### Unique

```{r V6,warning=FALSE, fig.height = 8, fig.width = 8}
unique_vec <- unique_vec <- c("AAMDC","AHDC1","ARL8A","CMTR1","DDX21","DR1","EYA4","GIPC1","MRPS36","MTPAP","PABPN1","PGAM5","ROR1","SMAD2","SNRPC","SNRPD3","TRIM56","UFD1L","UTS2","XRCC1","ZNF207","PRR12","SACS","BUD31","CDKN2A","CTSL","DCP1B","DENND4C","ESRP1","EWSR1","NCKAP1L","NIPSNAP1","RDH14","SEL1L","SRSF9","SYNGR2","TBCD","TFCP2","TMLHE","APOBEC3C","ECM1","GPNMB","PODN","SGCD","SRPX")
#unique_vec <- unique_vec <- c("MRPS36","SNRPC","BUD31","CDKN2A","CTSL","DCP1B","DENND4C","ESRP1","EWSR1","NCKAP1L","NIPSNAP1","RDH14","SEL1L","SRSF9","SYNGR2","TBCD","TFCP2","TMLHE","APOBEC3C","ECM1","GPNMB","PODN","SGCD","SRPX")
plot_volcano_2(dep, contrast=cont, add_names=T,label_size=4,adjusted = F)
```

```{r T6}
results %>%  select(starts_with("name") | starts_with(cont)) %>% filter(name %in% unique_vec) %>% arrange(across(ends_with("p.adj"))) %>% mutate(across(2:3, format, scientific = T, digits = 2)) %>% dplyr::rename_with(~ tolower(gsub(paste0(cont,"_"), "", .x)), .cols = starts_with(cont)) %>% DT::datatable()
```

### Volcano Plots T2_IS vs TME2_IS {.tabset}

#### Significant

```{r V3,warning=FALSE, fig.height = 8, fig.width = 8}
cont <- "T2_IS_vs_TME2_IS"
plot_volcano(dep, contrast=cont, add_names=T,label_size=4,adjusted = F)
```

```{r T3}
results %>%  select(starts_with("name") | starts_with(cont)) %>% filter(across(ends_with("significant"))) %>% arrange(across(ends_with("p.adj"))) %>% mutate(across(2:3, format, scientific = T, digits = 2)) %>% dplyr::rename_with(~ tolower(gsub(paste0(cont,"_"), "", .x)), .cols = starts_with(cont)) %>%  DT::datatable()
```

#### Unique

```{r V4,warning=FALSE, fig.height = 8, fig.width = 8}
unique_vec <- c("AAMDC","AHDC1","ARL8A","CMTR1","DDX21","DR1","EYA4","GIPC1","MRPS36","MTPAP","PABPN1","PGAM5","ROR1","SMAD2","SNRPC","SNRPD3","TRIM56","UFD1L","UTS2","XRCC1","ZNF207","PRR12","SACS","ATP5J2-PTCD1","C16orf13","GCA","GNG12","GNL3","MAVS","PARD6B","PDCD11","POLR2A","PTGES2","RTN1","SH3KBP1","SNRPA1","SYNGR2","TMEM201","TXLNA")
#unique_vec <- c("AAMDC","ATP5J2-PTCD1","C16orf13","GCA","GNG12","GNL3","MAVS","PARD6B","PDCD11","POLR2A","PTGES2","RTN1","SH3KBP1","SNRPA1","SYNGR2","TMEM201","TXLNA")
plot_volcano_2(dep, contrast=cont, add_names=T,label_size=4,adjusted = F)
```

```{r T4}
results %>%  select(starts_with("name") | starts_with(cont)) %>% filter(name %in% unique_vec) %>% arrange(across(ends_with("p.adj"))) %>% mutate(across(2:3, format, scientific = T, digits = 2)) %>% dplyr::rename_with(~ tolower(gsub(paste0(cont,"_"), "", .x)), .cols = starts_with(cont)) %>% DT::datatable()
```

## Gene ontology {.tabset .tabset-fade .tabset-pills}

***
```{r gene_name}
# ## Background list
# write.table(results$name,file = "gene_name_GO_background.txt", sep="\t", quote=FALSE, row.names=FALSE, col.names = F)
# 
# ## Significant list
# results %>% filter(significant) -> gene_name
# write.table(gene_name$name,file = "gene_name_GO_T2_TME2.txt", sep="\t", quote=FALSE, row.names=FALSE, col.names = F)
```

### Biological Process

```{r BP, warning=FALSE, fig.height = 7, fig.width = 10}
# A treemap R script produced by the REVIGO server at http://revigo.irb.hr/
# If you found REVIGO useful in your work, please cite the following reference:
# Supek F et al. "REVIGO summarizes and visualizes long lists of Gene Ontology
# terms" PLoS ONE 2011. doi:10.1371/journal.pone.0021800

# author: Anton Kratz <anton.kratz@gmail.com>, RIKEN Omics Science Center, Functional Genomics Technology Team, Japan
# created: Fri, Nov 02, 2012  7:25:52 PM
# last change: Fri, Nov 09, 2012  3:20:01 PM

# -----------------------------------------------------------------------------
# If you don't have the treemap package installed, uncomment the following line:
# install.packages( "treemap" );
library(treemap) 								# treemap package by Martijn Tennekes

# Set the working directory if necessary
# setwd("C:/Users/username/workingdir");

# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","freqInDbPercent","value","uniqueness","dispensability","representative");
revigo.data <- rbind(c("GO:0002526","acute inflammatory response",0.012,3.142,1.000,0.000,"acute inflammatory response"));

stuff <- data.frame(revigo.data);
names(stuff) <- revigo.names;

stuff$value <- as.numeric( as.character(stuff$value) );
stuff$freqInDbPercent <- as.numeric( as.character(stuff$freqInDbPercent) );
stuff$uniqueness <- as.numeric( as.character(stuff$uniqueness) );
stuff$dispensability <- as.numeric( as.character(stuff$dispensability) );


# check the tmPlot command documentation for all possible parameters - there are a lot more
treemap(
  stuff,
  index = c("representative","description"),
  vSize = "value",
  type = "categorical",
  vColor = "representative",
  fontsize.labels=c(16,14),
  title = "REVIGO TreeMap",
  inflate.labels = FALSE,      # set this to TRUE for space-filling group labels - good for posters
  lowerbound.cex.labels = 0,
  palette = "Set1",
  align.labels=list(
        c("center", "top"), 
        c("center", "bottom")
        ),# try to draw as many labels as possible (still, some small squares may not get a label)
  position.legend = "none"
)




```

### Molecular funtion

```{r MF, warning=FALSE, fig.height = 7, fig.width = 10}
# A treemap R script produced by the REVIGO server at http://revigo.irb.hr/
# If you found REVIGO useful in your work, please cite the following reference:
# Supek F et al. "REVIGO summarizes and visualizes long lists of Gene Ontology
# terms" PLoS ONE 2011. doi:10.1371/journal.pone.0021800

# author: Anton Kratz <anton.kratz@gmail.com>, RIKEN Omics Science Center, Functional Genomics Technology Team, Japan
# created: Fri, Nov 02, 2012  7:25:52 PM
# last change: Fri, Nov 09, 2012  3:20:01 PM

# -----------------------------------------------------------------------------
# If you don't have the treemap package installed, uncomment the following line:
# install.packages( "treemap" );
library(treemap) 								# treemap package by Martijn Tennekes

# Set the working directory if necessary
# setwd("C:/Users/username/workingdir");

# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","freqInDbPercent","value","uniqueness","dispensability","representative");
revigo.data <- rbind(c("GO:0071813","lipoprotein particle binding",0.004,3.277,0.264,0.000,"lipoprotein particle binding"),
c("GO:0071814","protein-lipid complex binding",0.004,3.277,0.264,0.487,"lipoprotein particle binding"));

stuff <- data.frame(revigo.data);
names(stuff) <- revigo.names;

stuff$value <- as.numeric( as.character(stuff$value) );
stuff$freqInDbPercent <- as.numeric( as.character(stuff$freqInDbPercent) );
stuff$uniqueness <- as.numeric( as.character(stuff$uniqueness) );
stuff$dispensability <- as.numeric( as.character(stuff$dispensability) );


# check the tmPlot command documentation for all possible parameters - there are a lot more
treemap(
  stuff,
  index = c("representative","description"),
  vSize = "value",
  type = "categorical",
  vColor = "representative",
  fontsize.labels=c(16,14),
  title = "REVIGO TreeMap",
  inflate.labels = FALSE,      # set this to TRUE for space-filling group labels - good for posters
  lowerbound.cex.labels = 0,
  palette = "Set1",
  align.labels=list(
        c("center", "top"), 
        c("center", "bottom")
        ),# try to draw as many labels as possible (still, some small squares may not get a label)
  position.legend = "none"
)




```


## Multiple sample ANOVA test {.tabset .tabset-fade .tabset-pills}

***

In this analysis, instead of comparing each possible condition individually, I used a multiple sample **ANOVA** test to observe the overall pattern of change.

```{r ANOVA, out.width = "200%"}
knitr::include_graphics(here::here("data/HeatMap_t2_tme2.png"))
```

### Cluster Table

Now i select only the 4 bigger cluster:

```{r ANOVA_table}
# anova_table %>%
#   group_by(Cluster) %>%
#   summarise(n = n())

name_cluster_order <- anova_table %>% 
  select(starts_with("T")) %>% colnames()

anova_table %>% 
  select(!c(Majority.protein.IDs, ANOVA.Significant, Protein.names, Protein.IDs)) %>% 
  dplyr::rename(Log.ANOVA.p.value = X.Log.ANOVA.p.value) %>% 
  pivot_longer(cols = starts_with("T"), names_to = "Samples", values_to = "Intensity") %>% 
  mutate(across(contains(c(".value", "Intensity")), format, scientific = T, digits = 2)) %>% 
  filter(Cluster == c("Cluster -691", "Cluster -687", "Cluster -689", "Cluster -690")) %>% 
  mutate(Cluster = case_when(Cluster == "Cluster -691" ~ "Cluster 1",
                             Cluster == "Cluster -687" ~ "Cluster 2",
                             Cluster == "Cluster -689" ~ "Cluster 3",
                             Cluster == "Cluster -690" ~ "Cluster 4")) %>% 
  relocate(Cluster, Samples, Intensity) %>% 
  DT::datatable()
  
```


### Cluster Profile {.tabset}

#### Cluster 1

```{r Cluster_1, warning=FALSE, message=FALSE, fig.height = 8, fig.width = 8}
anova_table %>% 
  filter(Cluster == "Cluster -691") %>% 
  select(starts_with("T") | contains("Gene")) %>% 
  pivot_longer(!Gene.names, names_to ="Sample", values_to = "Intensity") %>% 
  ggplot(mapping = aes(x = Sample, y = Intensity, group = Gene.names)) + 
  geom_line(color = "turquoise") + 
  geom_smooth(aes(group = 1)) + 
  scale_x_discrete(limits = name_cluster_order) +
  theme(axis.title = element_text(size = 16, face = "bold"),
        axis.text = element_text(size = 14), 
        axis.text.x = element_text(angle = 90), 
        plot.title = element_text(size = 20, face = "bold", hjust = 0.5),
        panel.background = element_rect(fill = NA)) +
  labs(title = "Cluster 1", x = "Samples", y = "log2 Intensity") + 
  theme(axis.line = element_line(size = 0.4, linetype = "solid"), 
        axis.text.x = element_text(vjust = 0.5))
```

#### Cluster 2

```{r Cluster_2, warning=FALSE, message=FALSE, fig.height = 8, fig.width = 8}
anova_table %>% 
  filter(Cluster == "Cluster -687") %>% 
  select(starts_with("T") | contains("Gene")) %>% 
  pivot_longer(!Gene.names, names_to ="Sample", values_to = "Intensity") %>% 
  ggplot(mapping = aes(x = Sample, y = Intensity, group = Gene.names)) + 
  geom_line(color = "khaki4") + 
  geom_smooth(aes(group = 1)) + 
  scale_x_discrete(limits = name_cluster_order) +
  theme(axis.title = element_text(size = 16, face = "bold"),
        axis.text = element_text(size = 14), 
        axis.text.x = element_text(angle = 90), 
        plot.title = element_text(size = 20, face = "bold", hjust = 0.5),
        panel.background = element_rect(fill = NA)) +
  labs(title = "Cluster 2", x = "Samples", y = "log2 Intensity") + 
  theme(axis.line = element_line(size = 0.4, linetype = "solid"), 
        axis.text.x = element_text(vjust = 0.5))
```

#### Cluster 3

```{r Cluster_3, warning=FALSE, message=FALSE, fig.height = 8, fig.width = 8}
anova_table %>% 
  filter(Cluster == "Cluster -689") %>% 
  select(starts_with("T") | contains("Gene")) %>% 
  pivot_longer(!Gene.names, names_to ="Sample", values_to = "Intensity") %>% 
  ggplot(mapping = aes(x = Sample, y = Intensity, group = Gene.names)) + 
  geom_line(color = "blue") + 
  geom_smooth(aes(group = 1)) + 
  scale_x_discrete(limits = name_cluster_order) +
  theme(axis.title = element_text(size = 16, face = "bold"),
        axis.text = element_text(size = 14), 
        axis.text.x = element_text(angle = 90), 
        plot.title = element_text(size = 20, face = "bold", hjust = 0.5),
        panel.background = element_rect(fill = NA)) +
  labs(title = "Cluster 3", x = "Samples", y = "log2 Intensity") + 
  theme(axis.line = element_line(size = 0.4, linetype = "solid"), 
        axis.text.x = element_text(vjust = 0.5))
```

#### Cluster 4

```{r Cluster_4, warning=FALSE, message=FALSE, fig.height = 8, fig.width = 8}
anova_table %>% 
  filter(Cluster == "Cluster -690") %>% 
  select(starts_with("T") | contains("Gene")) %>% 
  pivot_longer(!Gene.names, names_to ="Sample", values_to = "Intensity") %>% 
  ggplot(mapping = aes(x = Sample, y = Intensity, group = Gene.names)) + 
  geom_line(color = "firebrick4") + 
  geom_smooth(aes(group = 1)) + 
  scale_x_discrete(limits = name_cluster_order) +
  theme(axis.title = element_text(size = 16, face = "bold"),
        axis.text = element_text(size = 14), 
        axis.text.x = element_text(angle = 90), 
        plot.title = element_text(size = 20, face = "bold", hjust = 0.5),
        panel.background = element_rect(fill = NA)) +
  labs(title = "Cluster 4", x = "Samples", y = "log2 Intensity") + 
  theme(axis.line = element_line(size = 0.4, linetype = "solid"), 
        axis.text.x = element_text(vjust = 0.5))
```

### Gene Ontology {.tabset}

```{r}
b <- anova_table %>% select(Gene.names) %>% mutate(Gene.names = gsub(pattern = ";.*$", replacement = "", Gene.names)) %>% pull(Gene.names)

write.table(b,file = "gene_name_BP_anova.txt", sep="\t", quote=FALSE, row.names=FALSE, col.names = F)
```

#### BP

```{r BP_anova, warning=FALSE, fig.height = 7, fig.width = 10}
library(treemap) 								# treemap package by Martijn Tennekes

# Set the working directory if necessary
# setwd("C:/Users/username/workingdir");

# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.

revigo.names <- c("term_ID","description","freqInDbPercent","value","uniqueness","dispensability","representative");
revigo.data <- rbind(c("GO:0000726","(obsolete) non-recombinational repair",0.130,3.336,1.000,0.000,"(obsolete) non-recombinational repair"),
c("GO:0002376","immune system process",0.876,4.234,1.000,0.000,"immune system process"),
c("GO:0002673","regulation of acute inflammatory response",0.005,6.194,0.718,0.000,"regulation of acute inflammatory response"),
c("GO:0070613","regulation of protein processing",0.012,5.699,0.779,0.129,"regulation of acute inflammatory response"),
c("GO:0051346","negative regulation of hydrolase activity",0.274,3.491,0.801,0.152,"regulation of acute inflammatory response"),
c("GO:0032956","regulation of actin cytoskeleton organization",0.219,3.411,0.834,0.190,"regulation of acute inflammatory response"),
c("GO:0033238","regulation of cellular amine metabolic process",0.018,3.435,0.840,0.206,"regulation of acute inflammatory response"),
c("GO:0032970","regulation of actin filament-based process",0.224,3.078,0.834,0.213,"regulation of acute inflammatory response"),
c("GO:1903317","regulation of protein maturation",0.013,5.699,0.803,0.395,"regulation of acute inflammatory response"),
c("GO:2000257","regulation of protein activation cascade",0.002,5.312,0.710,0.420,"regulation of acute inflammatory response"),
c("GO:0002920","regulation of humoral immune response",0.011,5.312,0.620,0.459,"regulation of acute inflammatory response"),
c("GO:0030162","regulation of proteolysis",0.401,3.449,0.760,0.491,"regulation of acute inflammatory response"),
c("GO:0032101","regulation of response to external stimulus",0.211,4.423,0.697,0.571,"regulation of acute inflammatory response"),
c("GO:2001185","regulation of CD8-positive, alpha-beta T cell activation",0.002,3.917,0.685,0.588,"regulation of acute inflammatory response"),
c("GO:0030449","regulation of complement activation",0.005,5.312,0.609,0.662,"regulation of acute inflammatory response"),
c("GO:0015850","organic hydroxy compound transport",0.111,3.561,0.965,0.000,"organic hydroxy compound transport"),
c("GO:0016192","vesicle-mediated transport",1.371,3.423,0.974,0.243,"organic hydroxy compound transport"),
c("GO:0015918","sterol transport",0.036,3.561,0.949,0.369,"organic hydroxy compound transport"),
c("GO:0042590","antigen processing and presentation of exogenous peptide antigen via MHC class I",0.001,4.298,0.764,0.000,"antigen processing and presentation of exogenous peptide antigen via MHC class I"),
c("GO:0006959","humoral immune response",0.043,3.387,0.778,0.495,"antigen processing and presentation of exogenous peptide antigen via MHC class I"),
c("GO:0002252","immune effector process",0.068,3.163,0.800,0.625,"antigen processing and presentation of exogenous peptide antigen via MHC class I"),
c("GO:0002250","adaptive immune response",0.057,3.193,0.775,0.651,"antigen processing and presentation of exogenous peptide antigen via MHC class I"),
c("GO:0043062","extracellular structure organization",0.064,8.893,0.987,0.000,"extracellular structure organization"),
c("GO:0030198","extracellular matrix organization",0.062,6.714,0.987,0.325,"extracellular structure organization"),
c("GO:0048856","anatomical structure development",2.035,3.381,1.000,0.000,"anatomical structure development"),
c("GO:0072376","protein activation cascade",0.004,6.863,0.932,0.000,"protein activation cascade"),
c("GO:0006022","aminoglycan metabolic process",1.120,3.036,0.931,0.103,"protein activation cascade"),
c("GO:0043687","post-translational protein modification",0.028,4.551,0.972,0.141,"protein activation cascade"),
c("GO:0006954","inflammatory response",0.094,3.478,0.925,0.207,"protein activation cascade"),
c("GO:0006303","double-strand break repair via nonhomologous end joining",0.044,3.336,0.914,0.402,"protein activation cascade"),
c("GO:0006024","glycosaminoglycan biosynthetic process",0.773,3.077,0.904,0.524,"protein activation cascade"),
c("GO:0060337","type I interferon signaling pathway",0.002,3.243,0.683,0.548,"protein activation cascade"),
c("GO:0019835","cytolysis",0.061,4.407,0.999,0.007,"cytolysis"));

stuff <- data.frame(revigo.data);
names(stuff) <- revigo.names;

stuff$value <- as.numeric( as.character(stuff$value) );
stuff$freqInDbPercent <- as.numeric( as.character(stuff$freqInDbPercent) );
stuff$uniqueness <- as.numeric( as.character(stuff$uniqueness) );
stuff$dispensability <- as.numeric( as.character(stuff$dispensability) );

# check the tmPlot command documentation for all possible parameters - there are a lot more
treemap(
  stuff,
  index = c("representative","description"),
  vSize = "value",
  type = "categorical",
  vColor = "representative",
  fontsize.labels=c(16,14),
  title = "REVIGO TreeMap",
  inflate.labels = FALSE,      # set this to TRUE for space-filling group labels - good for posters
  lowerbound.cex.labels = 0,
  palette = "Set1",
  align.labels=list(
        c("center", "top"), 
        c("center", "bottom")
        ),# try to draw as many labels as possible (still, some small squares may not get a label)
  position.legend = "none"
)

```


#### MF

```{r MF_anova, warning=FALSE, fig.height = 7, fig.width = 10}
library(treemap) 								# treemap package by Martijn Tennekes

# Set the working directory if necessary
# setwd("C:/Users/username/workingdir");

# --------------------------------------------------------------------------
# Here is your data from REVIGO. Scroll down for plot configuration options.
revigo.names <- c("term_ID","description","freqInDbPercent","value","uniqueness","dispensability","representative");
revigo.data <- rbind(c("GO:0005201","extracellular matrix structural constituent",0.035,9.466,1.000,0.000,"extracellular matrix structural constituent"),
c("GO:0008236","serine-type peptidase activity",1.282,3.211,0.742,0.000,"serine-type peptidase activity"),
c("GO:0017171","serine hydrolase activity",1.283,3.211,0.872,0.303,"serine-type peptidase activity"),
c("GO:0046977","TAP binding",0.001,3.243,1.000,0.000,"TAP binding"),
c("GO:0061134","peptidase regulator activity",0.179,4.565,0.480,0.000,"peptidase regulator activity"),
c("GO:0098772","molecular function regulator",1.022,3.416,1.000,0.000,"molecular function regulator"));

stuff <- data.frame(revigo.data);
names(stuff) <- revigo.names;

stuff$value <- as.numeric( as.character(stuff$value) );
stuff$freqInDbPercent <- as.numeric( as.character(stuff$freqInDbPercent) );
stuff$uniqueness <- as.numeric( as.character(stuff$uniqueness) );
stuff$dispensability <- as.numeric( as.character(stuff$dispensability) );

# check the tmPlot command documentation for all possible parameters - there are a lot more
treemap(
  stuff,
  index = c("representative","description"),
  vSize = "value",
  type = "categorical",
  vColor = "representative",
  fontsize.labels=c(16,14),
  title = "REVIGO TreeMap",
  inflate.labels = FALSE,      # set this to TRUE for space-filling group labels - good for posters
  lowerbound.cex.labels = 0,
  palette = "Set1",
  align.labels=list(
        c("center", "top"), 
        c("center", "bottom")
        ),# try to draw as many labels as possible (still, some small squares may not get a label)
  position.legend = "none"
)
```

## {-}

***
